{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 158,
   "metadata": {},
   "outputs": [],
   "source": [
    "## IMPORT ##\n",
    "\n",
    "import pandas as pd\n",
    "from pandas import read_csv\n",
    "from Bio import Entrez\n",
    "import os\n",
    "import urllib\n",
    "import requests"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assembly</th>\n",
       "      <th>SRA_Accession</th>\n",
       "      <th>Submitters</th>\n",
       "      <th>Release_Date</th>\n",
       "      <th>Species</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Family</th>\n",
       "      <th>Molecule_type</th>\n",
       "      <th>Sequence_Type</th>\n",
       "      <th>Nuc_Completeness</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Segment</th>\n",
       "      <th>Publications</th>\n",
       "      <th>Geo_Location</th>\n",
       "      <th>USA</th>\n",
       "      <th>Host</th>\n",
       "      <th>Isolation_Source</th>\n",
       "      <th>Collection_Date</th>\n",
       "      <th>BioSample</th>\n",
       "      <th>GenBank_Title</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCF_009685705.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Brinks,E.</td>\n",
       "      <td>2021-03-08T13:59:31.973Z</td>\n",
       "      <td>Lactococcus phage P1048</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASM968570v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_002620465.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mahony,J., Moscarelli,A., Kelleher,P., Lugli,G...</td>\n",
       "      <td>2021-03-08T13:59:13.200Z</td>\n",
       "      <td>Lactococcus phage AM4</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Italy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASM262046v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_002620665.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mahony,J., Moscarelli,A., Kelleher,P., Lugli,G...</td>\n",
       "      <td>2021-03-08T13:58:52.907Z</td>\n",
       "      <td>Lactococcus phage LW81</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Italy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASM262066v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCF_002620365.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Mahony,J., Moscarelli,A., Kelleher,P., Lugli,G...</td>\n",
       "      <td>2021-03-08T13:58:33.207Z</td>\n",
       "      <td>Lactococcus phage AM1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>Italy</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASM262036v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCF_002625325.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Gilcrease,E.B., Casjens,S.R.</td>\n",
       "      <td>2021-03-08T13:57:57.437Z</td>\n",
       "      <td>Escherichia phage D6</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Myoviridae</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ASM262532v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3460</th>\n",
       "      <td>GCF_000836965.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Juhala,R.J., Ford,M.E., Duda,R.L., Youlton,A.,...</td>\n",
       "      <td>2015-02-12T16:29:08.923Z</td>\n",
       "      <td>Escherichia virus HK022</td>\n",
       "      <td>Hendrixvirus</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ViralProj14048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3461</th>\n",
       "      <td>GCF_000836945.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Miller,E.S., Kutter,E., Mosig,G., Arisaka,F., ...</td>\n",
       "      <td>2015-02-12T16:28:59.173Z</td>\n",
       "      <td>Escherichia virus T4</td>\n",
       "      <td>Tequatrovirus</td>\n",
       "      <td>Myoviridae</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>91.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ViralProj14044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3462</th>\n",
       "      <td>GCF_000836925.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hill,D.F., Hughes,G., McNaughton,J.C., Stockwe...</td>\n",
       "      <td>2015-02-12T16:28:52.943Z</td>\n",
       "      <td>Escherichia virus If1</td>\n",
       "      <td>Infulavirus</td>\n",
       "      <td>Inoviridae</td>\n",
       "      <td>ssDNA(+)</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ViralProj14039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3463</th>\n",
       "      <td>GCF_000836905.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Christie,G.E., Temple,L.M., Bartlett,B.A., Goo...</td>\n",
       "      <td>2015-02-12T16:28:47.170Z</td>\n",
       "      <td>Escherichia virus P2</td>\n",
       "      <td>Peduovirus</td>\n",
       "      <td>Myoviridae</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>22.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ViralProj14035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3464</th>\n",
       "      <td>GCF_000836805.1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Hsia,R.-c., Bavoil,P.M.</td>\n",
       "      <td>2015-02-12T16:28:10.820Z</td>\n",
       "      <td>Chlamydia virus CPG1</td>\n",
       "      <td>Chlamydiamicrovirus</td>\n",
       "      <td>Microviridae</td>\n",
       "      <td>ssDNA(+)</td>\n",
       "      <td>RefSeq</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Chlamydia psittaci</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ViralProj14012</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3465 rows × 20 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Assembly SRA_Accession  \\\n",
       "0     GCF_009685705.1           NaN   \n",
       "1     GCF_002620465.1           NaN   \n",
       "2     GCF_002620665.1           NaN   \n",
       "3     GCF_002620365.1           NaN   \n",
       "4     GCF_002625325.1           NaN   \n",
       "...               ...           ...   \n",
       "3460  GCF_000836965.1           NaN   \n",
       "3461  GCF_000836945.1           NaN   \n",
       "3462  GCF_000836925.1           NaN   \n",
       "3463  GCF_000836905.1           NaN   \n",
       "3464  GCF_000836805.1           NaN   \n",
       "\n",
       "                                             Submitters  \\\n",
       "0                                             Brinks,E.   \n",
       "1     Mahony,J., Moscarelli,A., Kelleher,P., Lugli,G...   \n",
       "2     Mahony,J., Moscarelli,A., Kelleher,P., Lugli,G...   \n",
       "3     Mahony,J., Moscarelli,A., Kelleher,P., Lugli,G...   \n",
       "4                          Gilcrease,E.B., Casjens,S.R.   \n",
       "...                                                 ...   \n",
       "3460  Juhala,R.J., Ford,M.E., Duda,R.L., Youlton,A.,...   \n",
       "3461  Miller,E.S., Kutter,E., Mosig,G., Arisaka,F., ...   \n",
       "3462  Hill,D.F., Hughes,G., McNaughton,J.C., Stockwe...   \n",
       "3463  Christie,G.E., Temple,L.M., Bartlett,B.A., Goo...   \n",
       "3464                            Hsia,R.-c., Bavoil,P.M.   \n",
       "\n",
       "                  Release_Date                  Species                Genus  \\\n",
       "0     2021-03-08T13:59:31.973Z  Lactococcus phage P1048                  NaN   \n",
       "1     2021-03-08T13:59:13.200Z    Lactococcus phage AM4                  NaN   \n",
       "2     2021-03-08T13:58:52.907Z   Lactococcus phage LW81                  NaN   \n",
       "3     2021-03-08T13:58:33.207Z    Lactococcus phage AM1                  NaN   \n",
       "4     2021-03-08T13:57:57.437Z     Escherichia phage D6                  NaN   \n",
       "...                        ...                      ...                  ...   \n",
       "3460  2015-02-12T16:29:08.923Z  Escherichia virus HK022         Hendrixvirus   \n",
       "3461  2015-02-12T16:28:59.173Z     Escherichia virus T4        Tequatrovirus   \n",
       "3462  2015-02-12T16:28:52.943Z    Escherichia virus If1          Infulavirus   \n",
       "3463  2015-02-12T16:28:47.170Z     Escherichia virus P2           Peduovirus   \n",
       "3464  2015-02-12T16:28:10.820Z     Chlamydia virus CPG1  Chlamydiamicrovirus   \n",
       "\n",
       "            Family Molecule_type Sequence_Type  Nuc_Completeness Genotype  \\\n",
       "0     Siphoviridae         dsDNA        RefSeq               NaN      NaN   \n",
       "1     Siphoviridae         dsDNA        RefSeq               NaN      NaN   \n",
       "2     Siphoviridae         dsDNA        RefSeq               NaN      NaN   \n",
       "3     Siphoviridae         dsDNA        RefSeq               NaN      NaN   \n",
       "4       Myoviridae         dsDNA        RefSeq               NaN      NaN   \n",
       "...            ...           ...           ...               ...      ...   \n",
       "3460  Siphoviridae         dsDNA        RefSeq               NaN      NaN   \n",
       "3461    Myoviridae         dsDNA        RefSeq               NaN      NaN   \n",
       "3462    Inoviridae      ssDNA(+)        RefSeq               NaN      NaN   \n",
       "3463    Myoviridae         dsDNA        RefSeq               NaN      NaN   \n",
       "3464  Microviridae      ssDNA(+)        RefSeq               NaN      NaN   \n",
       "\n",
       "      Segment  Publications Geo_Location  USA                Host  \\\n",
       "0         NaN           NaN          NaN  NaN  Lactococcus lactis   \n",
       "1         NaN           1.0        Italy  NaN  Lactococcus lactis   \n",
       "2         NaN           1.0        Italy  NaN  Lactococcus lactis   \n",
       "3         NaN           1.0        Italy  NaN  Lactococcus lactis   \n",
       "4         NaN           NaN          NaN  NaN    Escherichia coli   \n",
       "...       ...           ...          ...  ...                 ...   \n",
       "3460      NaN           1.0          NaN  NaN                 NaN   \n",
       "3461      NaN          91.0          NaN  NaN    Escherichia coli   \n",
       "3462      NaN           NaN          NaN  NaN    Escherichia coli   \n",
       "3463      NaN          22.0          NaN  NaN    Escherichia coli   \n",
       "3464      NaN           NaN          NaN  NaN  Chlamydia psittaci   \n",
       "\n",
       "     Isolation_Source Collection_Date BioSample   GenBank_Title  \n",
       "0                 NaN             NaN       NaN     ASM968570v1  \n",
       "1                 NaN             NaN       NaN     ASM262046v1  \n",
       "2                 NaN             NaN       NaN     ASM262066v1  \n",
       "3                 NaN             NaN       NaN     ASM262036v1  \n",
       "4                 NaN             NaN       NaN     ASM262532v1  \n",
       "...               ...             ...       ...             ...  \n",
       "3460              NaN             NaN       NaN  ViralProj14048  \n",
       "3461              NaN             NaN       NaN  ViralProj14044  \n",
       "3462              NaN             NaN       NaN  ViralProj14039  \n",
       "3463              NaN             NaN       NaN  ViralProj14035  \n",
       "3464              NaN             NaN       NaN  ViralProj14012  \n",
       "\n",
       "[3465 rows x 20 columns]"
      ]
     },
     "execution_count": 2,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "seq = pd.read_csv('sequences.csv') \n",
    "seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "#new dataframe with individual columns\n",
    "new_seq = seq[['Assembly', 'Species', 'Molecule_type', 'Family', 'Host', 'GenBank_Title']].copy()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assembly</th>\n",
       "      <th>Species</th>\n",
       "      <th>Molecule_type</th>\n",
       "      <th>Family</th>\n",
       "      <th>Host</th>\n",
       "      <th>GenBank_Title</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCF_009685705.1</td>\n",
       "      <td>Lactococcus phage P1048</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM968570v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_002620465.1</td>\n",
       "      <td>Lactococcus phage AM4</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM262046v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_002620665.1</td>\n",
       "      <td>Lactococcus phage LW81</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM262066v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCF_002620365.1</td>\n",
       "      <td>Lactococcus phage AM1</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM262036v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>GCF_002625325.1</td>\n",
       "      <td>Escherichia phage D6</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Myoviridae</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>ASM262532v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3460</th>\n",
       "      <td>GCF_000836965.1</td>\n",
       "      <td>Escherichia virus HK022</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ViralProj14048</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3461</th>\n",
       "      <td>GCF_000836945.1</td>\n",
       "      <td>Escherichia virus T4</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Myoviridae</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>ViralProj14044</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3462</th>\n",
       "      <td>GCF_000836925.1</td>\n",
       "      <td>Escherichia virus If1</td>\n",
       "      <td>ssDNA(+)</td>\n",
       "      <td>Inoviridae</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>ViralProj14039</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3463</th>\n",
       "      <td>GCF_000836905.1</td>\n",
       "      <td>Escherichia virus P2</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Myoviridae</td>\n",
       "      <td>Escherichia coli</td>\n",
       "      <td>ViralProj14035</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3464</th>\n",
       "      <td>GCF_000836805.1</td>\n",
       "      <td>Chlamydia virus CPG1</td>\n",
       "      <td>ssDNA(+)</td>\n",
       "      <td>Microviridae</td>\n",
       "      <td>Chlamydia psittaci</td>\n",
       "      <td>ViralProj14012</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3465 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Assembly                  Species Molecule_type        Family  \\\n",
       "0     GCF_009685705.1  Lactococcus phage P1048         dsDNA  Siphoviridae   \n",
       "1     GCF_002620465.1    Lactococcus phage AM4         dsDNA  Siphoviridae   \n",
       "2     GCF_002620665.1   Lactococcus phage LW81         dsDNA  Siphoviridae   \n",
       "3     GCF_002620365.1    Lactococcus phage AM1         dsDNA  Siphoviridae   \n",
       "4     GCF_002625325.1     Escherichia phage D6         dsDNA    Myoviridae   \n",
       "...               ...                      ...           ...           ...   \n",
       "3460  GCF_000836965.1  Escherichia virus HK022         dsDNA  Siphoviridae   \n",
       "3461  GCF_000836945.1     Escherichia virus T4         dsDNA    Myoviridae   \n",
       "3462  GCF_000836925.1    Escherichia virus If1      ssDNA(+)    Inoviridae   \n",
       "3463  GCF_000836905.1     Escherichia virus P2         dsDNA    Myoviridae   \n",
       "3464  GCF_000836805.1     Chlamydia virus CPG1      ssDNA(+)  Microviridae   \n",
       "\n",
       "                    Host   GenBank_Title  \n",
       "0     Lactococcus lactis     ASM968570v1  \n",
       "1     Lactococcus lactis     ASM262046v1  \n",
       "2     Lactococcus lactis     ASM262066v1  \n",
       "3     Lactococcus lactis     ASM262036v1  \n",
       "4       Escherichia coli     ASM262532v1  \n",
       "...                  ...             ...  \n",
       "3460                 NaN  ViralProj14048  \n",
       "3461    Escherichia coli  ViralProj14044  \n",
       "3462    Escherichia coli  ViralProj14039  \n",
       "3463    Escherichia coli  ViralProj14035  \n",
       "3464  Chlamydia psittaci  ViralProj14012  \n",
       "\n",
       "[3465 rows x 6 columns]"
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "new_seq"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assembly</th>\n",
       "      <th>Species</th>\n",
       "      <th>Molecule_type</th>\n",
       "      <th>Family</th>\n",
       "      <th>Host</th>\n",
       "      <th>GenBank_Title</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>3442</th>\n",
       "      <td>GCF_000837825.1</td>\n",
       "      <td>Acholeplasma virus L2</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Plasmaviridae</td>\n",
       "      <td>Acholeplasma laidlawii</td>\n",
       "      <td>ViralProj14066</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>85</th>\n",
       "      <td>GCF_006964305.1</td>\n",
       "      <td>Achromobacter phage Motura</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Myoviridae</td>\n",
       "      <td>Achromobacter animicus</td>\n",
       "      <td>ASM696430v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1299</th>\n",
       "      <td>GCF_002211055.1</td>\n",
       "      <td>Achromobacter virus Axp3</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Podoviridae</td>\n",
       "      <td>Achromobacter xylosoxidans</td>\n",
       "      <td>ViralProj308019</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2367</th>\n",
       "      <td>GCF_000914775.1</td>\n",
       "      <td>Achromobacter virus JWAlpha</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Podoviridae</td>\n",
       "      <td>Achromobacter xylosoxidans</td>\n",
       "      <td>ViralProj240045</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1605</th>\n",
       "      <td>GCF_001551405.1</td>\n",
       "      <td>Achromobacter phage JWF</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Achromobacter xylosoxidans</td>\n",
       "      <td>ViralProj310937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3323</th>\n",
       "      <td>GCF_000846545.1</td>\n",
       "      <td>Mycobacterium virus L5</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ViralProj14459</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3421</th>\n",
       "      <td>GCF_000839625.1</td>\n",
       "      <td>Escherichia virus N15</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ViralProj14086</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3448</th>\n",
       "      <td>GCF_000837485.1</td>\n",
       "      <td>Psimunavirus psiM2</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ViralProj14160</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3459</th>\n",
       "      <td>GCF_000836985.1</td>\n",
       "      <td>Streptococcus virus Sfi11</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ViralProj14054</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3460</th>\n",
       "      <td>GCF_000836965.1</td>\n",
       "      <td>Escherichia virus HK022</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>NaN</td>\n",
       "      <td>ViralProj14048</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>3465 rows × 6 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "             Assembly                      Species Molecule_type  \\\n",
       "3442  GCF_000837825.1        Acholeplasma virus L2         dsDNA   \n",
       "85    GCF_006964305.1   Achromobacter phage Motura         dsDNA   \n",
       "1299  GCF_002211055.1     Achromobacter virus Axp3         dsDNA   \n",
       "2367  GCF_000914775.1  Achromobacter virus JWAlpha         dsDNA   \n",
       "1605  GCF_001551405.1      Achromobacter phage JWF         dsDNA   \n",
       "...               ...                          ...           ...   \n",
       "3323  GCF_000846545.1       Mycobacterium virus L5         dsDNA   \n",
       "3421  GCF_000839625.1        Escherichia virus N15         dsDNA   \n",
       "3448  GCF_000837485.1           Psimunavirus psiM2         dsDNA   \n",
       "3459  GCF_000836985.1    Streptococcus virus Sfi11         dsDNA   \n",
       "3460  GCF_000836965.1      Escherichia virus HK022         dsDNA   \n",
       "\n",
       "             Family                        Host    GenBank_Title  \n",
       "3442  Plasmaviridae      Acholeplasma laidlawii   ViralProj14066  \n",
       "85       Myoviridae      Achromobacter animicus      ASM696430v1  \n",
       "1299    Podoviridae  Achromobacter xylosoxidans  ViralProj308019  \n",
       "2367    Podoviridae  Achromobacter xylosoxidans  ViralProj240045  \n",
       "1605   Siphoviridae  Achromobacter xylosoxidans  ViralProj310937  \n",
       "...             ...                         ...              ...  \n",
       "3323   Siphoviridae                         NaN   ViralProj14459  \n",
       "3421   Siphoviridae                         NaN   ViralProj14086  \n",
       "3448   Siphoviridae                         NaN   ViralProj14160  \n",
       "3459   Siphoviridae                         NaN   ViralProj14054  \n",
       "3460   Siphoviridae                         NaN   ViralProj14048  \n",
       "\n",
       "[3465 rows x 6 columns]"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#sorting values by Host and Family\n",
    "new_seq.sort_values(by=['Host', 'Family'])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Assembly</th>\n",
       "      <th>Species</th>\n",
       "      <th>Molecule_type</th>\n",
       "      <th>Family</th>\n",
       "      <th>Host</th>\n",
       "      <th>GenBank_Title</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>GCF_009685705.1</td>\n",
       "      <td>Lactococcus phage P1048</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM968570v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>GCF_002620465.1</td>\n",
       "      <td>Lactococcus phage AM4</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM262046v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>GCF_002620665.1</td>\n",
       "      <td>Lactococcus phage LW81</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM262066v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>GCF_002620365.1</td>\n",
       "      <td>Lactococcus phage AM1</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM262036v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>120</th>\n",
       "      <td>GCF_009685685.1</td>\n",
       "      <td>Lactococcus phage P1045</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM968568v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>121</th>\n",
       "      <td>GCF_002611385.1</td>\n",
       "      <td>Lactococcus phage 62503</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM261138v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>122</th>\n",
       "      <td>GCF_002611285.1</td>\n",
       "      <td>Lactococcus phage 50902</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM261128v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>123</th>\n",
       "      <td>GCF_002758415.1</td>\n",
       "      <td>Lactococcus phage Q33</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM275841v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>124</th>\n",
       "      <td>GCF_009662695.1</td>\n",
       "      <td>Lactococcus phage P596</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM966269v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>125</th>\n",
       "      <td>GCF_002620505.1</td>\n",
       "      <td>Lactococcus phage AM6</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM262050v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>126</th>\n",
       "      <td>GCF_002620585.1</td>\n",
       "      <td>Lactococcus phage LW31</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM262058v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>908</th>\n",
       "      <td>GCF_002602185.1</td>\n",
       "      <td>Lactococcus virus ASCC273</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM260218v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>916</th>\n",
       "      <td>GCF_002602245.1</td>\n",
       "      <td>Lactococcus virus ASCC532</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM260224v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>928</th>\n",
       "      <td>GCF_002602225.1</td>\n",
       "      <td>Lactococcus virus ASCC465</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM260222v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>929</th>\n",
       "      <td>GCF_002602205.1</td>\n",
       "      <td>Lactococcus virus ASCC281</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM260220v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1059</th>\n",
       "      <td>GCF_002630605.1</td>\n",
       "      <td>Lactococcus virus CB20</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM263060v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1060</th>\n",
       "      <td>GCF_002630585.1</td>\n",
       "      <td>Lactococcus virus CB19</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ASM263058v1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1432</th>\n",
       "      <td>GCF_001746015.1</td>\n",
       "      <td>Lactococcus phage M5938</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj344188</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1441</th>\n",
       "      <td>GCF_001745855.1</td>\n",
       "      <td>Lactococcus phage 63301</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj344153</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1463</th>\n",
       "      <td>GCF_001745355.1</td>\n",
       "      <td>Lactococcus phage D4412</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj344189</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1467</th>\n",
       "      <td>GCF_001745195.1</td>\n",
       "      <td>Lactococcus phage 50101</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj344155</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1488</th>\n",
       "      <td>GCF_001744695.1</td>\n",
       "      <td>Lactococcus phage M6162</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj344187</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1496</th>\n",
       "      <td>GCF_001744535.1</td>\n",
       "      <td>Lactococcus phage 98201</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj344152</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1515</th>\n",
       "      <td>GCF_001744035.1</td>\n",
       "      <td>Lactococcus phage D4410</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj344190</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1517</th>\n",
       "      <td>GCF_001744015.1</td>\n",
       "      <td>Lactococcus phage M6165</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj344186</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1523</th>\n",
       "      <td>GCF_001743875.1</td>\n",
       "      <td>Lactococcus phage 28201</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj344156</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1677</th>\n",
       "      <td>GCF_001505235.1</td>\n",
       "      <td>Lactococcus virus ASCC191</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj307952</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1801</th>\n",
       "      <td>GCF_001502115.1</td>\n",
       "      <td>Lactococcus virus CB13</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj307768</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1827</th>\n",
       "      <td>GCF_001501375.1</td>\n",
       "      <td>Lactococcus virus Sl4</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj307775</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1828</th>\n",
       "      <td>GCF_001501335.1</td>\n",
       "      <td>Lactococcus virus CB14</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj307767</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2044</th>\n",
       "      <td>GCF_001041075.1</td>\n",
       "      <td>Lactococcus phage WRP3</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj287977</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2063</th>\n",
       "      <td>GCF_001015305.1</td>\n",
       "      <td>Lactococcus phage 1358</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj284805</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2173</th>\n",
       "      <td>GCF_000923495.1</td>\n",
       "      <td>Lactococcus phage P162</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj251393</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2183</th>\n",
       "      <td>GCF_000922855.1</td>\n",
       "      <td>Lactococcus phage P118</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj251392</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2216</th>\n",
       "      <td>GCF_000920955.1</td>\n",
       "      <td>Lactococcus phage P078</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj251390</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2219</th>\n",
       "      <td>GCF_000920915.1</td>\n",
       "      <td>Lactococcus phage P092</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj251389</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2325</th>\n",
       "      <td>GCF_000916455.1</td>\n",
       "      <td>Lactococcus phage phiL47</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj240042</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2482</th>\n",
       "      <td>GCF_000910615.1</td>\n",
       "      <td>Lactococcus phage BM13</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj213076</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2967</th>\n",
       "      <td>GCF_000890755.1</td>\n",
       "      <td>Lactococcus phage 949</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj64559</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3040</th>\n",
       "      <td>GCF_000882895.1</td>\n",
       "      <td>Lactococcus phage P087</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj37887</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3091</th>\n",
       "      <td>GCF_000879895.1</td>\n",
       "      <td>Lactococcus phage 1706</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj29283</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3180</th>\n",
       "      <td>GCF_000870105.1</td>\n",
       "      <td>Lactococcus virus jj50</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj17759</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3181</th>\n",
       "      <td>GCF_000870085.1</td>\n",
       "      <td>Lactococcus phage Q54</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj17739</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3199</th>\n",
       "      <td>GCF_000868485.1</td>\n",
       "      <td>Lactococcus virus 712</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj17757</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3271</th>\n",
       "      <td>GCF_000859265.1</td>\n",
       "      <td>Lactococcus phage BK5-T</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj15244</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3365</th>\n",
       "      <td>GCF_000843105.1</td>\n",
       "      <td>Lactococcus phage P335 sensu lato</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj14281</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3393</th>\n",
       "      <td>GCF_000841645.1</td>\n",
       "      <td>Lactococcus phage ul36</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj14331</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3415</th>\n",
       "      <td>GCF_000840345.1</td>\n",
       "      <td>Lactococcus phage r1t</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj14225</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3430</th>\n",
       "      <td>GCF_000838985.1</td>\n",
       "      <td>Lactococcus phage Tuc2009</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj14131</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3434</th>\n",
       "      <td>GCF_000838765.1</td>\n",
       "      <td>Lactococcus virus bIL170</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj14087</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3437</th>\n",
       "      <td>GCF_000838065.1</td>\n",
       "      <td>Lactococcus phage TP901-1</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj14116</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3440</th>\n",
       "      <td>GCF_000837965.1</td>\n",
       "      <td>Lactococcus virus sk1</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj14096</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3447</th>\n",
       "      <td>GCF_000837665.1</td>\n",
       "      <td>Lactococcus virus c2</td>\n",
       "      <td>dsDNA</td>\n",
       "      <td>Siphoviridae</td>\n",
       "      <td>Lactococcus lactis</td>\n",
       "      <td>ViralProj14029</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "             Assembly                            Species Molecule_type  \\\n",
       "0     GCF_009685705.1            Lactococcus phage P1048         dsDNA   \n",
       "1     GCF_002620465.1              Lactococcus phage AM4         dsDNA   \n",
       "2     GCF_002620665.1             Lactococcus phage LW81         dsDNA   \n",
       "3     GCF_002620365.1              Lactococcus phage AM1         dsDNA   \n",
       "120   GCF_009685685.1            Lactococcus phage P1045         dsDNA   \n",
       "121   GCF_002611385.1            Lactococcus phage 62503         dsDNA   \n",
       "122   GCF_002611285.1            Lactococcus phage 50902         dsDNA   \n",
       "123   GCF_002758415.1              Lactococcus phage Q33         dsDNA   \n",
       "124   GCF_009662695.1             Lactococcus phage P596         dsDNA   \n",
       "125   GCF_002620505.1              Lactococcus phage AM6         dsDNA   \n",
       "126   GCF_002620585.1             Lactococcus phage LW31         dsDNA   \n",
       "908   GCF_002602185.1          Lactococcus virus ASCC273         dsDNA   \n",
       "916   GCF_002602245.1          Lactococcus virus ASCC532         dsDNA   \n",
       "928   GCF_002602225.1          Lactococcus virus ASCC465         dsDNA   \n",
       "929   GCF_002602205.1          Lactococcus virus ASCC281         dsDNA   \n",
       "1059  GCF_002630605.1             Lactococcus virus CB20         dsDNA   \n",
       "1060  GCF_002630585.1             Lactococcus virus CB19         dsDNA   \n",
       "1432  GCF_001746015.1            Lactococcus phage M5938         dsDNA   \n",
       "1441  GCF_001745855.1            Lactococcus phage 63301         dsDNA   \n",
       "1463  GCF_001745355.1            Lactococcus phage D4412         dsDNA   \n",
       "1467  GCF_001745195.1            Lactococcus phage 50101         dsDNA   \n",
       "1488  GCF_001744695.1            Lactococcus phage M6162         dsDNA   \n",
       "1496  GCF_001744535.1            Lactococcus phage 98201         dsDNA   \n",
       "1515  GCF_001744035.1            Lactococcus phage D4410         dsDNA   \n",
       "1517  GCF_001744015.1            Lactococcus phage M6165         dsDNA   \n",
       "1523  GCF_001743875.1            Lactococcus phage 28201         dsDNA   \n",
       "1677  GCF_001505235.1          Lactococcus virus ASCC191         dsDNA   \n",
       "1801  GCF_001502115.1             Lactococcus virus CB13         dsDNA   \n",
       "1827  GCF_001501375.1              Lactococcus virus Sl4         dsDNA   \n",
       "1828  GCF_001501335.1             Lactococcus virus CB14         dsDNA   \n",
       "2044  GCF_001041075.1             Lactococcus phage WRP3         dsDNA   \n",
       "2063  GCF_001015305.1             Lactococcus phage 1358         dsDNA   \n",
       "2173  GCF_000923495.1             Lactococcus phage P162         dsDNA   \n",
       "2183  GCF_000922855.1             Lactococcus phage P118         dsDNA   \n",
       "2216  GCF_000920955.1             Lactococcus phage P078         dsDNA   \n",
       "2219  GCF_000920915.1             Lactococcus phage P092         dsDNA   \n",
       "2325  GCF_000916455.1           Lactococcus phage phiL47         dsDNA   \n",
       "2482  GCF_000910615.1             Lactococcus phage BM13         dsDNA   \n",
       "2967  GCF_000890755.1              Lactococcus phage 949         dsDNA   \n",
       "3040  GCF_000882895.1             Lactococcus phage P087         dsDNA   \n",
       "3091  GCF_000879895.1             Lactococcus phage 1706         dsDNA   \n",
       "3180  GCF_000870105.1             Lactococcus virus jj50         dsDNA   \n",
       "3181  GCF_000870085.1              Lactococcus phage Q54         dsDNA   \n",
       "3199  GCF_000868485.1              Lactococcus virus 712         dsDNA   \n",
       "3271  GCF_000859265.1            Lactococcus phage BK5-T         dsDNA   \n",
       "3365  GCF_000843105.1  Lactococcus phage P335 sensu lato         dsDNA   \n",
       "3393  GCF_000841645.1             Lactococcus phage ul36         dsDNA   \n",
       "3415  GCF_000840345.1              Lactococcus phage r1t         dsDNA   \n",
       "3430  GCF_000838985.1          Lactococcus phage Tuc2009         dsDNA   \n",
       "3434  GCF_000838765.1           Lactococcus virus bIL170         dsDNA   \n",
       "3437  GCF_000838065.1          Lactococcus phage TP901-1         dsDNA   \n",
       "3440  GCF_000837965.1              Lactococcus virus sk1         dsDNA   \n",
       "3447  GCF_000837665.1               Lactococcus virus c2         dsDNA   \n",
       "\n",
       "            Family                Host    GenBank_Title  \n",
       "0     Siphoviridae  Lactococcus lactis      ASM968570v1  \n",
       "1     Siphoviridae  Lactococcus lactis      ASM262046v1  \n",
       "2     Siphoviridae  Lactococcus lactis      ASM262066v1  \n",
       "3     Siphoviridae  Lactococcus lactis      ASM262036v1  \n",
       "120   Siphoviridae  Lactococcus lactis      ASM968568v1  \n",
       "121   Siphoviridae  Lactococcus lactis      ASM261138v1  \n",
       "122   Siphoviridae  Lactococcus lactis      ASM261128v1  \n",
       "123   Siphoviridae  Lactococcus lactis      ASM275841v1  \n",
       "124   Siphoviridae  Lactococcus lactis      ASM966269v1  \n",
       "125   Siphoviridae  Lactococcus lactis      ASM262050v1  \n",
       "126   Siphoviridae  Lactococcus lactis      ASM262058v1  \n",
       "908   Siphoviridae  Lactococcus lactis      ASM260218v1  \n",
       "916   Siphoviridae  Lactococcus lactis      ASM260224v1  \n",
       "928   Siphoviridae  Lactococcus lactis      ASM260222v1  \n",
       "929   Siphoviridae  Lactococcus lactis      ASM260220v1  \n",
       "1059  Siphoviridae  Lactococcus lactis      ASM263060v1  \n",
       "1060  Siphoviridae  Lactococcus lactis      ASM263058v1  \n",
       "1432  Siphoviridae  Lactococcus lactis  ViralProj344188  \n",
       "1441  Siphoviridae  Lactococcus lactis  ViralProj344153  \n",
       "1463  Siphoviridae  Lactococcus lactis  ViralProj344189  \n",
       "1467  Siphoviridae  Lactococcus lactis  ViralProj344155  \n",
       "1488  Siphoviridae  Lactococcus lactis  ViralProj344187  \n",
       "1496  Siphoviridae  Lactococcus lactis  ViralProj344152  \n",
       "1515  Siphoviridae  Lactococcus lactis  ViralProj344190  \n",
       "1517  Siphoviridae  Lactococcus lactis  ViralProj344186  \n",
       "1523  Siphoviridae  Lactococcus lactis  ViralProj344156  \n",
       "1677  Siphoviridae  Lactococcus lactis  ViralProj307952  \n",
       "1801  Siphoviridae  Lactococcus lactis  ViralProj307768  \n",
       "1827  Siphoviridae  Lactococcus lactis  ViralProj307775  \n",
       "1828  Siphoviridae  Lactococcus lactis  ViralProj307767  \n",
       "2044  Siphoviridae  Lactococcus lactis  ViralProj287977  \n",
       "2063  Siphoviridae  Lactococcus lactis  ViralProj284805  \n",
       "2173  Siphoviridae  Lactococcus lactis  ViralProj251393  \n",
       "2183  Siphoviridae  Lactococcus lactis  ViralProj251392  \n",
       "2216  Siphoviridae  Lactococcus lactis  ViralProj251390  \n",
       "2219  Siphoviridae  Lactococcus lactis  ViralProj251389  \n",
       "2325  Siphoviridae  Lactococcus lactis  ViralProj240042  \n",
       "2482  Siphoviridae  Lactococcus lactis  ViralProj213076  \n",
       "2967  Siphoviridae  Lactococcus lactis   ViralProj64559  \n",
       "3040  Siphoviridae  Lactococcus lactis   ViralProj37887  \n",
       "3091  Siphoviridae  Lactococcus lactis   ViralProj29283  \n",
       "3180  Siphoviridae  Lactococcus lactis   ViralProj17759  \n",
       "3181  Siphoviridae  Lactococcus lactis   ViralProj17739  \n",
       "3199  Siphoviridae  Lactococcus lactis   ViralProj17757  \n",
       "3271  Siphoviridae  Lactococcus lactis   ViralProj15244  \n",
       "3365  Siphoviridae  Lactococcus lactis   ViralProj14281  \n",
       "3393  Siphoviridae  Lactococcus lactis   ViralProj14331  \n",
       "3415  Siphoviridae  Lactococcus lactis   ViralProj14225  \n",
       "3430  Siphoviridae  Lactococcus lactis   ViralProj14131  \n",
       "3434  Siphoviridae  Lactococcus lactis   ViralProj14087  \n",
       "3437  Siphoviridae  Lactococcus lactis   ViralProj14116  \n",
       "3440  Siphoviridae  Lactococcus lactis   ViralProj14096  \n",
       "3447  Siphoviridae  Lactococcus lactis   ViralProj14029  "
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#searching Family Siphoviridae which Host is Lactococcus lactis\n",
    "siphoviridae = new_seq.loc [(new_seq['Family'] == 'Siphoviridae') & (new_seq['Host'] == 'Lactococcus lactis')] \n",
    "siphoviridae"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "asb = siphoviridae[\"Assembly\"].tolist()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['GCF_009685705.1',\n",
       " 'GCF_002620465.1',\n",
       " 'GCF_002620665.1',\n",
       " 'GCF_002620365.1',\n",
       " 'GCF_009685685.1',\n",
       " 'GCF_002611385.1',\n",
       " 'GCF_002611285.1',\n",
       " 'GCF_002758415.1',\n",
       " 'GCF_009662695.1',\n",
       " 'GCF_002620505.1',\n",
       " 'GCF_002620585.1',\n",
       " 'GCF_002602185.1',\n",
       " 'GCF_002602245.1',\n",
       " 'GCF_002602225.1',\n",
       " 'GCF_002602205.1',\n",
       " 'GCF_002630605.1',\n",
       " 'GCF_002630585.1',\n",
       " 'GCF_001746015.1',\n",
       " 'GCF_001745855.1',\n",
       " 'GCF_001745355.1',\n",
       " 'GCF_001745195.1',\n",
       " 'GCF_001744695.1',\n",
       " 'GCF_001744535.1',\n",
       " 'GCF_001744035.1',\n",
       " 'GCF_001744015.1',\n",
       " 'GCF_001743875.1',\n",
       " 'GCF_001505235.1',\n",
       " 'GCF_001502115.1',\n",
       " 'GCF_001501375.1',\n",
       " 'GCF_001501335.1',\n",
       " 'GCF_001041075.1',\n",
       " 'GCF_001015305.1',\n",
       " 'GCF_000923495.1',\n",
       " 'GCF_000922855.1',\n",
       " 'GCF_000920955.1',\n",
       " 'GCF_000920915.1',\n",
       " 'GCF_000916455.1',\n",
       " 'GCF_000910615.1',\n",
       " 'GCF_000890755.1',\n",
       " 'GCF_000882895.1',\n",
       " 'GCF_000879895.1',\n",
       " 'GCF_000870105.1',\n",
       " 'GCF_000870085.1',\n",
       " 'GCF_000868485.1',\n",
       " 'GCF_000859265.1',\n",
       " 'GCF_000843105.1',\n",
       " 'GCF_000841645.1',\n",
       " 'GCF_000840345.1',\n",
       " 'GCF_000838985.1',\n",
       " 'GCF_000838765.1',\n",
       " 'GCF_000838065.1',\n",
       " 'GCF_000837965.1',\n",
       " 'GCF_000837665.1']"
      ]
     },
     "execution_count": 8,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "asb"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "with open('asb.txt', 'w') as f:\n",
    "    for item in asb:\n",
    "        f.write(\"%s\\n\" % item)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#from https://dmnfarrell.github.io/bioinformatics/assemblies-genbank-python\n",
    "def get_assembly_summary(id):\n",
    "    \"\"\"Get esummary for an entrez id\"\"\"\n",
    "    from Bio import Entrez\n",
    "    esummary_handle = Entrez.esummary(db=\"assembly\", id=id, report=\"full\")\n",
    "    esummary_record = Entrez.read(esummary_handle)\n",
    "    return esummary_record\n",
    "\n",
    "def get_assemblies(term, download=True, path='assemblies'):\n",
    "    \"\"\"Download genbank assemblies for a given search term.\n",
    "    Args:\n",
    "        term: search term, usually organism name\n",
    "        download: whether to download the results\n",
    "        path: folder to save to\n",
    "    \"\"\"\n",
    "    #provide your own mail here\n",
    "    Entrez.email = \"amandakoryl@gmail.com\"\n",
    "    handle = Entrez.esearch(db=\"assembly\", term=term, retmax='100')\n",
    "    record = Entrez.read(handle)\n",
    "    ids = record['IdList']\n",
    "    print (f'found {len(ids)} ids')\n",
    "    links = []\n",
    "    for id in ids:\n",
    "        #get summary\n",
    "        summary = get_assembly_summary(id)\n",
    "        #get ftp link\n",
    "        url = summary['DocumentSummarySet']['DocumentSummary'][0]['FtpPath_RefSeq']\n",
    "        if url == '':\n",
    "            continue\n",
    "        label = os.path.basename(url)\n",
    "        link = os.path.join(url,label+'_genomic.fna.gz')\n",
    "        print (link)\n",
    "        links.append(link)\n",
    "        if download == True:\n",
    "            #download link\n",
    "            urllib.request.urlretrieve(link, f'{label}.fna.gz')\n",
    "    return links"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "--2021-03-18 14:11:44--  https://ftp.ncbi.nlm.nih.gov/genomes/refseq/assembly_summary_refseq.txt\n",
      "Resolving ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)... 165.112.9.229, 165.112.9.228, 2607:f220:41e:250::7, ...\n",
      "Connecting to ftp.ncbi.nlm.nih.gov (ftp.ncbi.nlm.nih.gov)|165.112.9.229|:443... connected.\n",
      "HTTP request sent, awaiting response... 200 OK\n",
      "Length: 67248987 (64M) [text/plain]\n",
      "Saving to: â€�assembly_summary_refseq.txt.3â€™\n",
      "\n",
      "     0K .......... .......... .......... .......... ..........  0%  186K 5m53s\n",
      "    50K .......... .......... .......... .......... ..........  0%  370K 4m25s\n",
      "   100K .......... .......... .......... .......... ..........  0% 1.44M 3m11s\n",
      "   150K .......... .......... .......... .......... ..........  0% 2.21M 2m31s\n",
      "   200K .......... .......... .......... .......... ..........  0%  431K 2m31s\n",
      "   250K .......... .......... .......... .......... ..........  0% 2.80M 2m9s\n",
      "   300K .......... .......... .......... .......... ..........  0%  512K 2m9s\n",
      "   350K .......... .......... .......... .......... ..........  0% 1.23M 1m59s\n",
      "   400K .......... .......... .......... .......... ..........  0% 1.79M 1m50s\n",
      "   450K .......... .......... .......... .......... ..........  0% 2.91M 1m41s\n",
      "   500K .......... .......... .......... .......... ..........  0% 3.36M 94s\n",
      "   550K .......... .......... .......... .......... ..........  0% 1.40M 89s\n",
      "   600K .......... .......... .......... .......... ..........  0% 1.86M 85s\n",
      "   650K .......... .......... .......... .......... ..........  1% 2.37M 81s\n",
      "   700K .......... .......... .......... .......... ..........  1% 2.56M 77s\n",
      "   750K .......... .......... .......... .......... ..........  1% 1.26M 75s\n",
      "   800K .......... .......... .......... .......... ..........  1% 1.82M 73s\n",
      "   850K .......... .......... .......... .......... ..........  1% 2.81M 70s\n",
      "   900K .......... .......... .......... .......... ..........  1% 2.98M 67s\n",
      "   950K .......... .......... .......... .......... ..........  1% 2.26M 65s\n",
      "  1000K .......... .......... .......... .......... ..........  1% 2.78M 63s\n",
      "  1050K .......... .......... .......... .......... ..........  1% 1.83M 62s\n",
      "  1100K .......... .......... .......... .......... ..........  1% 2.96M 60s\n",
      "  1150K .......... .......... .......... .......... ..........  1% 1.85M 59s\n",
      "  1200K .......... .......... .......... .......... ..........  1% 2.73M 58s\n",
      "  1250K .......... .......... .......... .......... ..........  1% 2.79M 56s\n",
      "  1300K .......... .......... .......... .......... ..........  2% 2.23M 55s\n",
      "  1350K .......... .......... .......... .......... ..........  2% 2.92M 54s\n",
      "  1400K .......... .......... .......... .......... ..........  2% 2.99M 53s\n",
      "  1450K .......... .......... .......... .......... ..........  2%  361K 57s\n",
      "  1500K .......... .......... .......... .......... ..........  2% 2.78M 56s\n",
      "  1550K .......... .......... .......... .......... ..........  2% 2.13M 55s\n",
      "  1600K .......... .......... .......... .......... ..........  2% 1.79M 54s\n",
      "  1650K .......... .......... .......... .......... ..........  2% 2.03M 53s\n",
      "  1700K .......... .......... .......... .......... ..........  2% 2.71M 53s\n",
      "  1750K .......... .......... .......... .......... ..........  2% 2.75M 52s\n",
      "  1800K .......... .......... .......... .......... ..........  2% 2.89M 51s\n",
      "  1850K .......... .......... .......... .......... ..........  2% 2.18M 50s\n",
      "  1900K .......... .......... .......... .......... ..........  2% 3.04M 49s\n",
      "  1950K .......... .......... .......... .......... ..........  3% 2.23M 49s\n",
      "  2000K .......... .......... .......... .......... ..........  3% 1.70M 48s\n",
      "  2050K .......... .......... .......... .......... ..........  3% 1.27M 48s\n",
      "  2100K .......... .......... .......... .......... ..........  3% 2.91M 48s\n",
      "  2150K .......... .......... .......... .......... ..........  3% 2.07M 47s\n",
      "  2200K .......... .......... .......... .......... ..........  3% 2.66M 47s\n",
      "  2250K .......... .......... .......... .......... ..........  3% 2.61M 46s\n",
      "  2300K .......... .......... .......... .......... ..........  3% 1.64M 46s\n",
      "  2350K .......... .......... .......... .......... ..........  3% 2.03M 46s\n",
      "  2400K .......... .......... .......... .......... ..........  3% 2.92M 45s\n",
      "  2450K .......... .......... .......... .......... ..........  3%  875K 46s\n",
      "  2500K .......... .......... .......... .......... ..........  3% 1.00M 46s\n",
      "  2550K .......... .......... .......... .......... ..........  3%  831K 46s\n",
      "  2600K .......... .......... .......... .......... ..........  4%  713K 47s\n",
      "  2650K .......... .......... .......... .......... ..........  4% 1.13M 47s\n",
      "  2700K .......... .......... .......... .......... ..........  4%  609K 48s\n",
      "  2750K .......... .......... .......... .......... ..........  4% 1.19M 48s\n",
      "  2800K .......... .......... .......... .......... ..........  4% 19.2K 1m45s\n",
      "  2850K .......... .......... .......... .......... ..........  4% 72.9M 1m43s\n",
      "  2900K .......... .......... .......... .......... ..........  4% 85.7M 1m41s\n",
      "  2950K .......... .......... .......... .......... ..........  4%  134M 99s\n",
      "  3000K .......... .......... .......... .......... ..........  4% 96.7M 98s\n",
      "  3050K .......... .......... .......... .......... ..........  4%  126M 96s\n",
      "  3100K .......... .......... .......... .......... ..........  4%  119M 94s\n",
      "  3150K .......... .......... .......... .......... ..........  4% 98.0M 93s\n",
      "  3200K .......... .......... .......... .......... ..........  4%  114M 91s\n",
      "  3250K .......... .......... .......... .......... ..........  5%  121M 90s\n",
      "  3300K .......... .......... .......... .......... ..........  5%  105M 89s\n",
      "  3350K .......... .......... .......... .......... ..........  5% 28.1M 87s\n",
      "  3400K .......... .......... .......... .......... ..........  5% 25.5M 86s\n",
      "  3450K .......... .......... .......... .......... ..........  5%  121M 85s\n",
      "  3500K .......... .......... .......... .......... ..........  5%  115M 83s\n",
      "  3550K .......... .......... .......... .......... ..........  5% 87.0M 82s\n",
      "  3600K .......... .......... .......... .......... ..........  5%  110M 81s\n",
      "  3650K .......... .......... .......... .......... ..........  5% 64.9M 80s\n",
      "  3700K .......... .......... .......... .......... ..........  5%  101M 79s\n",
      "  3750K .......... .......... .......... .......... ..........  5%  104M 78s\n",
      "  3800K .......... .......... .......... .......... ..........  5% 78.9M 77s\n",
      "  3850K .......... .......... .......... .......... ..........  5% 5.55M 76s\n",
      "  3900K .......... .......... .......... .......... ..........  6% 88.9M 75s\n",
      "  3950K .......... .......... .......... .......... ..........  6% 84.7M 74s\n",
      "  4000K .......... .......... .......... .......... ..........  6% 83.6M 73s\n",
      "  4050K .......... .......... .......... .......... ..........  6%  132M 72s\n",
      "  4100K .......... .......... .......... .......... ..........  6%  121M 71s\n",
      "  4150K .......... .......... .......... .......... ..........  6% 9.93M 70s\n",
      "  4200K .......... .......... .......... .......... ..........  6% 76.4M 69s\n",
      "  4250K .......... .......... .......... .......... ..........  6% 94.8M 68s\n",
      "  4300K .......... .......... .......... .......... ..........  6% 82.4M 67s\n",
      "  4350K .......... .......... .......... .......... ..........  6% 62.2M 67s\n",
      "  4400K .......... .......... .......... .......... ..........  6% 97.6M 66s\n",
      "  4450K .......... .......... .......... .......... ..........  6%  122M 65s\n",
      "  4500K .......... .......... .......... .......... ..........  6%  135M 64s\n",
      "  4550K .......... .......... .......... .......... ..........  7% 72.0M 64s\n",
      "  4600K .......... .......... .......... .......... ..........  7%  117M 63s\n",
      "  4650K .......... .......... .......... .......... ..........  7%  125M 62s\n",
      "  4700K .......... .......... .......... .......... ..........  7%  124M 61s\n",
      "  4750K .......... .......... .......... .......... ..........  7%  135M 61s\n",
      "  4800K .......... .......... .......... .......... ..........  7% 94.5M 60s\n",
      "  4850K .......... .......... .......... .......... ..........  7%  410K 61s\n",
      "  4900K .......... .......... .......... .......... ..........  7% 1.20M 61s\n",
      "  4950K .......... .......... .......... .......... ..........  7% 1.20M 61s\n",
      "  5000K .......... .......... .......... .......... ..........  7% 1.02M 61s\n",
      "  5050K .......... .......... .......... .......... ..........  7%  967K 60s\n",
      "  5100K .......... .......... .......... .......... ..........  7% 1.31M 60s\n",
      "  5150K .......... .......... .......... .......... ..........  7%  593K 61s\n",
      "  5200K .......... .......... .......... .......... ..........  7%  681K 61s\n",
      "  5250K .......... .......... .......... .......... ..........  8%  561K 61s\n",
      "  5300K .......... .......... .......... .......... ..........  8%  606K 62s\n",
      "  5350K .......... .......... .......... .......... ..........  8%  735K 62s\n",
      "  5400K .......... .......... .......... .......... ..........  8%  596K 62s\n",
      "  5450K .......... .......... .......... .......... ..........  8% 1.34M 62s\n",
      "  5500K .......... .......... .......... .......... ..........  8% 1.07M 62s\n",
      "  5550K .......... .......... .......... .......... ..........  8% 1.23M 62s\n",
      "  5600K .......... .......... .......... .......... ..........  8% 1.73M 61s\n",
      "  5650K .......... .......... .......... .......... ..........  8% 1.77M 61s\n",
      "  5700K .......... .......... .......... .......... ..........  8%  929K 61s\n",
      "  5750K .......... .......... .......... .......... ..........  8% 1.41M 61s\n",
      "  5800K .......... .......... .......... .......... ..........  8% 1.06M 61s\n",
      "  5850K .......... .......... .......... .......... ..........  8% 1.65M 60s\n",
      "  5900K .......... .......... .......... .......... ..........  9% 2.30M 60s\n",
      "  5950K .......... .......... .......... .......... ..........  9% 1.78M 60s\n",
      "  6000K .......... .......... .......... .......... ..........  9%  829K 60s\n",
      "  6050K .......... .......... .......... .......... ..........  9% 1.53M 60s\n",
      "  6100K .......... .......... .......... .......... ..........  9% 1.01M 59s\n",
      "  6150K .......... .......... .......... .......... ..........  9% 2.15M 59s\n",
      "  6200K .......... .......... .......... .......... ..........  9% 2.91M 59s\n",
      "  6250K .......... .......... .......... .......... ..........  9% 1.79M 59s\n",
      "  6300K .......... .......... .......... .......... ..........  9%  759K 59s\n",
      "  6350K .......... .......... .......... .......... ..........  9% 2.19M 58s\n",
      "  6400K .......... .......... .......... .......... ..........  9%  438K 59s\n",
      "  6450K .......... .......... .......... .......... ..........  9%  457K 59s\n",
      "  6500K .......... .......... .......... .......... ..........  9%  493K 60s\n",
      "  6550K .......... .......... .......... .......... .......... 10%  746K 60s\n",
      "  6600K .......... .......... .......... .......... .......... 10% 19.8K 82s\n",
      "  6650K .......... .......... .......... .......... .......... 10% 13.4M 81s\n",
      "  6700K .......... .......... .......... .......... .......... 10% 13.8M 81s\n",
      "  6750K .......... .......... .......... .......... .......... 10% 27.8M 80s\n",
      "  6800K .......... .......... .......... .......... .......... 10% 79.4K 85s\n",
      "  6850K .......... .......... .......... .......... .......... 10% 64.7M 84s\n",
      "  6900K .......... .......... .......... .......... .......... 10% 61.9M 83s\n",
      "  6950K .......... .......... .......... .......... .......... 10% 68.9M 83s\n",
      "  7000K .......... .......... .......... .......... .......... 10% 72.1M 82s\n",
      "  7050K .......... .......... .......... .......... .......... 10% 85.6M 81s\n",
      "  7100K .......... .......... .......... .......... .......... 10% 97.2M 81s\n",
      "  7150K .......... .......... .......... .......... .......... 10% 64.8M 80s\n",
      "  7200K .......... .......... .......... .......... .......... 11% 51.5M 79s\n",
      "  7250K .......... .......... .......... .......... .......... 11% 70.1M 79s\n",
      "  7300K .......... .......... .......... .......... .......... 11% 82.4M 78s\n",
      "  7350K .......... .......... .......... .......... .......... 11% 88.8M 78s\n",
      "  7400K .......... .......... .......... .......... .......... 11% 63.3M 77s\n",
      "  7450K .......... .......... .......... .......... .......... 11% 77.3M 77s\n",
      "  7500K .......... .......... .......... .......... .......... 11% 83.0M 76s\n",
      "  7550K .......... .......... .......... .......... .......... 11% 67.7M 75s\n",
      "  7600K .......... .......... .......... .......... .......... 11% 57.1M 75s\n",
      "  7650K .......... .......... .......... .......... .......... 11% 63.1M 74s\n",
      "  7700K .......... .......... .......... .......... .......... 11% 59.5M 74s\n",
      "  7750K .......... .......... .......... .......... .......... 11% 76.8M 73s\n",
      "  7800K .......... .......... .......... .......... .......... 11% 58.5M 73s\n",
      "  7850K .......... .......... .......... .......... .......... 12% 71.4M 72s\n",
      "  7900K .......... .......... .......... .......... .......... 12% 63.9M 72s\n",
      "  7950K .......... .......... .......... .......... .......... 12% 76.9M 71s\n",
      "  8000K .......... .......... .......... .......... .......... 12% 40.9M 71s\n",
      "  8050K .......... .......... .......... .......... .......... 12% 65.5M 70s\n",
      "  8100K .......... .......... .......... .......... .......... 12% 61.1M 70s\n",
      "  8150K .......... .......... .......... .......... .......... 12% 57.9M 69s\n",
      "  8200K .......... .......... .......... .......... .......... 12% 77.6M 69s\n",
      "  8250K .......... .......... .......... .......... .......... 12%  107M 68s\n",
      "  8300K .......... .......... .......... .......... .......... 12%  112M 68s\n",
      "  8350K .......... .......... .......... .......... .......... 12%  111M 67s\n",
      "  8400K .......... .......... .......... .......... .......... 12% 72.7M 67s\n",
      "  8450K .......... .......... .......... .......... .......... 12%  121M 66s\n",
      "  8500K .......... .......... .......... .......... .......... 13%  122M 66s\n",
      "  8550K .......... .......... .......... .......... .......... 13%  126M 66s\n",
      "  8600K .......... .......... .......... .......... .......... 13%  113M 65s\n",
      "  8650K .......... .......... .......... .......... .......... 13%  133M 65s\n",
      "  8700K .......... .......... .......... .......... .......... 13% 1.70M 64s\n",
      "  8750K .......... .......... .......... .......... .......... 13% 1.30M 64s\n",
      "  8800K .......... .......... .......... .......... .......... 13%  608K 64s\n",
      "  8850K .......... .......... .......... .......... .......... 13% 1.11M 64s\n",
      "  8900K .......... .......... .......... .......... .......... 13%  963K 64s\n",
      "  8950K .......... .......... .......... .......... .......... 13% 1.91M 64s\n",
      "  9000K .......... .......... .......... .......... .......... 13% 3.26M 64s\n",
      "  9050K .......... .......... .......... .......... .......... 13% 3.05M 63s\n",
      "  9100K .......... .......... .......... .......... .......... 13% 2.89M 63s\n",
      "  9150K .......... .......... .......... .......... .......... 14% 2.87M 63s\n",
      "  9200K .......... .......... .......... .......... .......... 14% 1.82M 62s\n",
      "  9250K .......... .......... .......... .......... .......... 14% 2.80M 62s\n",
      "  9300K .......... .......... .......... .......... .......... 14% 2.72M 62s\n",
      "  9350K .......... .......... .......... .......... .......... 14%  474K 62s\n",
      "  9400K .......... .......... .......... .......... .......... 14% 1.07M 62s\n",
      "  9450K .......... .......... .......... .......... .......... 14% 2.77M 62s\n",
      "  9500K .......... .......... .......... .......... .......... 14% 1.44M 62s\n",
      "  9550K .......... .......... .......... .......... .......... 14% 3.02M 61s\n",
      "  9600K .......... .......... .......... .......... .......... 14% 2.06M 61s\n",
      "  9650K .......... .......... .......... .......... .......... 14% 2.86M 61s\n",
      "  9700K .......... .......... .......... .......... .......... 14% 2.91M 61s\n",
      "  9750K .......... .......... .......... .......... .......... 14% 2.67M 60s\n",
      "  9800K .......... .......... .......... .......... .......... 14% 2.12M 60s\n",
      "  9850K .......... .......... .......... .......... .......... 15% 2.85M 60s\n",
      "  9900K .......... .......... .......... .......... .......... 15% 2.11M 60s\n",
      "  9950K .......... .......... .......... .......... .......... 15% 2.14M 59s\n",
      " 10000K .......... .......... .......... .......... .......... 15% 2.05M 59s\n",
      " 10050K .......... .......... .......... .......... .......... 15% 2.70M 59s\n",
      " 10100K .......... .......... .......... .......... .......... 15% 2.11M 59s\n",
      " 10150K .......... .......... .......... .......... .......... 15% 2.96M 58s\n",
      " 10200K .......... .......... .......... .......... .......... 15% 2.71M 58s\n",
      " 10250K .......... .......... .......... .......... .......... 15% 2.90M 58s\n",
      " 10300K .......... .......... .......... .......... .......... 15% 1.50M 58s\n",
      " 10350K .......... .......... .......... .......... .......... 15% 1.67M 58s\n",
      " 10400K .......... .......... .......... .......... .......... 15% 3.21M 57s\n",
      " 10450K .......... .......... .......... .......... .......... 15%  831K 57s\n",
      " 10500K .......... .......... .......... .......... .......... 16%  951K 57s\n",
      " 10550K .......... .......... .......... .......... .......... 16%  694K 57s\n",
      " 10600K .......... .......... .......... .......... .......... 16%  840K 57s\n",
      " 10650K .......... .......... .......... .......... .......... 16%  693K 57s\n",
      " 10700K .......... .......... .......... .......... .......... 16%  943K 57s\n",
      " 10750K .......... .......... .......... .......... .......... 16%  672K 57s\n",
      " 10800K .......... .......... .......... .......... .......... 16%  590K 57s\n",
      " 10850K .......... .......... .......... .......... .......... 16% 1.71M 57s\n",
      " 10900K .......... .......... .......... .......... .......... 16% 2.85M 57s\n",
      " 10950K .......... .......... .......... .......... .......... 16% 1.71M 57s\n",
      " 11000K .......... .......... .......... .......... .......... 16% 1.90M 57s\n",
      " 11050K .......... .......... .......... .......... .......... 16% 2.52M 57s\n",
      " 11100K .......... .......... .......... .......... .......... 16% 2.18M 56s\n",
      " 11150K .......... .......... .......... .......... .......... 17% 2.30M 56s\n",
      " 11200K .......... .......... .......... .......... .......... 17% 1.75M 56s\n",
      " 11250K .......... .......... .......... .......... .......... 17% 2.99M 56s\n",
      " 11300K .......... .......... .......... .......... .......... 17% 2.24M 56s\n",
      " 11350K .......... .......... .......... .......... .......... 17% 2.97M 55s\n",
      " 11400K .......... .......... .......... .......... .......... 17% 2.80M 55s\n",
      " 11450K .......... .......... .......... .......... .......... 17% 2.90M 55s\n",
      " 11500K .......... .......... .......... .......... .......... 17% 3.03M 55s\n",
      " 11550K .......... .......... .......... .......... .......... 17% 2.60M 55s\n",
      " 11600K .......... .......... .......... .......... .......... 17% 2.00M 54s\n",
      " 11650K .......... .......... .......... .......... .......... 17% 1.54M 54s\n",
      " 11700K .......... .......... .......... .......... .......... 17%  855K 54s\n",
      " 11750K .......... .......... .......... .......... .......... 17% 2.87M 54s\n",
      " 11800K .......... .......... .......... .......... .......... 18% 2.73M 54s\n",
      " 11850K .......... .......... .......... .......... .......... 18% 2.15M 54s\n",
      " 11900K .......... .......... .......... .......... .......... 18% 1.62M 53s\n",
      " 11950K .......... .......... .......... .......... .......... 18% 2.93M 53s\n",
      " 12000K .......... .......... .......... .......... .......... 18% 1.81M 53s\n",
      " 12050K .......... .......... .......... .......... .......... 18% 2.85M 53s\n",
      " 12100K .......... .......... .......... .......... .......... 18% 2.98M 53s\n",
      " 12150K .......... .......... .......... .......... .......... 18% 2.84M 53s\n",
      " 12200K .......... .......... .......... .......... .......... 18% 2.04M 52s\n",
      " 12250K .......... .......... .......... .......... .......... 18% 1.84M 52s\n",
      " 12300K .......... .......... .......... .......... .......... 18% 2.09M 52s\n",
      " 12350K .......... .......... .......... .......... .......... 18% 2.18M 52s\n",
      " 12400K .......... .......... .......... .......... .......... 18% 1.43M 52s\n",
      " 12450K .......... .......... .......... .......... .......... 19% 1.77M 52s\n",
      " 12500K .......... .......... .......... .......... .......... 19% 1.72M 52s\n",
      " 12550K .......... .......... .......... .......... .......... 19% 2.56M 51s\n",
      " 12600K .......... .......... .......... .......... .......... 19% 1.07M 51s\n",
      " 12650K .......... .......... .......... .......... .......... 19%  807K 51s\n",
      " 12700K .......... .......... .......... .......... .......... 19%  705K 51s\n",
      " 12750K .......... .......... .......... .......... .......... 19%  790K 51s\n",
      " 12800K .......... .......... .......... .......... .......... 19%  436K 52s\n",
      " 12850K .......... .......... .......... .......... .......... 19%  773K 52s\n",
      " 12900K .......... .......... .......... .......... .......... 19%  796K 52s\n",
      " 12950K .......... .......... .......... .......... .......... 19%  684K 52s\n",
      " 13000K .......... .......... .......... .......... .......... 19% 1.34M 52s\n",
      " 13050K .......... .......... .......... .......... .......... 19% 2.99M 51s\n",
      " 13100K .......... .......... .......... .......... .......... 20% 2.77M 51s\n",
      " 13150K .......... .......... .......... .......... .......... 20% 2.95M 51s\n",
      " 13200K .......... .......... .......... .......... .......... 20% 1.64M 51s\n",
      " 13250K .......... .......... .......... .......... .......... 20% 2.77M 51s\n",
      " 13300K .......... .......... .......... .......... .......... 20% 2.55M 51s\n",
      " 13350K .......... .......... .......... .......... .......... 20% 2.72M 50s\n",
      " 13400K .......... .......... .......... .......... .......... 20% 1.61M 50s\n",
      " 13450K .......... .......... .......... .......... .......... 20% 2.90M 50s\n",
      " 13500K .......... .......... .......... .......... .......... 20% 3.05M 50s\n",
      " 13550K .......... .......... .......... .......... .......... 20% 2.07M 50s\n",
      " 13600K .......... .......... .......... .......... .......... 20% 1.88M 50s\n",
      " 13650K .......... .......... .......... .......... .......... 20% 2.82M 50s\n",
      " 13700K .......... .......... .......... .......... .......... 20% 2.97M 49s\n",
      " 13750K .......... .......... .......... .......... .......... 21% 1.39M 49s\n",
      " 13800K .......... .......... .......... .......... .......... 21% 1.80M 49s\n",
      " 13850K .......... .......... .......... .......... .......... 21% 4.04M 49s\n",
      " 13900K .......... .......... .......... .......... .......... 21% 3.01M 49s\n",
      " 13950K .......... .......... .......... .......... .......... 21% 2.04M 49s\n",
      " 14000K .......... .......... .......... .......... .......... 21% 2.04M 49s\n",
      " 14050K .......... .......... .......... .......... .......... 21% 2.37M 48s\n",
      " 14100K .......... .......... .......... .......... .......... 21% 1.99M 48s\n",
      " 14150K .......... .......... .......... .......... .......... 21% 2.78M 48s\n",
      " 14200K .......... .......... .......... .......... .......... 21% 2.16M 48s\n",
      " 14250K .......... .......... .......... .......... .......... 21% 2.71M 48s\n",
      " 14300K .......... .......... .......... .......... .......... 21% 2.96M 48s\n",
      " 14350K .......... .......... .......... .......... .......... 21% 2.23M 48s\n",
      " 14400K .......... .......... .......... .......... .......... 22% 1.47M 47s\n",
      " 14450K .......... .......... .......... .......... .......... 22% 1.49M 47s\n",
      " 14500K .......... .......... .......... .......... .......... 22% 2.89M 47s\n",
      " 14550K .......... .......... .......... .......... .......... 22% 1.77M 47s\n",
      " 14600K .......... .......... .......... .......... .......... 22% 4.16M 47s\n",
      " 14650K .......... .......... .......... .......... .......... 22% 1.68M 47s\n",
      " 14700K .......... .......... .......... .......... .......... 22% 2.05M 47s\n",
      " 14750K .......... .......... .......... .......... .......... 22% 1.77M 47s\n",
      " 14800K .......... .......... .......... .......... .......... 22% 2.22M 46s\n",
      " 14850K .......... .......... .......... .......... .......... 22% 1.65M 46s\n",
      " 14900K .......... .......... .......... .......... .......... 22% 1.51M 46s\n",
      " 14950K .......... .......... .......... .......... .......... 22% 3.99M 46s\n",
      " 15000K .......... .......... .......... .......... .......... 22% 1.78M 46s\n",
      " 15050K .......... .......... .......... .......... .......... 22% 3.04M 46s\n",
      " 15100K .......... .......... .......... .......... .......... 23% 2.98M 46s\n",
      " 15150K .......... .......... .......... .......... .......... 23% 2.31M 46s\n",
      " 15200K .......... .......... .......... .......... .......... 23% 1.59M 45s\n",
      " 15250K .......... .......... .......... .......... .......... 23% 1.27M 45s\n",
      " 15300K .......... .......... .......... .......... .......... 23% 2.15M 45s\n",
      " 15350K .......... .......... .......... .......... .......... 23% 1.75M 45s\n",
      " 15400K .......... .......... .......... .......... .......... 23% 2.92M 45s\n",
      " 15450K .......... .......... .......... .......... .......... 23% 2.11M 45s\n",
      " 15500K .......... .......... .......... .......... .......... 23% 3.45M 45s\n",
      " 15550K .......... .......... .......... .......... .......... 23%  707K 45s\n",
      " 15600K .......... .......... .......... .......... .......... 23%  587K 45s\n",
      " 15650K .......... .......... .......... .......... .......... 23%  703K 45s\n",
      " 15700K .......... .......... .......... .......... .......... 23%  659K 45s\n",
      " 15750K .......... .......... .......... .......... .......... 24% 1.07M 45s\n",
      " 15800K .......... .......... .......... .......... .......... 24%  707K 45s\n",
      " 15850K .......... .......... .......... .......... .......... 24% 1.18M 45s\n",
      " 15900K .......... .......... .......... .......... .......... 24% 2.93M 45s\n",
      " 15950K .......... .......... .......... .......... .......... 24% 2.96M 45s\n",
      " 16000K .......... .......... .......... .......... .......... 24% 2.24M 45s\n",
      " 16050K .......... .......... .......... .......... .......... 24% 2.74M 44s\n",
      " 16100K .......... .......... .......... .......... .......... 24% 2.96M 44s\n",
      " 16150K .......... .......... .......... .......... .......... 24% 2.95M 44s\n",
      " 16200K .......... .......... .......... .......... .......... 24% 2.92M 44s\n",
      " 16250K .......... .......... .......... .......... .......... 24% 2.57M 44s\n",
      " 16300K .......... .......... .......... .......... .......... 24% 2.98M 44s\n",
      " 16350K .......... .......... .......... .......... .......... 24% 2.16M 44s\n",
      " 16400K .......... .......... .......... .......... .......... 25% 1.67M 44s\n",
      " 16450K .......... .......... .......... .......... .......... 25% 2.91M 43s\n",
      " 16500K .......... .......... .......... .......... .......... 25%  481K 44s\n",
      " 16550K .......... .......... .......... .......... .......... 25% 2.15M 44s\n",
      " 16600K .......... .......... .......... .......... .......... 25% 3.94M 43s\n",
      " 16650K .......... .......... .......... .......... .......... 25% 2.68M 43s\n",
      " 16700K .......... .......... .......... .......... .......... 25% 2.06M 43s\n",
      " 16750K .......... .......... .......... .......... .......... 25% 1.67M 43s\n",
      " 16800K .......... .......... .......... .......... .......... 25% 1.76M 43s\n",
      " 16850K .......... .......... .......... .......... .......... 25% 2.71M 43s\n",
      " 16900K .......... .......... .......... .......... .......... 25% 1.93M 43s\n",
      " 16950K .......... .......... .......... .......... .......... 25% 2.93M 43s\n",
      " 17000K .......... .......... .......... .......... .......... 25% 2.95M 43s\n",
      " 17050K .......... .......... .......... .......... .......... 26% 2.00M 42s\n",
      " 17100K .......... .......... .......... .......... .......... 26% 2.14M 42s\n",
      " 17150K .......... .......... .......... .......... .......... 26% 2.21M 42s\n",
      " 17200K .......... .......... .......... .......... .......... 26% 2.29M 42s\n",
      " 17250K .......... .......... .......... .......... .......... 26% 3.01M 42s\n",
      " 17300K .......... .......... .......... .......... .......... 26% 3.33M 42s\n",
      " 17350K .......... .......... .......... .......... .......... 26% 2.13M 42s\n",
      " 17400K .......... .......... .......... .......... .......... 26% 4.20M 42s\n",
      " 17450K .......... .......... .......... .......... .......... 26% 3.05M 42s\n",
      " 17500K .......... .......... .......... .......... .......... 26% 2.36M 41s\n",
      " 17550K .......... .......... .......... .......... .......... 26% 2.91M 41s\n",
      " 17600K .......... .......... .......... .......... .......... 26% 2.25M 41s\n",
      " 17650K .......... .......... .......... .......... .......... 26%  323K 41s\n",
      " 17700K .......... .......... .......... .......... .......... 27% 2.96M 41s\n",
      " 17750K .......... .......... .......... .......... .......... 27% 2.88M 41s\n",
      " 17800K .......... .......... .......... .......... .......... 27% 2.14M 41s\n",
      " 17850K .......... .......... .......... .......... .......... 27% 2.88M 41s\n",
      " 17900K .......... .......... .......... .......... .......... 27% 2.94M 41s\n",
      " 17950K .......... .......... .......... .......... .......... 27% 2.67M 41s\n",
      " 18000K .......... .......... .......... .......... .......... 27% 1.48M 41s\n",
      " 18050K .......... .......... .......... .......... .......... 27% 3.00M 41s\n",
      " 18100K .......... .......... .......... .......... .......... 27% 3.00M 40s\n",
      " 18150K .......... .......... .......... .......... .......... 27% 2.21M 40s\n",
      " 18200K .......... .......... .......... .......... .......... 27% 2.95M 40s\n",
      " 18250K .......... .......... .......... .......... .......... 27% 2.72M 40s\n",
      " 18300K .......... .......... .......... .......... .......... 27% 2.79M 40s\n",
      " 18350K .......... .......... .......... .......... .......... 28% 2.70M 40s\n",
      " 18400K .......... .......... .......... .......... .......... 28% 2.11M 40s\n",
      " 18450K .......... .......... .......... .......... .......... 28% 2.95M 40s\n",
      " 18500K .......... .......... .......... .......... .......... 28% 2.07M 40s\n",
      " 18550K .......... .......... .......... .......... .......... 28% 2.87M 40s\n",
      " 18600K .......... .......... .......... .......... .......... 28% 2.04M 39s\n",
      " 18650K .......... .......... .......... .......... .......... 28% 2.92M 39s\n",
      " 18700K .......... .......... .......... .......... .......... 28% 2.77M 39s\n",
      " 18750K .......... .......... .......... .......... .......... 28% 2.19M 39s\n",
      " 18800K .......... .......... .......... .......... .......... 28% 2.17M 39s\n",
      " 18850K .......... .......... .......... .......... .......... 28% 2.87M 39s\n",
      " 18900K .......... .......... .......... .......... .......... 28% 2.66M 39s\n",
      " 18950K .......... .......... .......... .......... .......... 28% 3.17M 39s\n",
      " 19000K .......... .......... .......... .......... .......... 29% 2.58M 39s\n",
      " 19050K .......... .......... .......... .......... .......... 29% 2.75M 39s\n",
      " 19100K .......... .......... .......... .......... .......... 29% 2.99M 38s\n",
      " 19150K .......... .......... .......... .......... .......... 29%  991K 38s\n",
      " 19200K .......... .......... .......... .......... .......... 29%  650K 39s\n",
      " 19250K .......... .......... .......... .......... .......... 29%  820K 39s\n",
      " 19300K .......... .......... .......... .......... .......... 29%  760K 39s\n",
      " 19350K .......... .......... .......... .......... .......... 29% 1.00M 39s\n",
      " 19400K .......... .......... .......... .......... .......... 29%  565K 39s\n",
      " 19450K .......... .......... .......... .......... .......... 29%  770K 39s\n",
      " 19500K .......... .......... .......... .......... .......... 29%  737K 39s\n",
      " 19550K .......... .......... .......... .......... .......... 29% 1.02M 39s\n",
      " 19600K .......... .......... .......... .......... .......... 29% 2.07M 39s\n",
      " 19650K .......... .......... .......... .......... .......... 29% 2.74M 38s\n",
      " 19700K .......... .......... .......... .......... .......... 30% 2.83M 38s\n",
      " 19750K .......... .......... .......... .......... .......... 30% 2.67M 38s\n",
      " 19800K .......... .......... .......... .......... .......... 30% 2.88M 38s\n",
      " 19850K .......... .......... .......... .......... .......... 30% 2.19M 38s\n",
      " 19900K .......... .......... .......... .......... .......... 30% 2.30M 38s\n",
      " 19950K .......... .......... .......... .......... .......... 30% 2.19M 38s\n",
      " 20000K .......... .......... .......... .......... .......... 30% 2.18M 38s\n",
      " 20050K .......... .......... .......... .......... .......... 30% 2.87M 38s\n",
      " 20100K .......... .......... .......... .......... .......... 30% 2.92M 38s\n",
      " 20150K .......... .......... .......... .......... .......... 30% 2.84M 37s\n",
      " 20200K .......... .......... .......... .......... .......... 30% 2.87M 37s\n",
      " 20250K .......... .......... .......... .......... .......... 30% 2.96M 37s\n",
      " 20300K .......... .......... .......... .......... .......... 30% 2.96M 37s\n",
      " 20350K .......... .......... .......... .......... .......... 31% 2.03M 37s\n",
      " 20400K .......... .......... .......... .......... .......... 31% 1.64M 37s\n",
      " 20450K .......... .......... .......... .......... .......... 31% 4.39M 37s\n",
      " 20500K .......... .......... .......... .......... .......... 31% 2.15M 37s\n",
      " 20550K .......... .......... .......... .......... .......... 31% 2.95M 37s\n",
      " 20600K .......... .......... .......... .......... .......... 31% 2.68M 37s\n",
      " 20650K .......... .......... .......... .......... .......... 31% 1.78M 37s\n",
      " 20700K .......... .......... .......... .......... .......... 31% 2.25M 37s\n",
      " 20750K .......... .......... .......... .......... .......... 31% 2.61M 36s\n",
      " 20800K .......... .......... .......... .......... .......... 31% 1.77M 36s\n",
      " 20850K .......... .......... .......... .......... .......... 31% 2.24M 36s\n",
      " 20900K .......... .......... .......... .......... .......... 31% 3.01M 36s\n",
      " 20950K .......... .......... .......... .......... .......... 31% 2.87M 36s\n",
      " 21000K .......... .......... .......... .......... .......... 32% 1.76M 36s\n",
      " 21050K .......... .......... .......... .......... .......... 32% 2.68M 36s\n",
      " 21100K .......... .......... .......... .......... .......... 32% 2.90M 36s\n",
      " 21150K .......... .......... .......... .......... .......... 32% 2.22M 36s\n",
      " 21200K .......... .......... .......... .......... .......... 32% 1.78M 36s\n",
      " 21250K .......... .......... .......... .......... .......... 32% 2.54M 36s\n",
      " 21300K .......... .......... .......... .......... .......... 32% 2.93M 36s\n",
      " 21350K .......... .......... .......... .......... .......... 32% 2.66M 35s\n",
      " 21400K .......... .......... .......... .......... .......... 32% 2.21M 35s\n",
      " 21450K .......... .......... .......... .......... .......... 32% 1.75M 35s\n",
      " 21500K .......... .......... .......... .......... .......... 32% 2.85M 35s\n",
      " 21550K .......... .......... .......... .......... .......... 32% 2.13M 35s\n",
      " 21600K .......... .......... .......... .......... .......... 32% 1.42M 35s\n",
      " 21650K .......... .......... .......... .......... .......... 33% 2.06M 35s\n",
      " 21700K .......... .......... .......... .......... .......... 33% 2.82M 35s\n",
      " 21750K .......... .......... .......... .......... .......... 33% 2.32M 35s\n",
      " 21800K .......... .......... .......... .......... .......... 33% 2.90M 35s\n",
      " 21850K .......... .......... .......... .......... .......... 33% 1.89M 35s\n",
      " 21900K .......... .......... .......... .......... .......... 33% 2.33M 35s\n",
      " 21950K .......... .......... .......... .......... .......... 33% 2.83M 35s\n",
      " 22000K .......... .......... .......... .......... .......... 33% 1.42M 35s\n",
      " 22050K .......... .......... .......... .......... .......... 33% 2.19M 34s\n",
      " 22100K .......... .......... .......... .......... .......... 33% 2.73M 34s\n",
      " 22150K .......... .......... .......... .......... .......... 33% 2.94M 34s\n",
      " 22200K .......... .......... .......... .......... .......... 33% 2.52M 34s\n",
      " 22250K .......... .......... .......... .......... .......... 33% 2.20M 34s\n",
      " 22300K .......... .......... .......... .......... .......... 34% 2.00M 34s\n",
      " 22350K .......... .......... .......... .......... .......... 34% 2.28M 34s\n",
      " 22400K .......... .......... .......... .......... .......... 34% 1.75M 34s\n",
      " 22450K .......... .......... .......... .......... .......... 34% 2.95M 34s\n",
      " 22500K .......... .......... .......... .......... .......... 34% 3.02M 34s\n",
      " 22550K .......... .......... .......... .......... .......... 34% 1.33M 34s\n",
      " 22600K .......... .......... .......... .......... .......... 34% 2.89M 34s\n",
      " 22650K .......... .......... .......... .......... .......... 34% 3.42M 34s\n",
      " 22700K .......... .......... .......... .......... .......... 34% 2.30M 33s\n",
      " 22750K .......... .......... .......... .......... .......... 34% 3.13M 33s\n",
      " 22800K .......... .......... .......... .......... .......... 34% 1.31M 33s\n",
      " 22850K .......... .......... .......... .......... .......... 34% 3.00M 33s\n",
      " 22900K .......... .......... .......... .......... .......... 34% 2.98M 33s\n",
      " 22950K .......... .......... .......... .......... .......... 35% 2.26M 33s\n",
      " 23000K .......... .......... .......... .......... .......... 35% 2.26M 33s\n",
      " 23050K .......... .......... .......... .......... .......... 35% 2.93M 33s\n",
      " 23100K .......... .......... .......... .......... .......... 35%  908K 33s\n",
      " 23150K .......... .......... .......... .......... .......... 35%  755K 33s\n",
      " 23200K .......... .......... .......... .......... .......... 35% 1.14M 33s\n",
      " 23250K .......... .......... .......... .......... .......... 35% 2.85M 33s\n",
      " 23300K .......... .......... .......... .......... .......... 35% 2.45M 33s\n",
      " 23350K .......... .......... .......... .......... .......... 35% 2.78M 33s\n",
      " 23400K .......... .......... .......... .......... .......... 35% 2.08M 33s\n",
      " 23450K .......... .......... .......... .......... .......... 35% 3.02M 33s\n",
      " 23500K .......... .......... .......... .......... .......... 35% 2.77M 32s\n",
      " 23550K .......... .......... .......... .......... .......... 35% 2.86M 32s\n",
      " 23600K .......... .......... .......... .......... .......... 36% 1.67M 32s\n",
      " 23650K .......... .......... .......... .......... .......... 36% 4.27M 32s\n",
      " 23700K .......... .......... .......... .......... .......... 36% 2.49M 32s\n",
      " 23750K .......... .......... .......... .......... .......... 36% 2.89M 32s\n",
      " 23800K .......... .......... .......... .......... .......... 36% 2.76M 32s\n",
      " 23850K .......... .......... .......... .......... .......... 36% 2.91M 32s\n",
      " 23900K .......... .......... .......... .......... .......... 36% 2.11M 32s\n",
      " 23950K .......... .......... .......... .......... .......... 36% 3.10M 32s\n",
      " 24000K .......... .......... .......... .......... .......... 36% 1.75M 32s\n",
      " 24050K .......... .......... .......... .......... .......... 36% 3.04M 32s\n",
      " 24100K .......... .......... .......... .......... .......... 36% 2.28M 32s\n",
      " 24150K .......... .......... .......... .......... .......... 36% 2.21M 32s\n",
      " 24200K .......... .......... .......... .......... .......... 36% 2.86M 31s\n",
      " 24250K .......... .......... .......... .......... .......... 37% 2.96M 31s\n",
      " 24300K .......... .......... .......... .......... .......... 37% 3.01M 31s\n",
      " 24350K .......... .......... .......... .......... .......... 37% 1.53M 31s\n",
      " 24400K .......... .......... .......... .......... .......... 37% 2.23M 31s\n",
      " 24450K .......... .......... .......... .......... .......... 37% 2.52M 31s\n",
      " 24500K .......... .......... .......... .......... .......... 37% 2.21M 31s\n",
      " 24550K .......... .......... .......... .......... .......... 37% 2.90M 31s\n",
      " 24600K .......... .......... .......... .......... .......... 37% 2.16M 31s\n",
      " 24650K .......... .......... .......... .......... .......... 37% 1.84M 31s\n",
      " 24700K .......... .......... .......... .......... .......... 37% 2.87M 31s\n",
      " 24750K .......... .......... .......... .......... .......... 37% 2.86M 31s\n",
      " 24800K .......... .......... .......... .......... .......... 37% 2.25M 31s\n",
      " 24850K .......... .......... .......... .......... .......... 37% 2.41M 31s\n",
      " 24900K .......... .......... .......... .......... .......... 37% 1.26M 31s\n",
      " 24950K .......... .......... .......... .......... .......... 38% 1.79M 31s\n",
      " 25000K .......... .......... .......... .......... .......... 38% 2.99M 30s\n",
      " 25050K .......... .......... .......... .......... .......... 38% 2.89M 30s\n",
      " 25100K .......... .......... .......... .......... .......... 38% 1.63M 30s\n",
      " 25150K .......... .......... .......... .......... .......... 38% 2.69M 30s\n",
      " 25200K .......... .......... .......... .......... .......... 38% 1.72M 30s\n",
      " 25250K .......... .......... .......... .......... .......... 38% 1.95M 30s\n",
      " 25300K .......... .......... .......... .......... .......... 38% 2.24M 30s\n",
      " 25350K .......... .......... .......... .......... .......... 38% 2.47M 30s\n",
      " 25400K .......... .......... .......... .......... .......... 38% 2.80M 30s\n",
      " 25450K .......... .......... .......... .......... .......... 38% 2.11M 30s\n",
      " 25500K .......... .......... .......... .......... .......... 38% 2.90M 30s\n",
      " 25550K .......... .......... .......... .......... .......... 38% 2.13M 30s\n",
      " 25600K .......... .......... .......... .......... .......... 39% 2.96M 30s\n",
      " 25650K .......... .......... .......... .......... .......... 39% 2.77M 30s\n",
      " 25700K .......... .......... .......... .......... .......... 39% 2.18M 30s\n",
      " 25750K .......... .......... .......... .......... .......... 39% 1.72M 30s\n",
      " 25800K .......... .......... .......... .......... .......... 39% 2.73M 29s\n",
      " 25850K .......... .......... .......... .......... .......... 39% 1.03M 29s\n",
      " 25900K .......... .......... .......... .......... .......... 39% 2.21M 29s\n",
      " 25950K .......... .......... .......... .......... .......... 39%  724K 29s\n",
      " 26000K .......... .......... .......... .......... .......... 39%  608K 29s\n",
      " 26050K .......... .......... .......... .......... .......... 39%  694K 29s\n",
      " 26100K .......... .......... .......... .......... .......... 39% 1.01M 29s\n",
      " 26150K .......... .......... .......... .......... .......... 39%  677K 29s\n",
      " 26200K .......... .......... .......... .......... .......... 39%  715K 29s\n",
      " 26250K .......... .......... .......... .......... .......... 40%  771K 29s\n",
      " 26300K .......... .......... .......... .......... .......... 40%  849K 29s\n",
      " 26350K .......... .......... .......... .......... .......... 40%  711K 29s\n",
      " 26400K .......... .......... .......... .......... .......... 40% 1.27M 29s\n",
      " 26450K .......... .......... .......... .......... .......... 40% 1.74M 29s\n",
      " 26500K .......... .......... .......... .......... .......... 40% 2.85M 29s\n",
      " 26550K .......... .......... .......... .......... .......... 40% 2.21M 29s\n",
      " 26600K .......... .......... .......... .......... .......... 40% 1.68M 29s\n",
      " 26650K .......... .......... .......... .......... .......... 40% 4.49M 29s\n",
      " 26700K .......... .......... .......... .......... .......... 40% 2.19M 29s\n",
      " 26750K .......... .......... .......... .......... .......... 40% 2.92M 29s\n",
      " 26800K .......... .......... .......... .......... .......... 40% 1.44M 29s\n",
      " 26850K .......... .......... .......... .......... .......... 40% 2.26M 29s\n",
      " 26900K .......... .......... .......... .......... .......... 41% 2.84M 29s\n",
      " 26950K .......... .......... .......... .......... .......... 41% 2.83M 29s\n",
      " 27000K .......... .......... .......... .......... .......... 41% 2.92M 29s\n",
      " 27050K .......... .......... .......... .......... .......... 41% 2.80M 29s\n",
      " 27100K .......... .......... .......... .......... .......... 41% 2.96M 29s\n",
      " 27150K .......... .......... .......... .......... .......... 41% 1.77M 29s\n",
      " 27200K .......... .......... .......... .......... .......... 41% 2.16M 28s\n",
      " 27250K .......... .......... .......... .......... .......... 41% 3.00M 28s\n",
      " 27300K .......... .......... .......... .......... .......... 41% 2.94M 28s\n",
      " 27350K .......... .......... .......... .......... .......... 41% 3.06M 28s\n",
      " 27400K .......... .......... .......... .......... .......... 41% 2.80M 28s\n",
      " 27450K .......... .......... .......... .......... .......... 41% 2.88M 28s\n",
      " 27500K .......... .......... .......... .......... .......... 41% 2.10M 28s\n",
      " 27550K .......... .......... .......... .......... .......... 42% 2.19M 28s\n",
      " 27600K .......... .......... .......... .......... .......... 42% 2.24M 28s\n",
      " 27650K .......... .......... .......... .......... .......... 42% 1.98M 28s\n",
      " 27700K .......... .......... .......... .......... .......... 42% 2.98M 28s\n",
      " 27750K .......... .......... .......... .......... .......... 42% 2.22M 28s\n",
      " 27800K .......... .......... .......... .......... .......... 42% 3.04M 28s\n",
      " 27850K .......... .......... .......... .......... .......... 42% 2.82M 28s\n",
      " 27900K .......... .......... .......... .......... .......... 42% 2.57M 28s\n",
      " 27950K .......... .......... .......... .......... .......... 42% 2.22M 28s\n",
      " 28000K .......... .......... .......... .......... .......... 42% 2.09M 27s\n",
      " 28050K .......... .......... .......... .......... .......... 42% 3.04M 27s\n",
      " 28100K .......... .......... .......... .......... .......... 42% 2.23M 27s\n",
      " 28150K .......... .......... .......... .......... .......... 42% 2.36M 27s\n",
      " 28200K .......... .......... .......... .......... .......... 43% 2.91M 27s\n",
      " 28250K .......... .......... .......... .......... .......... 43% 2.97M 27s\n",
      " 28300K .......... .......... .......... .......... .......... 43% 2.99M 27s\n",
      " 28350K .......... .......... .......... .......... .......... 43% 2.05M 27s\n",
      " 28400K .......... .......... .......... .......... .......... 43% 1.86M 27s\n",
      " 28450K .......... .......... .......... .......... .......... 43% 1.63M 27s\n",
      " 28500K .......... .......... .......... .......... .......... 43% 2.80M 27s\n",
      " 28550K .......... .......... .......... .......... .......... 43% 2.82M 27s\n",
      " 28600K .......... .......... .......... .......... .......... 43% 2.90M 27s\n",
      " 28650K .......... .......... .......... .......... .......... 43% 2.09M 27s\n",
      " 28700K .......... .......... .......... .......... .......... 43% 1.17M 27s\n",
      " 28750K .......... .......... .......... .......... .......... 43%  609K 27s\n",
      " 28800K .......... .......... .......... .......... .......... 43%  580K 27s\n",
      " 28850K .......... .......... .......... .......... .......... 44%  738K 27s\n",
      " 28900K .......... .......... .......... .......... .......... 44%  715K 27s\n",
      " 28950K .......... .......... .......... .......... .......... 44%  857K 27s\n",
      " 29000K .......... .......... .......... .......... .......... 44%  694K 27s\n",
      " 29050K .......... .......... .......... .......... .......... 44%  725K 27s\n",
      " 29100K .......... .......... .......... .......... .......... 44%  702K 27s\n",
      " 29150K .......... .......... .......... .......... .......... 44% 2.07M 27s\n",
      " 29200K .......... .......... .......... .......... .......... 44% 1.67M 27s\n",
      " 29250K .......... .......... .......... .......... .......... 44% 2.99M 27s\n",
      " 29300K .......... .......... .......... .......... .......... 44% 2.94M 27s\n",
      " 29350K .......... .......... .......... .......... .......... 44% 2.14M 27s\n",
      " 29400K .......... .......... .......... .......... .......... 44% 2.96M 26s\n",
      " 29450K .......... .......... .......... .......... .......... 44% 2.87M 26s\n",
      " 29500K .......... .......... .......... .......... .......... 44% 2.15M 26s\n",
      " 29550K .......... .......... .......... .......... .......... 45% 2.50M 26s\n",
      " 29600K .......... .......... .......... .......... .......... 45% 2.45M 26s\n",
      " 29650K .......... .......... .......... .......... .......... 45% 3.08M 26s\n",
      " 29700K .......... .......... .......... .......... .......... 45% 2.08M 26s\n",
      " 29750K .......... .......... .......... .......... .......... 45% 1.62M 26s\n",
      " 29800K .......... .......... .......... .......... .......... 45% 1.86M 26s\n",
      " 29850K .......... .......... .......... .......... .......... 45% 2.06M 26s\n",
      " 29900K .......... .......... .......... .......... .......... 45% 2.80M 26s\n",
      " 29950K .......... .......... .......... .......... .......... 45% 1.82M 26s\n",
      " 30000K .......... .......... .......... .......... .......... 45% 1.37M 26s\n",
      " 30050K .......... .......... .......... .......... .......... 45% 1.78M 26s\n",
      " 30100K .......... .......... .......... .......... .......... 45% 2.85M 26s\n",
      " 30150K .......... .......... .......... .......... .......... 45% 1.48M 26s\n",
      " 30200K .......... .......... .......... .......... .......... 46% 1.47M 26s\n",
      " 30250K .......... .......... .......... .......... .......... 46% 1.50M 26s\n",
      " 30300K .......... .......... .......... .......... .......... 46% 1.66M 26s\n",
      " 30350K .......... .......... .......... .......... .......... 46% 2.00M 25s\n",
      " 30400K .......... .......... .......... .......... .......... 46% 1.68M 25s\n",
      " 30450K .......... .......... .......... .......... .......... 46% 2.17M 25s\n",
      " 30500K .......... .......... .......... .......... .......... 46% 4.39M 25s\n",
      " 30550K .......... .......... .......... .......... .......... 46% 1.77M 25s\n",
      " 30600K .......... .......... .......... .......... .......... 46% 2.97M 25s\n",
      " 30650K .......... .......... .......... .......... .......... 46% 1.40M 25s\n",
      " 30700K .......... .......... .......... .......... .......... 46% 2.17M 25s\n",
      " 30750K .......... .......... .......... .......... .......... 46% 3.03M 25s\n",
      " 30800K .......... .......... .......... .......... .......... 46% 2.18M 25s\n",
      " 30850K .......... .......... .......... .......... .......... 47% 3.07M 25s\n",
      " 30900K .......... .......... .......... .......... .......... 47% 1.74M 25s\n",
      " 30950K .......... .......... .......... .......... .......... 47% 3.01M 25s\n",
      " 31000K .......... .......... .......... .......... .......... 47% 2.83M 25s\n",
      " 31050K .......... .......... .......... .......... .......... 47% 3.65M 25s\n",
      " 31100K .......... .......... .......... .......... .......... 47% 1.70M 25s\n",
      " 31150K .......... .......... .......... .......... .......... 47% 4.00M 25s\n",
      " 31200K .......... .......... .......... .......... .......... 47% 1.25M 25s\n",
      " 31250K .......... .......... .......... .......... .......... 47% 2.07M 25s\n",
      " 31300K .......... .......... .......... .......... .......... 47% 2.87M 25s\n",
      " 31350K .......... .......... .......... .......... .......... 47% 2.13M 24s\n",
      " 31400K .......... .......... .......... .......... .......... 47% 2.74M 24s\n",
      " 31450K .......... .......... .......... .......... .......... 47% 1.26M 24s\n",
      " 31500K .......... .......... .......... .......... .......... 48% 2.36M 24s\n",
      " 31550K .......... .......... .......... .......... .......... 48% 2.17M 24s\n",
      " 31600K .......... .......... .......... .......... .......... 48% 2.07M 24s\n",
      " 31650K .......... .......... .......... .......... .......... 48% 2.24M 24s\n",
      " 31700K .......... .......... .......... .......... .......... 48% 1.82M 24s\n",
      " 31750K .......... .......... .......... .......... .......... 48% 3.01M 24s\n",
      " 31800K .......... .......... .......... .......... .......... 48% 2.80M 24s\n",
      " 31850K .......... .......... .......... .......... .......... 48% 2.92M 24s\n",
      " 31900K .......... .......... .......... .......... .......... 48% 2.95M 24s\n",
      " 31950K .......... .......... .......... .......... .......... 48% 3.08M 24s\n",
      " 32000K .......... .......... .......... .......... .......... 48% 1.53M 24s\n",
      " 32050K .......... .......... .......... .......... .......... 48% 2.18M 24s\n",
      " 32100K .......... .......... .......... .......... .......... 48% 3.94M 24s\n",
      " 32150K .......... .......... .......... .......... .......... 49% 2.98M 24s\n",
      " 32200K .......... .......... .......... .......... .......... 49% 1.84M 24s\n",
      " 32250K .......... .......... .......... .......... .......... 49% 2.14M 24s\n",
      " 32300K .......... .......... .......... .......... .......... 49% 2.69M 23s\n",
      " 32350K .......... .......... .......... .......... .......... 49% 2.03M 23s\n",
      " 32400K .......... .......... .......... .......... .......... 49% 2.15M 23s\n",
      " 32450K .......... .......... .......... .......... .......... 49% 1.29M 23s\n",
      " 32500K .......... .......... .......... .......... .......... 49%  755K 23s\n",
      " 32550K .......... .......... .......... .......... .......... 49%  905K 23s\n",
      " 32600K .......... .......... .......... .......... .......... 49%  637K 23s\n",
      " 32650K .......... .......... .......... .......... .......... 49%  844K 23s\n",
      " 32700K .......... .......... .......... .......... .......... 49%  866K 23s\n",
      " 32750K .......... .......... .......... .......... .......... 49%  711K 23s\n",
      " 32800K .......... .......... .......... .......... .......... 50%  502K 23s\n",
      " 32850K .......... .......... .......... .......... .......... 50%  658K 23s\n",
      " 32900K .......... .......... .......... .......... .......... 50% 1.49M 23s\n",
      " 32950K .......... .......... .......... .......... .......... 50% 2.89M 23s\n",
      " 33000K .......... .......... .......... .......... .......... 50% 2.80M 23s\n",
      " 33050K .......... .......... .......... .......... .......... 50% 2.04M 23s\n",
      " 33100K .......... .......... .......... .......... .......... 50% 3.23M 23s\n",
      " 33150K .......... .......... .......... .......... .......... 50% 1.98M 23s\n",
      " 33200K .......... .......... .......... .......... .......... 50% 1.51M 23s\n",
      " 33250K .......... .......... .......... .......... .......... 50% 2.94M 23s\n",
      " 33300K .......... .......... .......... .......... .......... 50% 2.09M 23s\n",
      " 33350K .......... .......... .......... .......... .......... 50% 3.02M 23s\n",
      " 33400K .......... .......... .......... .......... .......... 50% 2.26M 23s\n",
      " 33450K .......... .......... .......... .......... .......... 51% 3.05M 23s\n",
      " 33500K .......... .......... .......... .......... .......... 51% 1.74M 23s\n",
      " 33550K .......... .......... .......... .......... .......... 51% 2.86M 23s\n",
      " 33600K .......... .......... .......... .......... .......... 51% 2.25M 23s\n",
      " 33650K .......... .......... .......... .......... .......... 51% 2.73M 23s\n",
      " 33700K .......... .......... .......... .......... .......... 51% 3.18M 23s\n",
      " 33750K .......... .......... .......... .......... .......... 51% 2.99M 22s\n",
      " 33800K .......... .......... .......... .......... .......... 51% 2.95M 22s\n",
      " 33850K .......... .......... .......... .......... .......... 51% 2.20M 22s\n",
      " 33900K .......... .......... .......... .......... .......... 51% 3.06M 22s\n",
      " 33950K .......... .......... .......... .......... .......... 51% 2.61M 22s\n",
      " 34000K .......... .......... .......... .......... .......... 51% 1.77M 22s\n",
      " 34050K .......... .......... .......... .......... .......... 51% 2.75M 22s\n",
      " 34100K .......... .......... .......... .......... .......... 52% 2.76M 22s\n",
      " 34150K .......... .......... .......... .......... .......... 52% 1.73M 22s\n",
      " 34200K .......... .......... .......... .......... .......... 52% 4.52M 22s\n",
      " 34250K .......... .......... .......... .......... .......... 52% 1.39M 22s\n",
      " 34300K .......... .......... .......... .......... .......... 52% 2.71M 22s\n",
      " 34350K .......... .......... .......... .......... .......... 52% 2.85M 22s\n",
      " 34400K .......... .......... .......... .......... .......... 52% 2.12M 22s\n",
      " 34450K .......... .......... .......... .......... .......... 52% 2.16M 22s\n",
      " 34500K .......... .......... .......... .......... .......... 52% 2.69M 22s\n",
      " 34550K .......... .......... .......... .......... .......... 52% 2.10M 22s\n",
      " 34600K .......... .......... .......... .......... .......... 52% 1.77M 22s\n",
      " 34650K .......... .......... .......... .......... .......... 52% 1.73M 22s\n",
      " 34700K .......... .......... .......... .......... .......... 52% 2.91M 22s\n",
      " 34750K .......... .......... .......... .......... .......... 52% 3.00M 22s\n",
      " 34800K .......... .......... .......... .......... .......... 53% 2.25M 21s\n",
      " 34850K .......... .......... .......... .......... .......... 53% 2.28M 21s\n",
      " 34900K .......... .......... .......... .......... .......... 53% 4.13M 21s\n",
      " 34950K .......... .......... .......... .......... .......... 53% 2.35M 21s\n",
      " 35000K .......... .......... .......... .......... .......... 53% 2.90M 21s\n",
      " 35050K .......... .......... .......... .......... .......... 53% 2.23M 21s\n",
      " 35100K .......... .......... .......... .......... .......... 53% 2.90M 21s\n",
      " 35150K .......... .......... .......... .......... .......... 53% 1.84M 21s\n",
      " 35200K .......... .......... .......... .......... .......... 53% 1.81M 21s\n",
      " 35250K .......... .......... .......... .......... .......... 53% 2.98M 21s\n",
      " 35300K .......... .......... .......... .......... .......... 53% 2.89M 21s\n",
      " 35350K .......... .......... .......... .......... .......... 53% 2.94M 21s\n",
      " 35400K .......... .......... .......... .......... .......... 53% 2.38M 21s\n",
      " 35450K .......... .......... .......... .......... .......... 54% 2.36M 21s\n",
      " 35500K .......... .......... .......... .......... .......... 54% 1.72M 21s\n",
      " 35550K .......... .......... .......... .......... .......... 54% 2.89M 21s\n",
      " 35600K .......... .......... .......... .......... .......... 54% 1.91M 21s\n",
      " 35650K .......... .......... .......... .......... .......... 54% 2.31M 21s\n",
      " 35700K .......... .......... .......... .......... .......... 54% 3.08M 21s\n",
      " 35750K .......... .......... .......... .......... .......... 54% 2.81M 21s\n",
      " 35800K .......... .......... .......... .......... .......... 54% 2.72M 21s\n",
      " 35850K .......... .......... .......... .......... .......... 54% 2.98M 20s\n",
      " 35900K .......... .......... .......... .......... .......... 54% 2.68M 20s\n",
      " 35950K .......... .......... .......... .......... .......... 54% 2.92M 20s\n",
      " 36000K .......... .......... .......... .......... .......... 54% 1.48M 20s\n",
      " 36050K .......... .......... .......... .......... .......... 54% 2.84M 20s\n",
      " 36100K .......... .......... .......... .......... .......... 55% 2.24M 20s\n",
      " 36150K .......... .......... .......... .......... .......... 55% 3.02M 20s\n",
      " 36200K .......... .......... .......... .......... .......... 55% 2.85M 20s\n",
      " 36250K .......... .......... .......... .......... .......... 55% 3.13M 20s\n",
      " 36300K .......... .......... .......... .......... .......... 55% 2.20M 20s\n",
      " 36350K .......... .......... .......... .......... .......... 55% 1.14M 20s\n",
      " 36400K .......... .......... .......... .......... .......... 55%  473K 20s\n",
      " 36450K .......... .......... .......... .......... .......... 55% 1.03M 20s\n",
      " 36500K .......... .......... .......... .......... .......... 55%  696K 20s\n",
      " 36550K .......... .......... .......... .......... .......... 55%  919K 20s\n",
      " 36600K .......... .......... .......... .......... .......... 55% 1010K 20s\n",
      " 36650K .......... .......... .......... .......... .......... 55%  731K 20s\n",
      " 36700K .......... .......... .......... .......... .......... 55%  915K 20s\n",
      " 36750K .......... .......... .......... .......... .......... 56%  961K 20s\n",
      " 36800K .......... .......... .......... .......... .......... 56% 2.27M 20s\n",
      " 36850K .......... .......... .......... .......... .......... 56% 2.96M 20s\n",
      " 36900K .......... .......... .......... .......... .......... 56% 3.06M 20s\n",
      " 36950K .......... .......... .......... .......... .......... 56% 2.96M 20s\n",
      " 37000K .......... .......... .......... .......... .......... 56% 2.99M 20s\n",
      " 37050K .......... .......... .......... .......... .......... 56% 3.04M 20s\n",
      " 37100K .......... .......... .......... .......... .......... 56% 3.11M 20s\n",
      " 37150K .......... .......... .......... .......... .......... 56% 2.13M 20s\n",
      " 37200K .......... .......... .......... .......... .......... 56% 1.59M 20s\n",
      " 37250K .......... .......... .......... .......... .......... 56% 2.21M 19s\n",
      " 37300K .......... .......... .......... .......... .......... 56% 2.81M 19s\n",
      " 37350K .......... .......... .......... .......... .......... 56% 2.12M 19s\n",
      " 37400K .......... .......... .......... .......... .......... 57% 3.02M 19s\n",
      " 37450K .......... .......... .......... .......... .......... 57% 3.07M 19s\n",
      " 37500K .......... .......... .......... .......... .......... 57% 2.89M 19s\n",
      " 37550K .......... .......... .......... .......... .......... 57% 2.20M 19s\n",
      " 37600K .......... .......... .......... .......... .......... 57% 2.30M 19s\n",
      " 37650K .......... .......... .......... .......... .......... 57% 2.99M 19s\n",
      " 37700K .......... .......... .......... .......... .......... 57% 1.81M 19s\n",
      " 37750K .......... .......... .......... .......... .......... 57% 2.90M 19s\n",
      " 37800K .......... .......... .......... .......... .......... 57% 2.87M 19s\n",
      " 37850K .......... .......... .......... .......... .......... 57% 2.27M 19s\n",
      " 37900K .......... .......... .......... .......... .......... 57%  280K 19s\n",
      " 37950K .......... .......... .......... .......... .......... 57% 36.9M 19s\n",
      " 38000K .......... .......... .......... .......... .......... 57% 3.34M 19s\n",
      " 38050K .......... .......... .......... .......... .......... 58% 2.23M 19s\n",
      " 38100K .......... .......... .......... .......... .......... 58% 2.11M 19s\n",
      " 38150K .......... .......... .......... .......... .......... 58% 2.83M 19s\n",
      " 38200K .......... .......... .......... .......... .......... 58% 2.21M 19s\n",
      " 38250K .......... .......... .......... .......... .......... 58% 3.01M 19s\n",
      " 38300K .......... .......... .......... .......... .......... 58% 2.87M 19s\n",
      " 38350K .......... .......... .......... .......... .......... 58% 2.33M 19s\n",
      " 38400K .......... .......... .......... .......... .......... 58% 1.78M 19s\n",
      " 38450K .......... .......... .......... .......... .......... 58% 2.99M 19s\n",
      " 38500K .......... .......... .......... .......... .......... 58% 2.95M 18s\n",
      " 38550K .......... .......... .......... .......... .......... 58% 3.02M 18s\n",
      " 38600K .......... .......... .......... .......... .......... 58% 2.95M 18s\n",
      " 38650K .......... .......... .......... .......... .......... 58% 3.01M 18s\n",
      " 38700K .......... .......... .......... .......... .......... 59% 1.83M 18s\n",
      " 38750K .......... .......... .......... .......... .......... 59% 1.97M 18s\n",
      " 38800K .......... .......... .......... .......... .......... 59% 2.43M 18s\n",
      " 38850K .......... .......... .......... .......... .......... 59% 2.94M 18s\n",
      " 38900K .......... .......... .......... .......... .......... 59% 2.93M 18s\n",
      " 38950K .......... .......... .......... .......... .......... 59% 2.88M 18s\n",
      " 39000K .......... .......... .......... .......... .......... 59% 2.84M 18s\n",
      " 39050K .......... .......... .......... .......... .......... 59% 2.96M 18s\n",
      " 39100K .......... .......... .......... .......... .......... 59% 1.80M 18s\n",
      " 39150K .......... .......... .......... .......... .......... 59% 2.27M 18s\n",
      " 39200K .......... .......... .......... .......... .......... 59% 1.71M 18s\n",
      " 39250K .......... .......... .......... .......... .......... 59% 1.13M 18s\n",
      " 39300K .......... .......... .......... .......... .......... 59% 2.41M 18s\n",
      " 39350K .......... .......... .......... .......... .......... 59%  566K 18s\n",
      " 39400K .......... .......... .......... .......... .......... 60%  730K 18s\n",
      " 39450K .......... .......... .......... .......... .......... 60%  727K 18s\n",
      " 39500K .......... .......... .......... .......... .......... 60%  852K 18s\n",
      " 39550K .......... .......... .......... .......... .......... 60%  665K 18s\n",
      " 39600K .......... .......... .......... .......... .......... 60%  635K 18s\n",
      " 39650K .......... .......... .......... .......... .......... 60%  702K 18s\n",
      " 39700K .......... .......... .......... .......... .......... 60%  688K 18s\n",
      " 39750K .......... .......... .......... .......... .......... 60%  988K 18s\n",
      " 39800K .......... .......... .......... .......... .......... 60% 2.80M 18s\n",
      " 39850K .......... .......... .......... .......... .......... 60% 1.30M 18s\n",
      " 39900K .......... .......... .......... .......... .......... 60% 2.96M 18s\n",
      " 39950K .......... .......... .......... .......... .......... 60% 2.92M 18s\n",
      " 40000K .......... .......... .......... .......... .......... 60% 1.81M 17s\n",
      " 40050K .......... .......... .......... .......... .......... 61% 1.80M 17s\n",
      " 40100K .......... .......... .......... .......... .......... 61% 2.99M 17s\n",
      " 40150K .......... .......... .......... .......... .......... 61% 2.28M 17s\n",
      " 40200K .......... .......... .......... .......... .......... 61% 2.24M 17s\n",
      " 40250K .......... .......... .......... .......... .......... 61% 3.05M 17s\n",
      " 40300K .......... .......... .......... .......... .......... 61% 2.20M 17s\n",
      " 40350K .......... .......... .......... .......... .......... 61% 2.99M 17s\n",
      " 40400K .......... .......... .......... .......... .......... 61% 2.18M 17s\n",
      " 40450K .......... .......... .......... .......... .......... 61% 2.03M 17s\n",
      " 40500K .......... .......... .......... .......... .......... 61% 1.41M 17s\n",
      " 40550K .......... .......... .......... .......... .......... 61% 2.21M 17s\n",
      " 40600K .......... .......... .......... .......... .......... 61% 2.27M 17s\n",
      " 40650K .......... .......... .......... .......... .......... 61% 2.99M 17s\n",
      " 40700K .......... .......... .......... .......... .......... 62% 2.10M 17s\n",
      " 40750K .......... .......... .......... .......... .......... 62% 2.64M 17s\n",
      " 40800K .......... .......... .......... .......... .......... 62% 1.58M 17s\n",
      " 40850K .......... .......... .......... .......... .......... 62% 3.05M 17s\n",
      " 40900K .......... .......... .......... .......... .......... 62% 1.42M 17s\n",
      " 40950K .......... .......... .......... .......... .......... 62% 2.10M 17s\n",
      " 41000K .......... .......... .......... .......... .......... 62% 3.88M 17s\n",
      " 41050K .......... .......... .......... .......... .......... 62% 1.74M 17s\n",
      " 41100K .......... .......... .......... .......... .......... 62% 4.61M 17s\n",
      " 41150K .......... .......... .......... .......... .......... 62% 2.67M 17s\n",
      " 41200K .......... .......... .......... .......... .......... 62% 1.66M 16s\n",
      " 41250K .......... .......... .......... .......... .......... 62% 2.96M 16s\n",
      " 41300K .......... .......... .......... .......... .......... 62% 2.89M 16s\n",
      " 41350K .......... .......... .......... .......... .......... 63%  721K 16s\n",
      " 41400K .......... .......... .......... .......... .......... 63%  916K 16s\n",
      " 41450K .......... .......... .......... .......... .......... 63%  680K 16s\n",
      " 41500K .......... .......... .......... .......... .......... 63%  701K 16s\n",
      " 41550K .......... .......... .......... .......... .......... 63% 1.02M 16s\n",
      " 41600K .......... .......... .......... .......... .......... 63%  520K 16s\n",
      " 41650K .......... .......... .......... .......... .......... 63%  806K 16s\n",
      " 41700K .......... .......... .......... .......... .......... 63%  623K 16s\n",
      " 41750K .......... .......... .......... .......... .......... 63% 1.18M 16s\n",
      " 41800K .......... .......... .......... .......... .......... 63% 2.89M 16s\n",
      " 41850K .......... .......... .......... .......... .......... 63% 2.11M 16s\n",
      " 41900K .......... .......... .......... .......... .......... 63% 5.06M 16s\n",
      " 41950K .......... .......... .......... .......... .......... 63% 3.04M 16s\n",
      " 42000K .......... .......... .......... .......... .......... 64% 1.36M 16s\n",
      " 42050K .......... .......... .......... .......... .......... 64% 1.49M 16s\n",
      " 42100K .......... .......... .......... .......... .......... 64% 1.50M 16s\n",
      " 42150K .......... .......... .......... .......... .......... 64% 1.72M 16s\n",
      " 42200K .......... .......... .......... .......... .......... 64% 2.74M 16s\n",
      " 42250K .......... .......... .......... .......... .......... 64% 2.41M 16s\n",
      " 42300K .......... .......... .......... .......... .......... 64% 2.95M 16s\n",
      " 42350K .......... .......... .......... .......... .......... 64% 2.19M 16s\n",
      " 42400K .......... .......... .......... .......... .......... 64% 1.51M 16s\n",
      " 42450K .......... .......... .......... .......... .......... 64% 1.80M 16s\n",
      " 42500K .......... .......... .......... .......... .......... 64% 2.84M 16s\n",
      " 42550K .......... .......... .......... .......... .......... 64% 2.61M 16s\n",
      " 42600K .......... .......... .......... .......... .......... 64% 1.28M 16s\n",
      " 42650K .......... .......... .......... .......... .......... 65% 1018K 16s\n",
      " 42700K .......... .......... .......... .......... .......... 65%  957K 16s\n",
      " 42750K .......... .......... .......... .......... .......... 65% 1.17M 16s\n",
      " 42800K .......... .......... .......... .......... .......... 65%  915K 15s\n",
      " 42850K .......... .......... .......... .......... .......... 65%  769K 15s\n",
      " 42900K .......... .......... .......... .......... .......... 65%  786K 15s\n",
      " 42950K .......... .......... .......... .......... .......... 65%  774K 15s\n",
      " 43000K .......... .......... .......... .......... .......... 65%  920K 15s\n",
      " 43050K .......... .......... .......... .......... .......... 65% 2.95M 15s\n",
      " 43100K .......... .......... .......... .......... .......... 65% 2.75M 15s\n",
      " 43150K .......... .......... .......... .......... .......... 65% 2.79M 15s\n",
      " 43200K .......... .......... .......... .......... .......... 65% 2.10M 15s\n",
      " 43250K .......... .......... .......... .......... .......... 65% 2.93M 15s\n",
      " 43300K .......... .......... .......... .......... .......... 66% 2.07M 15s\n",
      " 43350K .......... .......... .......... .......... .......... 66% 2.97M 15s\n",
      " 43400K .......... .......... .......... .......... .......... 66% 2.86M 15s\n",
      " 43450K .......... .......... .......... .......... .......... 66% 2.07M 15s\n",
      " 43500K .......... .......... .......... .......... .......... 66% 3.13M 15s\n",
      " 43550K .......... .......... .......... .......... .......... 66% 2.72M 15s\n",
      " 43600K .......... .......... .......... .......... .......... 66% 1.48M 15s\n",
      " 43650K .......... .......... .......... .......... .......... 66% 2.83M 15s\n",
      " 43700K .......... .......... .......... .......... .......... 66% 3.04M 15s\n",
      " 43750K .......... .......... .......... .......... .......... 66% 2.99M 15s\n",
      " 43800K .......... .......... .......... .......... .......... 66% 2.00M 15s\n",
      " 43850K .......... .......... .......... .......... .......... 66% 2.21M 15s\n",
      " 43900K .......... .......... .......... .......... .......... 66% 2.27M 15s\n",
      " 43950K .......... .......... .......... .......... .......... 66% 2.87M 15s\n",
      " 44000K .......... .......... .......... .......... .......... 67% 2.26M 15s\n",
      " 44050K .......... .......... .......... .......... .......... 67% 2.70M 15s\n",
      " 44100K .......... .......... .......... .......... .......... 67% 2.01M 15s\n",
      " 44150K .......... .......... .......... .......... .......... 67% 2.99M 14s\n",
      " 44200K .......... .......... .......... .......... .......... 67% 3.33M 14s\n",
      " 44250K .......... .......... .......... .......... .......... 67% 2.73M 14s\n",
      " 44300K .......... .......... .......... .......... .......... 67% 2.22M 14s\n",
      " 44350K .......... .......... .......... .......... .......... 67% 2.98M 14s\n",
      " 44400K .......... .......... .......... .......... .......... 67% 1.78M 14s\n",
      " 44450K .......... .......... .......... .......... .......... 67% 2.96M 14s\n",
      " 44500K .......... .......... .......... .......... .......... 67% 2.94M 14s\n",
      " 44550K .......... .......... .......... .......... .......... 67% 2.93M 14s\n",
      " 44600K .......... .......... .......... .......... .......... 67% 3.02M 14s\n",
      " 44650K .......... .......... .......... .......... .......... 68% 2.25M 14s\n",
      " 44700K .......... .......... .......... .......... .......... 68% 3.01M 14s\n",
      " 44750K .......... .......... .......... .......... .......... 68% 4.29M 14s\n",
      " 44800K .......... .......... .......... .......... .......... 68% 2.24M 14s\n",
      " 44850K .......... .......... .......... .......... .......... 68% 2.17M 14s\n",
      " 44900K .......... .......... .......... .......... .......... 68% 2.96M 14s\n",
      " 44950K .......... .......... .......... .......... .......... 68% 2.91M 14s\n",
      " 45000K .......... .......... .......... .......... .......... 68% 2.19M 14s\n",
      " 45050K .......... .......... .......... .......... .......... 68% 2.25M 14s\n",
      " 45100K .......... .......... .......... .......... .......... 68% 2.93M 14s\n",
      " 45150K .......... .......... .......... .......... .......... 68% 2.00M 14s\n",
      " 45200K .......... .......... .......... .......... .......... 68% 1.72M 14s\n",
      " 45250K .......... .......... .......... .......... .......... 68% 2.75M 14s\n",
      " 45300K .......... .......... .......... .......... .......... 69% 3.05M 14s\n",
      " 45350K .......... .......... .......... .......... .......... 69% 2.23M 14s\n",
      " 45400K .......... .......... .......... .......... .......... 69% 2.24M 13s\n",
      " 45450K .......... .......... .......... .......... .......... 69% 2.28M 13s\n",
      " 45500K .......... .......... .......... .......... .......... 69% 2.95M 13s\n",
      " 45550K .......... .......... .......... .......... .......... 69% 1.10M 13s\n",
      " 45600K .......... .......... .......... .......... .......... 69% 1.55M 13s\n",
      " 45650K .......... .......... .......... .......... .......... 69% 3.02M 13s\n",
      " 45700K .......... .......... .......... .......... .......... 69% 2.16M 13s\n",
      " 45750K .......... .......... .......... .......... .......... 69% 2.47M 13s\n",
      " 45800K .......... .......... .......... .......... .......... 69% 1.85M 13s\n",
      " 45850K .......... .......... .......... .......... .......... 69% 2.28M 13s\n",
      " 45900K .......... .......... .......... .......... .......... 69% 4.31M 13s\n",
      " 45950K .......... .......... .......... .......... .......... 70% 2.02M 13s\n",
      " 46000K .......... .......... .......... .......... .......... 70% 1.96M 13s\n",
      " 46050K .......... .......... .......... .......... .......... 70% 2.68M 13s\n",
      " 46100K .......... .......... .......... .......... .......... 70% 2.93M 13s\n",
      " 46150K .......... .......... .......... .......... .......... 70% 1.94M 13s\n",
      " 46200K .......... .......... .......... .......... .......... 70% 2.89M 13s\n",
      " 46250K .......... .......... .......... .......... .......... 70% 2.03M 13s\n",
      " 46300K .......... .......... .......... .......... .......... 70% 2.57M 13s\n",
      " 46350K .......... .......... .......... .......... .......... 70% 2.13M 13s\n",
      " 46400K .......... .......... .......... .......... .......... 70% 2.12M 13s\n",
      " 46450K .......... .......... .......... .......... .......... 70% 3.01M 13s\n",
      " 46500K .......... .......... .......... .......... .......... 70% 2.57M 13s\n",
      " 46550K .......... .......... .......... .......... .......... 70% 2.16M 13s\n",
      " 46600K .......... .......... .......... .......... .......... 71% 2.94M 13s\n",
      " 46650K .......... .......... .......... .......... .......... 71% 2.82M 13s\n",
      " 46700K .......... .......... .......... .......... .......... 71% 2.88M 12s\n",
      " 46750K .......... .......... .......... .......... .......... 71% 1.81M 12s\n",
      " 46800K .......... .......... .......... .......... .......... 71% 1.01M 12s\n",
      " 46850K .......... .......... .......... .......... .......... 71%  627K 12s\n",
      " 46900K .......... .......... .......... .......... .......... 71%  755K 12s\n",
      " 46950K .......... .......... .......... .......... .......... 71%  842K 12s\n",
      " 47000K .......... .......... .......... .......... .......... 71%  996K 12s\n",
      " 47050K .......... .......... .......... .......... .......... 71%  912K 12s\n",
      " 47100K .......... .......... .......... .......... .......... 71%  886K 12s\n",
      " 47150K .......... .......... .......... .......... .......... 71%  763K 12s\n",
      " 47200K .......... .......... .......... .......... .......... 71%  698K 12s\n",
      " 47250K .......... .......... .......... .......... .......... 72% 2.29M 12s\n",
      " 47300K .......... .......... .......... .......... .......... 72% 2.79M 12s\n",
      " 47350K .......... .......... .......... .......... .......... 72%  865K 12s\n",
      " 47400K .......... .......... .......... .......... .......... 72% 2.14M 12s\n",
      " 47450K .......... .......... .......... .......... .......... 72%  515K 12s\n",
      " 47500K .......... .......... .......... .......... .......... 72%  929K 12s\n",
      " 47550K .......... .......... .......... .......... .......... 72%  857K 12s\n",
      " 47600K .......... .......... .......... .......... .......... 72%  781K 12s\n",
      " 47650K .......... .......... .......... .......... .......... 72% 1.24M 12s\n",
      " 47700K .......... .......... .......... .......... .......... 72% 2.14M 12s\n",
      " 47750K .......... .......... .......... .......... .......... 72% 2.06M 12s\n",
      " 47800K .......... .......... .......... .......... .......... 72% 1.24M 12s\n",
      " 47850K .......... .......... .......... .......... .......... 72% 1.16M 12s\n",
      " 47900K .......... .......... .......... .......... .......... 73%  700K 12s\n",
      " 47950K .......... .......... .......... .......... .......... 73%  877K 12s\n",
      " 48000K .......... .......... .......... .......... .......... 73%  689K 12s\n",
      " 48050K .......... .......... .......... .......... .......... 73% 1.10M 12s\n",
      " 48100K .......... .......... .......... .......... .......... 73%  716K 12s\n",
      " 48150K .......... .......... .......... .......... .......... 73%  877K 12s\n",
      " 48200K .......... .......... .......... .......... .......... 73% 1.07M 12s\n",
      " 48250K .......... .......... .......... .......... .......... 73% 19.1K 13s\n",
      " 48300K .......... .......... .......... .......... .......... 73% 26.7M 13s\n",
      " 48350K .......... .......... .......... .......... .......... 73% 37.6M 12s\n",
      " 48400K .......... .......... .......... .......... .......... 73%  338K 12s\n",
      " 48450K .......... .......... .......... .......... .......... 73% 3.29M 12s\n",
      " 48500K .......... .......... .......... .......... .......... 73%  711K 12s\n",
      " 48550K .......... .......... .......... .......... .......... 74% 36.9M 12s\n",
      " 48600K .......... .......... .......... .......... .......... 74% 1008K 12s\n",
      " 48650K .......... .......... .......... .......... .......... 74% 3.01M 12s\n",
      " 48700K .......... .......... .......... .......... .......... 74% 34.5M 12s\n",
      " 48750K .......... .......... .......... .......... .......... 74% 30.6M 12s\n",
      " 48800K .......... .......... .......... .......... .......... 74%  443K 12s\n",
      " 48850K .......... .......... .......... .......... .......... 74% 40.5M 12s\n",
      " 48900K .......... .......... .......... .......... .......... 74% 28.8M 12s\n",
      " 48950K .......... .......... .......... .......... .......... 74% 30.1M 12s\n",
      " 49000K .......... .......... .......... .......... .......... 74% 30.4M 12s\n",
      " 49050K .......... .......... .......... .......... .......... 74% 29.3M 12s\n",
      " 49100K .......... .......... .......... .......... .......... 74% 41.5M 12s\n",
      " 49150K .......... .......... .......... .......... .......... 74% 34.2M 12s\n",
      " 49200K .......... .......... .......... .......... .......... 74% 19.7M 12s\n",
      " 49250K .......... .......... .......... .......... .......... 75% 31.6M 12s\n",
      " 49300K .......... .......... .......... .......... .......... 75% 33.1M 12s\n",
      " 49350K .......... .......... .......... .......... .......... 75% 32.2M 12s\n",
      " 49400K .......... .......... .......... .......... .......... 75% 26.9M 12s\n",
      " 49450K .......... .......... .......... .......... .......... 75% 30.0M 12s\n",
      " 49500K .......... .......... .......... .......... .......... 75% 43.9M 12s\n",
      " 49550K .......... .......... .......... .......... .......... 75% 51.8M 11s\n",
      " 49600K .......... .......... .......... .......... .......... 75% 25.7M 11s\n",
      " 49650K .......... .......... .......... .......... .......... 75%  135M 11s\n",
      " 49700K .......... .......... .......... .......... .......... 75%  119M 11s\n",
      " 49750K .......... .......... .......... .......... .......... 75%  139M 11s\n",
      " 49800K .......... .......... .......... .......... .......... 75%  102M 11s\n",
      " 49850K .......... .......... .......... .......... .......... 75%  115M 11s\n",
      " 49900K .......... .......... .......... .......... .......... 76% 13.8M 11s\n",
      " 49950K .......... .......... .......... .......... .......... 76%  581K 11s\n",
      " 50000K .......... .......... .......... .......... .......... 76%  468K 11s\n",
      " 50050K .......... .......... .......... .......... .......... 76%  777K 11s\n",
      " 50100K .......... .......... .......... .......... .......... 76% 1.17M 11s\n",
      " 50150K .......... .......... .......... .......... .......... 76%  608K 11s\n",
      " 50200K .......... .......... .......... .......... .......... 76%  736K 11s\n",
      " 50250K .......... .......... .......... .......... .......... 76%  897K 11s\n",
      " 50300K .......... .......... .......... .......... .......... 76% 1.55M 11s\n",
      " 50350K .......... .......... .......... .......... .......... 76% 1.10M 11s\n",
      " 50400K .......... .......... .......... .......... .......... 76%  744K 11s\n",
      " 50450K .......... .......... .......... .......... .......... 76%  551K 11s\n",
      " 50500K .......... .......... .......... .......... .......... 76%  686K 11s\n",
      " 50550K .......... .......... .......... .......... .......... 77% 44.3K 11s\n",
      " 50600K .......... .......... .......... .......... .......... 77% 54.8M 11s\n",
      " 50650K .......... .......... .......... .......... .......... 77%  760K 11s\n",
      " 50700K .......... .......... .......... .......... .......... 77% 79.8M 11s\n",
      " 50750K .......... .......... .......... .......... .......... 77% 1.42M 11s\n",
      " 50800K .......... .......... .......... .......... .......... 77% 69.1M 11s\n",
      " 50850K .......... .......... .......... .......... .......... 77%  121M 11s\n",
      " 50900K .......... .......... .......... .......... .......... 77%  127M 11s\n",
      " 50950K .......... .......... .......... .......... .......... 77%  121M 11s\n",
      " 51000K .......... .......... .......... .......... .......... 77%  112M 11s\n",
      " 51050K .......... .......... .......... .......... .......... 77%  123M 11s\n",
      " 51100K .......... .......... .......... .......... .......... 77%  114M 11s\n",
      " 51150K .......... .......... .......... .......... .......... 77% 70.3M 11s\n",
      " 51200K .......... .......... .......... .......... .......... 78% 97.9M 11s\n",
      " 51250K .......... .......... .......... .......... .......... 78%  126M 11s\n",
      " 51300K .......... .......... .......... .......... .......... 78%  127M 10s\n",
      " 51350K .......... .......... .......... .......... .......... 78% 1.28M 10s\n",
      " 51400K .......... .......... .......... .......... .......... 78% 1.20M 10s\n",
      " 51450K .......... .......... .......... .......... .......... 78%  703K 10s\n",
      " 51500K .......... .......... .......... .......... .......... 78%  945K 10s\n",
      " 51550K .......... .......... .......... .......... .......... 78% 1.01M 10s\n",
      " 51600K .......... .......... .......... .......... .......... 78%  436K 10s\n",
      " 51650K .......... .......... .......... .......... .......... 78%  764K 10s\n",
      " 51700K .......... .......... .......... .......... .......... 78%  778K 10s\n",
      " 51750K .......... .......... .......... .......... .......... 78%  867K 10s\n",
      " 51800K .......... .......... .......... .......... .......... 78%  946K 10s\n",
      " 51850K .......... .......... .......... .......... .......... 79% 1.14M 10s\n",
      " 51900K .......... .......... .......... .......... .......... 79%  993K 10s\n",
      " 51950K .......... .......... .......... .......... .......... 79% 1.15M 10s\n",
      " 52000K .......... .......... .......... .......... .......... 79%  988K 10s\n",
      " 52050K .......... .......... .......... .......... .......... 79% 1.29M 10s\n",
      " 52100K .......... .......... .......... .......... .......... 79% 1.04M 10s\n",
      " 52150K .......... .......... .......... .......... .......... 79%  954K 10s\n",
      " 52200K .......... .......... .......... .......... .......... 79% 1.33M 10s\n",
      " 52250K .......... .......... .......... .......... .......... 79% 1.35M 10s\n",
      " 52300K .......... .......... .......... .......... .......... 79% 1.21M 10s\n",
      " 52350K .......... .......... .......... .......... .......... 79% 1.10M 10s\n",
      " 52400K .......... .......... .......... .......... .......... 79% 1007K 10s\n",
      " 52450K .......... .......... .......... .......... .......... 79%  710K 10s\n",
      " 52500K .......... .......... .......... .......... .......... 80% 1.15M 10s\n",
      " 52550K .......... .......... .......... .......... .......... 80% 1.35M 10s\n",
      " 52600K .......... .......... .......... .......... .......... 80%  961K 10s\n",
      " 52650K .......... .......... .......... .......... .......... 80%  980K 10s\n",
      " 52700K .......... .......... .......... .......... .......... 80% 1.12M 10s\n",
      " 52750K .......... .......... .......... .......... .......... 80% 1.37M 9s\n",
      " 52800K .......... .......... .......... .......... .......... 80% 1017K 9s\n",
      " 52850K .......... .......... .......... .......... .......... 80%  880K 9s\n",
      " 52900K .......... .......... .......... .......... .......... 80%  484K 9s\n",
      " 52950K .......... .......... .......... .......... .......... 80% 1.16M 9s\n",
      " 53000K .......... .......... .......... .......... .......... 80% 1.34M 9s\n",
      " 53050K .......... .......... .......... .......... .......... 80% 1.15M 9s\n",
      " 53100K .......... .......... .......... .......... .......... 80% 1.33M 9s\n",
      " 53150K .......... .......... .......... .......... .......... 81% 1.32M 9s\n",
      " 53200K .......... .......... .......... .......... .......... 81%  953K 9s\n",
      " 53250K .......... .......... .......... .......... .......... 81% 1.17M 9s\n",
      " 53300K .......... .......... .......... .......... .......... 81% 1.33M 9s\n",
      " 53350K .......... .......... .......... .......... .......... 81% 1.33M 9s\n",
      " 53400K .......... .......... .......... .......... .......... 81% 1.18M 9s\n",
      " 53450K .......... .......... .......... .......... .......... 81% 1023K 9s\n",
      " 53500K .......... .......... .......... .......... .......... 81% 1.07M 9s\n",
      " 53550K .......... .......... .......... .......... .......... 81%  937K 9s\n",
      " 53600K .......... .......... .......... .......... .......... 81%  822K 9s\n",
      " 53650K .......... .......... .......... .......... .......... 81%  913K 9s\n",
      " 53700K .......... .......... .......... .......... .......... 81% 1.51M 9s\n",
      " 53750K .......... .......... .......... .......... .......... 81%  559K 9s\n",
      " 53800K .......... .......... .......... .......... .......... 81%  829K 9s\n",
      " 53850K .......... .......... .......... .......... .......... 82% 1.71M 9s\n",
      " 53900K .......... .......... .......... .......... .......... 82% 2.49M 9s\n",
      " 53950K .......... .......... .......... .......... .......... 82% 1.18M 9s\n",
      " 54000K .......... .......... .......... .......... .......... 82% 1.20M 9s\n",
      " 54050K .......... .......... .......... .......... .......... 82% 1.13M 9s\n",
      " 54100K .......... .......... .......... .......... .......... 82%  115K 9s\n",
      " 54150K .......... .......... .......... .......... .......... 82% 54.0M 9s\n",
      " 54200K .......... .......... .......... .......... .......... 82% 93.8M 9s\n",
      " 54250K .......... .......... .......... .......... .......... 82%  113M 9s\n",
      " 54300K .......... .......... .......... .......... .......... 82% 88.3M 8s\n",
      " 54350K .......... .......... .......... .......... .......... 82%  121M 8s\n",
      " 54400K .......... .......... .......... .......... .......... 82% 42.9M 8s\n",
      " 54450K .......... .......... .......... .......... .......... 82%  117M 8s\n",
      " 54500K .......... .......... .......... .......... .......... 83% 1.88M 8s\n",
      " 54550K .......... .......... .......... .......... .......... 83% 1.23M 8s\n",
      " 54600K .......... .......... .......... .......... .......... 83% 1.33M 8s\n",
      " 54650K .......... .......... .......... .......... .......... 83% 1.32M 8s\n",
      " 54700K .......... .......... .......... .......... .......... 83%  951K 8s\n",
      " 54750K .......... .......... .......... .......... .......... 83% 1.38M 8s\n",
      " 54800K .......... .......... .......... .......... .......... 83% 1.02M 8s\n",
      " 54850K .......... .......... .......... .......... .......... 83% 1006K 8s\n",
      " 54900K .......... .......... .......... .......... .......... 83% 1.31M 8s\n",
      " 54950K .......... .......... .......... .......... .......... 83% 1.28M 8s\n",
      " 55000K .......... .......... .......... .......... .......... 83% 1.31M 8s\n",
      " 55050K .......... .......... .......... .......... .......... 83% 1.48M 8s\n",
      " 55100K .......... .......... .......... .......... .......... 83% 1.38M 8s\n",
      " 55150K .......... .......... .......... .......... .......... 84%  874K 8s\n",
      " 55200K .......... .......... .......... .......... .......... 84% 1.02M 8s\n",
      " 55250K .......... .......... .......... .......... .......... 84% 1.19M 8s\n",
      " 55300K .......... .......... .......... .......... .......... 84% 1.32M 8s\n",
      " 55350K .......... .......... .......... .......... .......... 84% 1.03M 8s\n",
      " 55400K .......... .......... .......... .......... .......... 84% 1.40M 8s\n",
      " 55450K .......... .......... .......... .......... .......... 84%  604K 8s\n",
      " 55500K .......... .......... .......... .......... .......... 84% 1.31M 8s\n",
      " 55550K .......... .......... .......... .......... .......... 84% 1.13M 8s\n",
      " 55600K .......... .......... .......... .......... .......... 84% 1.04M 7s\n",
      " 55650K .......... .......... .......... .......... .......... 84%  855K 7s\n",
      " 55700K .......... .......... .......... .......... .......... 84% 1.42M 7s\n",
      " 55750K .......... .......... .......... .......... .......... 84% 1.48M 7s\n",
      " 55800K .......... .......... .......... .......... .......... 85% 1.95M 7s\n",
      " 55850K .......... .......... .......... .......... .......... 85% 1.05M 7s\n",
      " 55900K .......... .......... .......... .......... .......... 85% 1.23M 7s\n",
      " 55950K .......... .......... .......... .......... .......... 85% 1.24M 7s\n",
      " 56000K .......... .......... .......... .......... .......... 85%  574K 7s\n",
      " 56050K .......... .......... .......... .......... .......... 85% 1.04M 7s\n",
      " 56100K .......... .......... .......... .......... .......... 85% 1.99M 7s\n",
      " 56150K .......... .......... .......... .......... .......... 85% 3.38M 7s\n",
      " 56200K .......... .......... .......... .......... .......... 85% 1021K 7s\n",
      " 56250K .......... .......... .......... .......... .......... 85%  430K 7s\n",
      " 56300K .......... .......... .......... .......... .......... 85% 1.46M 7s\n",
      " 56350K .......... .......... .......... .......... .......... 85% 2.73M 7s\n",
      " 56400K .......... .......... .......... .......... .......... 85%  622K 7s\n",
      " 56450K .......... .......... .......... .......... .......... 86% 1.66M 7s\n",
      " 56500K .......... .......... .......... .......... .......... 86% 2.32M 7s\n",
      " 56550K .......... .......... .......... .......... .......... 86% 1.62M 7s\n",
      " 56600K .......... .......... .......... .......... .......... 86% 2.33M 7s\n",
      " 56650K .......... .......... .......... .......... .......... 86%  715K 7s\n",
      " 56700K .......... .......... .......... .......... .......... 86%  581K 7s\n",
      " 56750K .......... .......... .......... .......... .......... 86% 2.19M 7s\n",
      " 56800K .......... .......... .......... .......... .......... 86% 1.77M 7s\n",
      " 56850K .......... .......... .......... .......... .......... 86% 2.37M 7s\n",
      " 56900K .......... .......... .......... .......... .......... 86%  785K 7s\n",
      " 56950K .......... .......... .......... .......... .......... 86%  835K 7s\n",
      " 57000K .......... .......... .......... .......... .......... 86% 1.28M 6s\n",
      " 57050K .......... .......... .......... .......... .......... 86% 3.11M 6s\n",
      " 57100K .......... .......... .......... .......... .......... 87% 2.63M 6s\n",
      " 57150K .......... .......... .......... .......... .......... 87% 1.63M 6s\n",
      " 57200K .......... .......... .......... .......... .......... 87%  617K 6s\n",
      " 57250K .......... .......... .......... .......... .......... 87% 3.46M 6s\n",
      " 57300K .......... .......... .......... .......... .......... 87% 2.12M 6s\n",
      " 57350K .......... .......... .......... .......... .......... 87% 1.50M 6s\n",
      " 57400K .......... .......... .......... .......... .......... 87% 1.99M 6s\n",
      " 57450K .......... .......... .......... .......... .......... 87%  957K 6s\n",
      " 57500K .......... .......... .......... .......... .......... 87% 2.22M 6s\n",
      " 57550K .......... .......... .......... .......... .......... 87% 2.42M 6s\n",
      " 57600K .......... .......... .......... .......... .......... 87%  745K 6s\n",
      " 57650K .......... .......... .......... .......... .......... 87% 3.15M 6s\n",
      " 57700K .......... .......... .......... .......... .......... 87% 2.48M 6s\n",
      " 57750K .......... .......... .......... .......... .......... 88% 1.49M 6s\n",
      " 57800K .......... .......... .......... .......... .......... 88%  802K 6s\n",
      " 57850K .......... .......... .......... .......... .......... 88% 2.02M 6s\n",
      " 57900K .......... .......... .......... .......... .......... 88% 2.12M 6s\n",
      " 57950K .......... .......... .......... .......... .......... 88% 2.93M 6s\n",
      " 58000K .......... .......... .......... .......... .......... 88% 2.10M 6s\n",
      " 58050K .......... .......... .......... .......... .......... 88%  880K 6s\n",
      " 58100K .......... .......... .......... .......... .......... 88% 2.76M 6s\n",
      " 58150K .......... .......... .......... .......... .......... 88%  541K 6s\n",
      " 58200K .......... .......... .......... .......... .......... 88% 2.68M 6s\n",
      " 58250K .......... .......... .......... .......... .......... 88% 1.43M 6s\n",
      " 58300K .......... .......... .......... .......... .......... 88% 2.12M 5s\n",
      " 58350K .......... .......... .......... .......... .......... 88% 2.35M 5s\n",
      " 58400K .......... .......... .......... .......... .......... 89%  747K 5s\n",
      " 58450K .......... .......... .......... .......... .......... 89%  673K 5s\n",
      " 58500K .......... .......... .......... .......... .......... 89% 2.07M 5s\n",
      " 58550K .......... .......... .......... .......... .......... 89% 2.93M 5s\n",
      " 58600K .......... .......... .......... .......... .......... 89% 1.45M 5s\n",
      " 58650K .......... .......... .......... .......... .......... 89% 1.77M 5s\n",
      " 58700K .......... .......... .......... .......... .......... 89% 1.83M 5s\n",
      " 58750K .......... .......... .......... .......... .......... 89%  311K 5s\n",
      " 58800K .......... .......... .......... .......... .......... 89% 1006K 5s\n",
      " 58850K .......... .......... .......... .......... .......... 89% 2.71M 5s\n",
      " 58900K .......... .......... .......... .......... .......... 89% 1.95M 5s\n",
      " 58950K .......... .......... .......... .......... .......... 89%  766K 5s\n",
      " 59000K .......... .......... .......... .......... .......... 89%  596K 5s\n",
      " 59050K .......... .......... .......... .......... .......... 89%  350K 5s\n",
      " 59100K .......... .......... .......... .......... .......... 90% 1.73M 5s\n",
      " 59150K .......... .......... .......... .......... .......... 90% 2.32M 5s\n",
      " 59200K .......... .......... .......... .......... .......... 90% 1.74M 5s\n",
      " 59250K .......... .......... .......... .......... .......... 90% 2.83M 5s\n",
      " 59300K .......... .......... .......... .......... .......... 90%  340K 5s\n",
      " 59350K .......... .......... .......... .......... .......... 90% 2.17M 5s\n",
      " 59400K .......... .......... .......... .......... .......... 90% 2.72M 5s\n",
      " 59450K .......... .......... .......... .......... .......... 90% 2.87M 5s\n",
      " 59500K .......... .......... .......... .......... .......... 90% 2.78M 5s\n",
      " 59550K .......... .......... .......... .......... .......... 90% 1.91M 5s\n",
      " 59600K .......... .......... .......... .......... .......... 90%  876K 5s\n",
      " 59650K .......... .......... .......... .......... .......... 90%  944K 5s\n",
      " 59700K .......... .......... .......... .......... .......... 90% 2.86M 4s\n",
      " 59750K .......... .......... .......... .......... .......... 91% 2.07M 4s\n",
      " 59800K .......... .......... .......... .......... .......... 91% 1.99M 4s\n",
      " 59850K .......... .......... .......... .......... .......... 91% 3.40M 4s\n",
      " 59900K .......... .......... .......... .......... .......... 91% 2.32M 4s\n",
      " 59950K .......... .......... .......... .......... .......... 91% 1.03M 4s\n",
      " 60000K .......... .......... .......... .......... .......... 91% 1.14M 4s\n",
      " 60050K .......... .......... .......... .......... .......... 91% 1.90M 4s\n",
      " 60100K .......... .......... .......... .......... .......... 91% 2.14M 4s\n",
      " 60150K .......... .......... .......... .......... .......... 91% 2.02M 4s\n",
      " 60200K .......... .......... .......... .......... .......... 91% 2.85M 4s\n",
      " 60250K .......... .......... .......... .......... .......... 91% 2.93M 4s\n",
      " 60300K .......... .......... .......... .......... .......... 91% 2.31M 4s\n",
      " 60350K .......... .......... .......... .......... .......... 91%  224K 4s\n",
      " 60400K .......... .......... .......... .......... .......... 92% 3.06M 4s\n",
      " 60450K .......... .......... .......... .......... .......... 92%  109M 4s\n",
      " 60500K .......... .......... .......... .......... .......... 92%  134M 4s\n",
      " 60550K .......... .......... .......... .......... .......... 92%  118M 4s\n",
      " 60600K .......... .......... .......... .......... .......... 92% 91.7M 4s\n",
      " 60650K .......... .......... .......... .......... .......... 92% 63.3M 4s\n",
      " 60700K .......... .......... .......... .......... .......... 92%  993K 4s\n",
      " 60750K .......... .......... .......... .......... .......... 92% 1.14M 4s\n",
      " 60800K .......... .......... .......... .......... .......... 92% 1007K 4s\n",
      " 60850K .......... .......... .......... .......... .......... 92%  669K 4s\n",
      " 60900K .......... .......... .......... .......... .......... 92% 1.78M 4s\n",
      " 60950K .......... .......... .......... .......... .......... 92% 1.21M 4s\n",
      " 61000K .......... .......... .......... .......... .......... 92%  695K 3s\n",
      " 61050K .......... .......... .......... .......... .......... 93% 1.37M 3s\n",
      " 61100K .......... .......... .......... .......... .......... 93% 2.17M 3s\n",
      " 61150K .......... .......... .......... .......... .......... 93% 1.48M 3s\n",
      " 61200K .......... .......... .......... .......... .......... 93%  532K 3s\n",
      " 61250K .......... .......... .......... .......... .......... 93% 3.12M 3s\n",
      " 61300K .......... .......... .......... .......... .......... 93% 1.97M 3s\n",
      " 61350K .......... .......... .......... .......... .......... 93%  180K 3s\n",
      " 61400K .......... .......... .......... .......... .......... 93% 56.7M 3s\n",
      " 61450K .......... .......... .......... .......... .......... 93% 72.9M 3s\n",
      " 61500K .......... .......... .......... .......... .......... 93% 1.94M 3s\n",
      " 61550K .......... .......... .......... .......... .......... 93% 1.90M 3s\n",
      " 61600K .......... .......... .......... .......... .......... 93%  273K 3s\n",
      " 61650K .......... .......... .......... .......... .......... 93%  639K 3s\n",
      " 61700K .......... .......... .......... .......... .......... 94%  501K 3s\n",
      " 61750K .......... .......... .......... .......... .......... 94% 1.17M 3s\n",
      " 61800K .......... .......... .......... .......... .......... 94%  475K 3s\n",
      " 61850K .......... .......... .......... .......... .......... 94%  653K 3s\n",
      " 61900K .......... .......... .......... .......... .......... 94%  718K 3s\n",
      " 61950K .......... .......... .......... .......... .......... 94%  475K 3s\n",
      " 62000K .......... .......... .......... .......... .......... 94%  669K 3s\n",
      " 62050K .......... .......... .......... .......... .......... 94%  587K 3s\n",
      " 62100K .......... .......... .......... .......... .......... 94%  934K 3s\n",
      " 62150K .......... .......... .......... .......... .......... 94%  563K 3s\n",
      " 62200K .......... .......... .......... .......... .......... 94%  748K 3s\n",
      " 62250K .......... .......... .......... .......... .......... 94%  683K 3s\n",
      " 62300K .......... .......... .......... .......... .......... 94%  482K 3s\n",
      " 62350K .......... .......... .......... .......... .......... 95%  911K 3s\n",
      " 62400K .......... .......... .......... .......... .......... 95%  464K 2s\n",
      " 62450K .......... .......... .......... .......... .......... 95%  617K 2s\n",
      " 62500K .......... .......... .......... .......... .......... 95%  744K 2s\n",
      " 62550K .......... .......... .......... .......... .......... 95% 2.46M 2s\n",
      " 62600K .......... .......... .......... .......... .......... 95%  561K 2s\n",
      " 62650K .......... .......... .......... .......... .......... 95%  908K 2s\n",
      " 62700K .......... .......... .......... .......... .......... 95%  704K 2s\n",
      " 62750K .......... .......... .......... .......... .......... 95%  866K 2s\n",
      " 62800K .......... .......... .......... .......... .......... 95%  678K 2s\n",
      " 62850K .......... .......... .......... .......... .......... 95%  879K 2s\n",
      " 62900K .......... .......... .......... .......... .......... 95%  736K 2s\n",
      " 62950K .......... .......... .......... .......... .......... 95%  817K 2s\n",
      " 63000K .......... .......... .......... .......... .......... 96% 1.97M 2s\n",
      " 63050K .......... .......... .......... .......... .......... 96%  813K 2s\n",
      " 63100K .......... .......... .......... .......... .......... 96%  889K 2s\n",
      " 63150K .......... .......... .......... .......... .......... 96% 1.87M 2s\n",
      " 63200K .......... .......... .......... .......... .......... 96%  573K 2s\n",
      " 63250K .......... .......... .......... .......... .......... 96% 1.46M 2s\n",
      " 63300K .......... .......... .......... .......... .......... 96%  644K 2s\n",
      " 63350K .......... .......... .......... .......... .......... 96% 1.47M 2s\n",
      " 63400K .......... .......... .......... .......... .......... 96%  876K 2s\n",
      " 63450K .......... .......... .......... .......... .......... 96%  666K 2s\n",
      " 63500K .......... .......... .......... .......... .......... 96% 2.81M 2s\n",
      " 63550K .......... .......... .......... .......... .......... 96% 1.07M 2s\n",
      " 63600K .......... .......... .......... .......... .......... 96%  547K 2s\n",
      " 63650K .......... .......... .......... .......... .......... 96% 1.92M 2s\n",
      " 63700K .......... .......... .......... .......... .......... 97% 1.20M 1s\n",
      " 63750K .......... .......... .......... .......... .......... 97%  574K 1s\n",
      " 63800K .......... .......... .......... .......... .......... 97% 2.86M 1s\n",
      " 63850K .......... .......... .......... .......... .......... 97% 1.83M 1s\n",
      " 63900K .......... .......... .......... .......... .......... 97% 1.94M 1s\n",
      " 63950K .......... .......... .......... .......... .......... 97%  550K 1s\n",
      " 64000K .......... .......... .......... .......... .......... 97% 2.40M 1s\n",
      " 64050K .......... .......... .......... .......... .......... 97% 1.71M 1s\n",
      " 64100K .......... .......... .......... .......... .......... 97%  396K 1s\n",
      " 64150K .......... .......... .......... .......... .......... 97%  741K 1s\n",
      " 64200K .......... .......... .......... .......... .......... 97% 1.01M 1s\n",
      " 64250K .......... .......... .......... .......... .......... 97% 1.89M 1s\n",
      " 64300K .......... .......... .......... .......... .......... 97%  561K 1s\n",
      " 64350K .......... .......... .......... .......... .......... 98% 2.93M 1s\n",
      " 64400K .......... .......... .......... .......... .......... 98%  974K 1s\n",
      " 64450K .......... .......... .......... .......... .......... 98% 3.17M 1s\n",
      " 64500K .......... .......... .......... .......... .......... 98%  455K 1s\n",
      " 64550K .......... .......... .......... .......... .......... 98% 8.24M 1s\n",
      " 64600K .......... .......... .......... .......... .......... 98% 1.96M 1s\n",
      " 64650K .......... .......... .......... .......... .......... 98%  421K 1s\n",
      " 64700K .......... .......... .......... .......... .......... 98% 3.02M 1s\n",
      " 64750K .......... .......... .......... .......... .......... 98% 2.92M 1s\n",
      " 64800K .......... .......... .......... .......... .......... 98% 1.66M 1s\n",
      " 64850K .......... .......... .......... .......... .......... 98%  552K 1s\n",
      " 64900K .......... .......... .......... .......... .......... 98% 2.63M 1s\n",
      " 64950K .......... .......... .......... .......... .......... 98% 2.26M 1s\n",
      " 65000K .......... .......... .......... .......... .......... 99% 2.14M 0s\n",
      " 65050K .......... .......... .......... .......... .......... 99%  609K 0s\n",
      " 65100K .......... .......... .......... .......... .......... 99% 3.24M 0s\n",
      " 65150K .......... .......... .......... .......... .......... 99% 2.25M 0s\n",
      " 65200K .......... .......... .......... .......... .......... 99% 2.00M 0s\n",
      " 65250K .......... .......... .......... .......... .......... 99%  196K 0s\n",
      " 65300K .......... .......... .......... .......... .......... 99% 30.9M 0s\n",
      " 65350K .......... .......... .......... .......... .......... 99% 28.3M 0s\n",
      " 65400K .......... .......... .......... .......... .......... 99% 21.9M 0s\n",
      " 65450K .......... .......... .......... .......... .......... 99% 15.9M 0s\n",
      " 65500K .......... .......... .......... .......... .......... 99% 2.70M 0s\n",
      " 65550K .......... .......... .......... .......... .......... 99%  431K 0s\n",
      " 65600K .......... .......... .......... .......... .......... 99% 2.12M 0s\n",
      " 65650K .......... .......... ..                              100% 23.1M=51s\n",
      "\n",
      "2021-03-18 14:12:36 (1.25 MB/s) - â€�assembly_summary_refseq.txt.3â€™ saved [67248987/67248987]\n",
      "\n"
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "\n",
    "wget https://ftp.ncbi.nlm.nih.gov/genomes/refseq/assembly_summary_refseq.txt\n",
    "\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 150,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "wget: invalid option -- 'f'\n",
      "Usage: wget [OPTION]... [URL]...\n",
      "\n",
      "Try `wget --help' for more options.\n"
     ]
    },
    {
     "ename": "CalledProcessError",
     "evalue": "Command 'b'\\ngrep GCF_000910615.1 assembly_summary_refseq.txt \\\\\\n    | awk \\'BEGIN{FS=\"\\\\t\"}{if($12==\"Complete Genome\"){print $20}}\\' \\\\\\n    | awk \\'BEGIN{OFS=FS=\"/\"}{print $0,$NF\"_genomic.fna.gz\"}\\' \\\\\\n    > urls.txt\\n\\n#Now you can go through your urls file\\n\\nIFS=$\\'\\\\n\\'; for NEXT in $(cat urls.txt); do wget \"$NEXT\" iconv -f ISO-8859-1 -t UTF-8; done\\n'' returned non-zero exit status 2.",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mCalledProcessError\u001b[0m                        Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-150-6013122ac2ff>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mget_ipython\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mrun_cell_magic\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'bash'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m''\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'\\ngrep GCF_000910615.1 assembly_summary_refseq.txt \\\\\\n    | awk \\'BEGIN{FS=\"\\\\t\"}{if($12==\"Complete Genome\"){print $20}}\\' \\\\\\n    | awk \\'BEGIN{OFS=FS=\"/\"}{print $0,$NF\"_genomic.fna.gz\"}\\' \\\\\\n    > urls.txt\\n\\n#Now you can go through your urls file\\n\\nIFS=$\\'\\\\n\\'; for NEXT in $(cat urls.txt); do wget \"$NEXT\" iconv -f ISO-8859-1 -t UTF-8; done\\n'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32mD:\\anaconndaa\\lib\\site-packages\\IPython\\core\\interactiveshell.py\u001b[0m in \u001b[0;36mrun_cell_magic\u001b[1;34m(self, magic_name, line, cell)\u001b[0m\n\u001b[0;32m   2369\u001b[0m             \u001b[1;32mwith\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mbuiltin_trap\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2370\u001b[0m                 \u001b[0margs\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mmagic_arg_s\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 2371\u001b[1;33m                 \u001b[0mresult\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mfn\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0margs\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mkwargs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   2372\u001b[0m             \u001b[1;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   2373\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mD:\\anaconndaa\\lib\\site-packages\\IPython\\core\\magics\\script.py\u001b[0m in \u001b[0;36mnamed_script_magic\u001b[1;34m(line, cell)\u001b[0m\n\u001b[0;32m    140\u001b[0m             \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    141\u001b[0m                 \u001b[0mline\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mscript\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 142\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mshebang\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mline\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    143\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    144\u001b[0m         \u001b[1;31m# write a basic docstring:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32m<decorator-gen-111>\u001b[0m in \u001b[0;36mshebang\u001b[1;34m(self, line, cell)\u001b[0m\n",
      "\u001b[1;32mD:\\anaconndaa\\lib\\site-packages\\IPython\\core\\magic.py\u001b[0m in \u001b[0;36m<lambda>\u001b[1;34m(f, *a, **k)\u001b[0m\n\u001b[0;32m    185\u001b[0m     \u001b[1;31m# but it's overkill for just that one bit of state.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    186\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0mmagic_deco\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0marg\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 187\u001b[1;33m         \u001b[0mcall\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mlambda\u001b[0m \u001b[0mf\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m*\u001b[0m\u001b[0ma\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mk\u001b[0m\u001b[1;33m:\u001b[0m \u001b[0mf\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0ma\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m**\u001b[0m\u001b[0mk\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    188\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    189\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mcallable\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0marg\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mD:\\anaconndaa\\lib\\site-packages\\IPython\\core\\magics\\script.py\u001b[0m in \u001b[0;36mshebang\u001b[1;34m(self, line, cell)\u001b[0m\n\u001b[0;32m    243\u001b[0m             \u001b[0msys\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mstderr\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mflush\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    244\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0margs\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mraise_error\u001b[0m \u001b[1;32mand\u001b[0m \u001b[0mp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[1;33m!=\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 245\u001b[1;33m             \u001b[1;32mraise\u001b[0m \u001b[0mCalledProcessError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mp\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mreturncode\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0moutput\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mout\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mstderr\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0merr\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    246\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    247\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_run_script\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mp\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcell\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mto_close\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mCalledProcessError\u001b[0m: Command 'b'\\ngrep GCF_000910615.1 assembly_summary_refseq.txt \\\\\\n    | awk \\'BEGIN{FS=\"\\\\t\"}{if($12==\"Complete Genome\"){print $20}}\\' \\\\\\n    | awk \\'BEGIN{OFS=FS=\"/\"}{print $0,$NF\"_genomic.fna.gz\"}\\' \\\\\\n    > urls.txt\\n\\n#Now you can go through your urls file\\n\\nIFS=$\\'\\\\n\\'; for NEXT in $(cat urls.txt); do wget \"$NEXT\" iconv -f ISO-8859-1 -t UTF-8; done\\n'' returned non-zero exit status 2."
     ]
    }
   ],
   "source": [
    "%%bash\n",
    "\n",
    "grep GCF_000910615.1 assembly_summary_refseq.txt \\\n",
    "    | awk 'BEGIN{FS=\"\\t\"}{if($12==\"Complete Genome\"){print $20}}' \\\n",
    "    | awk 'BEGIN{OFS=FS=\"/\"}{print $0,$NF\"_genomic.fna.gz\"}' \\\n",
    "    > urls.txt\n",
    "\n",
    "#Now you can go through your urls file\n",
    "\n",
    "IFS=$'\\n'; for NEXT in $(cat urls.txt); do wget \"$NEXT\" iconv -f ISO-8859-1 -t UTF-8 ; done\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 144,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "D:\\anaconndaa\\lib\\site-packages\\IPython\\core\\interactiveshell.py:3071: DtypeWarning: Columns (5,6) have mixed types.Specify dtype option on import or set low_memory=False.\n",
      "  has_raised = await self.run_ast_nodes(code_ast.body, cell_name,\n"
     ]
    }
   ],
   "source": [
    "import codecs\n",
    "\n",
    "doc = codecs.open('assembly_summary_refseq.txt','rU','UTF-8') #open for reading with \"universal\" type set\n",
    "\n",
    "df = pd.read_csv(doc, sep='\\t')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 145,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>#   See ftp://ftp.ncbi.nlm.nih.gov/genomes/README_assembly_summary.txt for a description of the columns in this file.</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th># assembly_accession</th>\n",
       "      <th>bioproject</th>\n",
       "      <th>biosample</th>\n",
       "      <th>wgs_master</th>\n",
       "      <th>refseq_category</th>\n",
       "      <th>taxid</th>\n",
       "      <th>species_taxid</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>infraspecific_name</th>\n",
       "      <th>isolate</th>\n",
       "      <th>version_status</th>\n",
       "      <th>assembly_level</th>\n",
       "      <th>release_type</th>\n",
       "      <th>genome_rep</th>\n",
       "      <th>seq_rel_date</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>submitter</th>\n",
       "      <th>gbrs_paired_asm</th>\n",
       "      <th>paired_asm_comp</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>excluded_from_refseq</th>\n",
       "      <td>relation_to_type_material</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000001215.4</th>\n",
       "      <th>PRJNA164</th>\n",
       "      <th>SAMN02803731</th>\n",
       "      <th>NaN</th>\n",
       "      <th>reference genome</th>\n",
       "      <th>7227</th>\n",
       "      <th>7227</th>\n",
       "      <th>Drosophila melanogaster</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>latest</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Major</th>\n",
       "      <th>Full</th>\n",
       "      <th>2014/08/01</th>\n",
       "      <th>Release 6 plus ISO1 MT</th>\n",
       "      <th>The FlyBase Consortium/Berkeley Drosophila Genome Project/Celera Genomics</th>\n",
       "      <th>GCA_000001215.4</th>\n",
       "      <th>identical</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/215/GCF_000001215.4_Release_6_plus_ISO1_MT</th>\n",
       "      <th>NaN</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000001405.39</th>\n",
       "      <th>PRJNA168</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>reference genome</th>\n",
       "      <th>9606</th>\n",
       "      <th>9606</th>\n",
       "      <th>Homo sapiens</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>latest</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Patch</th>\n",
       "      <th>Full</th>\n",
       "      <th>2019/02/28</th>\n",
       "      <th>GRCh38.p13</th>\n",
       "      <th>Genome Reference Consortium</th>\n",
       "      <th>GCA_000001405.28</th>\n",
       "      <th>different</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.39_GRCh38.p13</th>\n",
       "      <th>NaN</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000001635.27</th>\n",
       "      <th>PRJNA169</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>reference genome</th>\n",
       "      <th>10090</th>\n",
       "      <th>10090</th>\n",
       "      <th>Mus musculus</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>latest</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Major</th>\n",
       "      <th>Full</th>\n",
       "      <th>2020/06/24</th>\n",
       "      <th>GRCm39</th>\n",
       "      <th>Genome Reference Consortium</th>\n",
       "      <th>GCA_000001635.9</th>\n",
       "      <th>identical</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/635/GCF_000001635.27_GRCm39</th>\n",
       "      <th>NaN</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000001735.4</th>\n",
       "      <th>PRJNA116</th>\n",
       "      <th>SAMN03081427</th>\n",
       "      <th>NaN</th>\n",
       "      <th>reference genome</th>\n",
       "      <th>3702</th>\n",
       "      <th>3702</th>\n",
       "      <th>Arabidopsis thaliana</th>\n",
       "      <th>ecotype=Columbia</th>\n",
       "      <th>NaN</th>\n",
       "      <th>latest</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Minor</th>\n",
       "      <th>Full</th>\n",
       "      <th>2018/03/15</th>\n",
       "      <th>TAIR10.1</th>\n",
       "      <th>The Arabidopsis Information Resource (TAIR)</th>\n",
       "      <th>GCA_000001735.2</th>\n",
       "      <th>identical</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/735/GCF_000001735.4_TAIR10.1</th>\n",
       "      <th>NaN</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_905236995.1</th>\n",
       "      <th>PRJNA224116</th>\n",
       "      <th>SAMEA7833113</th>\n",
       "      <th>CAJNNO000000000.1</th>\n",
       "      <th>na</th>\n",
       "      <th>150055</th>\n",
       "      <th>150055</th>\n",
       "      <th>Streptococcus lutetiensis</th>\n",
       "      <th>NaN</th>\n",
       "      <th>BR_UFV01</th>\n",
       "      <th>latest</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Major</th>\n",
       "      <th>Full</th>\n",
       "      <th>2021/02/20</th>\n",
       "      <th>ASM90523699v1</th>\n",
       "      <th>Universidade Federal de Vicosa</th>\n",
       "      <th>GCA_905236995.1</th>\n",
       "      <th>identical</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/236/995/GCF_905236995.1_ASM90523699v1</th>\n",
       "      <th>NaN</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_905237015.1</th>\n",
       "      <th>PRJNA224116</th>\n",
       "      <th>SAMEA7833114</th>\n",
       "      <th>CAJNNP000000000.1</th>\n",
       "      <th>na</th>\n",
       "      <th>150055</th>\n",
       "      <th>150055</th>\n",
       "      <th>Streptococcus lutetiensis</th>\n",
       "      <th>NaN</th>\n",
       "      <th>BR_UFV02</th>\n",
       "      <th>latest</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Major</th>\n",
       "      <th>Full</th>\n",
       "      <th>2021/02/20</th>\n",
       "      <th>ASM90523701v1</th>\n",
       "      <th>Universidade Federal de Vicosa</th>\n",
       "      <th>GCA_905237015.1</th>\n",
       "      <th>identical</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/237/015/GCF_905237015.1_ASM90523701v1</th>\n",
       "      <th>NaN</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_905237025.1</th>\n",
       "      <th>PRJNA224116</th>\n",
       "      <th>SAMEA7833116</th>\n",
       "      <th>CAJNNS000000000.1</th>\n",
       "      <th>na</th>\n",
       "      <th>150055</th>\n",
       "      <th>150055</th>\n",
       "      <th>Streptococcus lutetiensis</th>\n",
       "      <th>NaN</th>\n",
       "      <th>BR_UFV04</th>\n",
       "      <th>latest</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Major</th>\n",
       "      <th>Full</th>\n",
       "      <th>2021/02/20</th>\n",
       "      <th>ASM90523702v1</th>\n",
       "      <th>Universidade Federal de Vicosa</th>\n",
       "      <th>GCA_905237025.1</th>\n",
       "      <th>identical</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/237/025/GCF_905237025.1_ASM90523702v1</th>\n",
       "      <th>NaN</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_905237035.1</th>\n",
       "      <th>PRJNA224116</th>\n",
       "      <th>SAMEA7833115</th>\n",
       "      <th>CAJNNR000000000.1</th>\n",
       "      <th>na</th>\n",
       "      <th>150055</th>\n",
       "      <th>150055</th>\n",
       "      <th>Streptococcus lutetiensis</th>\n",
       "      <th>NaN</th>\n",
       "      <th>BR_UFV03</th>\n",
       "      <th>latest</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Major</th>\n",
       "      <th>Full</th>\n",
       "      <th>2021/02/20</th>\n",
       "      <th>ASM90523703v1</th>\n",
       "      <th>Universidade Federal de Vicosa</th>\n",
       "      <th>GCA_905237035.1</th>\n",
       "      <th>identical</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/237/035/GCF_905237035.1_ASM90523703v1</th>\n",
       "      <th>NaN</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_905237045.1</th>\n",
       "      <th>PRJNA224116</th>\n",
       "      <th>SAMEA7833117</th>\n",
       "      <th>CAJNNQ000000000.1</th>\n",
       "      <th>na</th>\n",
       "      <th>150055</th>\n",
       "      <th>150055</th>\n",
       "      <th>Streptococcus lutetiensis</th>\n",
       "      <th>NaN</th>\n",
       "      <th>BR_UFV05</th>\n",
       "      <th>latest</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Major</th>\n",
       "      <th>Full</th>\n",
       "      <th>2021/02/20</th>\n",
       "      <th>ASM90523704v1</th>\n",
       "      <th>Universidade Federal de Vicosa</th>\n",
       "      <th>GCA_905237045.1</th>\n",
       "      <th>identical</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/237/045/GCF_905237045.1_ASM90523704v1</th>\n",
       "      <th>NaN</th>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>222706 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                                                                                                                                                                                                                                                                  #   See ftp://ftp.ncbi.nlm.nih.gov/genomes/README_assembly_summary.txt for a description of the columns in this file.\n",
       "# assembly_accession bioproject  biosample    wgs_master        refseq_category  taxid  species_taxid organism_name             infraspecific_name isolate  version_status assembly_level release_type genome_rep seq_rel_date asm_name               submitter                                          gbrs_paired_asm  paired_asm_comp ftp_path                                           excluded_from_refseq                          relation_to_type_material                                                                   \n",
       "GCF_000001215.4      PRJNA164    SAMN02803731 NaN               reference genome 7227   7227          Drosophila melanogaster   NaN                NaN      latest         Chromosome     Major        Full       2014/08/01   Release 6 plus ISO1 MT The FlyBase Consortium/Berkeley Drosophila Geno... GCA_000001215.4  identical       ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/... NaN                                                                 NaN                                                                   \n",
       "GCF_000001405.39     PRJNA168    NaN          NaN               reference genome 9606   9606          Homo sapiens              NaN                NaN      latest         Chromosome     Patch        Full       2019/02/28   GRCh38.p13             Genome Reference Consortium                        GCA_000001405.28 different       ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/... NaN                                                                 NaN                                                                   \n",
       "GCF_000001635.27     PRJNA169    NaN          NaN               reference genome 10090  10090         Mus musculus              NaN                NaN      latest         Chromosome     Major        Full       2020/06/24   GRCm39                 Genome Reference Consortium                        GCA_000001635.9  identical       ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/... NaN                                                                 NaN                                                                   \n",
       "GCF_000001735.4      PRJNA116    SAMN03081427 NaN               reference genome 3702   3702          Arabidopsis thaliana      ecotype=Columbia   NaN      latest         Chromosome     Minor        Full       2018/03/15   TAIR10.1               The Arabidopsis Information Resource (TAIR)        GCA_000001735.2  identical       ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/... NaN                                                                 NaN                                                                   \n",
       "...                                                                                                                                                                                                                                                                                                                                                                                                                                                              ...                                                                   \n",
       "GCF_905236995.1      PRJNA224116 SAMEA7833113 CAJNNO000000000.1 na               150055 150055        Streptococcus lutetiensis NaN                BR_UFV01 latest         Contig         Major        Full       2021/02/20   ASM90523699v1          Universidade Federal de Vicosa                     GCA_905236995.1  identical       ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/... NaN                                                                 NaN                                                                   \n",
       "GCF_905237015.1      PRJNA224116 SAMEA7833114 CAJNNP000000000.1 na               150055 150055        Streptococcus lutetiensis NaN                BR_UFV02 latest         Contig         Major        Full       2021/02/20   ASM90523701v1          Universidade Federal de Vicosa                     GCA_905237015.1  identical       ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/... NaN                                                                 NaN                                                                   \n",
       "GCF_905237025.1      PRJNA224116 SAMEA7833116 CAJNNS000000000.1 na               150055 150055        Streptococcus lutetiensis NaN                BR_UFV04 latest         Contig         Major        Full       2021/02/20   ASM90523702v1          Universidade Federal de Vicosa                     GCA_905237025.1  identical       ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/... NaN                                                                 NaN                                                                   \n",
       "GCF_905237035.1      PRJNA224116 SAMEA7833115 CAJNNR000000000.1 na               150055 150055        Streptococcus lutetiensis NaN                BR_UFV03 latest         Contig         Major        Full       2021/02/20   ASM90523703v1          Universidade Federal de Vicosa                     GCA_905237035.1  identical       ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/... NaN                                                                 NaN                                                                   \n",
       "GCF_905237045.1      PRJNA224116 SAMEA7833117 CAJNNQ000000000.1 na               150055 150055        Streptococcus lutetiensis NaN                BR_UFV05 latest         Contig         Major        Full       2021/02/20   ASM90523704v1          Universidade Federal de Vicosa                     GCA_905237045.1  identical       ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/... NaN                                                                 NaN                                                                   \n",
       "\n",
       "[222706 rows x 1 columns]"
      ]
     },
     "execution_count": 145,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "# assembly_accession\t"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 185,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_assemblies(asb, download=True):\n",
    "    from Bio import Entrez\n",
    "    Entrez.email = 'amandakoryl@gmail.com'\n",
    "    for id in asb:\n",
    "        handle = Entrez.esearch(db=\"assembly\", term=id)\n",
    "        record = Entrez.read(handle)\n",
    "    for i in record['IdList']:\n",
    "    # Get Assembly Summary\n",
    "        esummary_handle = Entrez.esummary(db=\"assembly\", id=i, report=\"full\")\n",
    "        esummary_record = Entrez.read(esummary_handle)\n",
    "    links = []\n",
    "    for id in asb:\n",
    "\n",
    "        handle1 = Entrez.efetch(db=\"nucleotide\", id=id, retmode='text')\n",
    "        record = Entrez.read(handle1)\n",
    "        url = esummary_record['DocumentSummarySet']['DocumentSummary'][0]['FtpPath_RefSeq']\n",
    "        if url == '':\n",
    "            print(\"ups\")\n",
    "            break\n",
    "        label = os.path.basename(url)\n",
    "        print(label)\n",
    "        link = os.path.join(url,label+'_genomic.fna.gz')\n",
    "        print(link)\n",
    "        links.append(link)\n",
    "        if download == True:\n",
    "        #download link\n",
    "            urllib.request.urlretrieve(link, f'{label}.fna.gz')\n",
    "        return links\n",
    "\n",
    "\n",
    "\n",
    "\n",
    "    \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 186,
   "metadata": {},
   "outputs": [
    {
     "ename": "HTTPError",
     "evalue": "HTTP Error 400: Bad Request",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mHTTPError\u001b[0m                                 Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-186-b52b324ee497>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mget_assemblies\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0masb\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdownload\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mTrue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m<ipython-input-185-9286f5b4318f>\u001b[0m in \u001b[0;36mget_assemblies\u001b[1;34m(asb, download)\u001b[0m\n\u001b[0;32m     12\u001b[0m     \u001b[1;32mfor\u001b[0m \u001b[0mid\u001b[0m \u001b[1;32min\u001b[0m \u001b[0masb\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     13\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 14\u001b[1;33m         \u001b[0mhandle1\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mEntrez\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mefetch\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mdb\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m\"nucleotide\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mid\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mid\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mretmode\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m'text'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     15\u001b[0m         \u001b[0mrecord\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mEntrez\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhandle1\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     16\u001b[0m         \u001b[0murl\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mesummary_record\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'DocumentSummarySet'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'DocumentSummary'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;36m0\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m[\u001b[0m\u001b[1;34m'FtpPath_RefSeq'\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mD:\\anaconndaa\\lib\\site-packages\\Bio\\Entrez\\__init__.py\u001b[0m in \u001b[0;36mefetch\u001b[1;34m(db, **keywords)\u001b[0m\n\u001b[0;32m    205\u001b[0m             \u001b[1;31m# more than about 200 IDs\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    206\u001b[0m             \u001b[0mpost\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;32mTrue\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 207\u001b[1;33m     \u001b[1;32mreturn\u001b[0m \u001b[0m_open\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcgi\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mvariables\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mpost\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0mpost\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    208\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    209\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mD:\\anaconndaa\\lib\\site-packages\\Bio\\Entrez\\__init__.py\u001b[0m in \u001b[0;36m_open\u001b[1;34m(cgi, params, post, ecitmatch)\u001b[0m\n\u001b[0;32m    604\u001b[0m                 \u001b[0mhandle\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0murlopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcgi\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[1;33m=\u001b[0m\u001b[0moptions\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mencode\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m\"utf8\"\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    605\u001b[0m             \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 606\u001b[1;33m                 \u001b[0mhandle\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0murlopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mcgi\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    607\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[0mHTTPError\u001b[0m \u001b[1;32mas\u001b[0m \u001b[0mexception\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    608\u001b[0m             \u001b[1;31m# Reraise if the final try fails\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mD:\\anaconndaa\\lib\\urllib\\request.py\u001b[0m in \u001b[0;36murlopen\u001b[1;34m(url, data, timeout, cafile, capath, cadefault, context)\u001b[0m\n\u001b[0;32m    220\u001b[0m     \u001b[1;32melse\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    221\u001b[0m         \u001b[0mopener\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0m_opener\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 222\u001b[1;33m     \u001b[1;32mreturn\u001b[0m \u001b[0mopener\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0murl\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    223\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    224\u001b[0m \u001b[1;32mdef\u001b[0m \u001b[0minstall_opener\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mopener\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mD:\\anaconndaa\\lib\\urllib\\request.py\u001b[0m in \u001b[0;36mopen\u001b[1;34m(self, fullurl, data, timeout)\u001b[0m\n\u001b[0;32m    529\u001b[0m         \u001b[1;32mfor\u001b[0m \u001b[0mprocessor\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mprocess_response\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mget\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mprotocol\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    530\u001b[0m             \u001b[0mmeth\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mprocessor\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmeth_name\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 531\u001b[1;33m             \u001b[0mresponse\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mmeth\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mreq\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mresponse\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    532\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    533\u001b[0m         \u001b[1;32mreturn\u001b[0m \u001b[0mresponse\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mD:\\anaconndaa\\lib\\urllib\\request.py\u001b[0m in \u001b[0;36mhttp_response\u001b[1;34m(self, request, response)\u001b[0m\n\u001b[0;32m    638\u001b[0m         \u001b[1;31m# request was successfully received, understood, and accepted.\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    639\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;33m(\u001b[0m\u001b[1;36m200\u001b[0m \u001b[1;33m<=\u001b[0m \u001b[0mcode\u001b[0m \u001b[1;33m<\u001b[0m \u001b[1;36m300\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 640\u001b[1;33m             response = self.parent.error(\n\u001b[0m\u001b[0;32m    641\u001b[0m                 'http', request, response, code, msg, hdrs)\n\u001b[0;32m    642\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mD:\\anaconndaa\\lib\\urllib\\request.py\u001b[0m in \u001b[0;36merror\u001b[1;34m(self, proto, *args)\u001b[0m\n\u001b[0;32m    567\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mhttp_err\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    568\u001b[0m             \u001b[0margs\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m(\u001b[0m\u001b[0mdict\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'default'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m'http_error_default'\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m+\u001b[0m \u001b[0morig_args\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 569\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_call_chain\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0margs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    570\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    571\u001b[0m \u001b[1;31m# XXX probably also want an abstract factory that knows when it makes\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mD:\\anaconndaa\\lib\\urllib\\request.py\u001b[0m in \u001b[0;36m_call_chain\u001b[1;34m(self, chain, kind, meth_name, *args)\u001b[0m\n\u001b[0;32m    500\u001b[0m         \u001b[1;32mfor\u001b[0m \u001b[0mhandler\u001b[0m \u001b[1;32min\u001b[0m \u001b[0mhandlers\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    501\u001b[0m             \u001b[0mfunc\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mhandler\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmeth_name\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 502\u001b[1;33m             \u001b[0mresult\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mfunc\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m*\u001b[0m\u001b[0margs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    503\u001b[0m             \u001b[1;32mif\u001b[0m \u001b[0mresult\u001b[0m \u001b[1;32mis\u001b[0m \u001b[1;32mnot\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    504\u001b[0m                 \u001b[1;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mD:\\anaconndaa\\lib\\urllib\\request.py\u001b[0m in \u001b[0;36mhttp_error_default\u001b[1;34m(self, req, fp, code, msg, hdrs)\u001b[0m\n\u001b[0;32m    647\u001b[0m \u001b[1;32mclass\u001b[0m \u001b[0mHTTPDefaultErrorHandler\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mBaseHandler\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    648\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0mhttp_error_default\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mreq\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfp\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcode\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmsg\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mhdrs\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 649\u001b[1;33m         \u001b[1;32mraise\u001b[0m \u001b[0mHTTPError\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mreq\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mfull_url\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mcode\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mmsg\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mhdrs\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mfp\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    650\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    651\u001b[0m \u001b[1;32mclass\u001b[0m \u001b[0mHTTPRedirectHandler\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mBaseHandler\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mHTTPError\u001b[0m: HTTP Error 400: Bad Request"
     ]
    }
   ],
   "source": [
    "get_assemblies(asb, download=True)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 273,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import re\n",
    "first = []\n",
    "with open('assembly_summary_refseq.txt', \"r\", encoding = \"utf-8\") as f:\n",
    "    lines = f.readlines()[1:]\\\n",
    "\n",
    "    for line in lines:\n",
    "        line = str(line)\n",
    "        re.sub(r'(\\t)+', '\\t', line)\n",
    "        line.replace(\"#\", \"\")\n",
    "        first.append(line)\n",
    "\n",
    "        \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 274,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "['# assembly_accession\\tbioproject\\tbiosample\\twgs_master\\trefseq_category\\ttaxid\\tspecies_taxid\\torganism_name\\tinfraspecific_name\\tisolate\\tversion_status\\tassembly_level\\trelease_type\\tgenome_rep\\tseq_rel_date\\tasm_name\\tsubmitter\\tgbrs_paired_asm\\tpaired_asm_comp\\tftp_path\\texcluded_from_refseq\\trelation_to_type_material\\n', 'GCF_000001215.4\\tPRJNA164\\tSAMN02803731\\t\\treference genome\\t7227\\t7227\\tDrosophila melanogaster\\t\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2014/08/01\\tRelease 6 plus ISO1 MT\\tThe FlyBase Consortium/Berkeley Drosophila Genome Project/Celera Genomics\\tGCA_000001215.4\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/215/GCF_000001215.4_Release_6_plus_ISO1_MT\\t\\t\\n', 'GCF_000001405.39\\tPRJNA168\\t\\t\\treference genome\\t9606\\t9606\\tHomo sapiens\\t\\t\\tlatest\\tChromosome\\tPatch\\tFull\\t2019/02/28\\tGRCh38.p13\\tGenome Reference Consortium\\tGCA_000001405.28\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.39_GRCh38.p13\\t\\t\\n', 'GCF_000001635.27\\tPRJNA169\\t\\t\\treference genome\\t10090\\t10090\\tMus musculus\\t\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2020/06/24\\tGRCm39\\tGenome Reference Consortium\\tGCA_000001635.9\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/635/GCF_000001635.27_GRCm39\\t\\t\\n', 'GCF_000001735.4\\tPRJNA116\\tSAMN03081427\\t\\treference genome\\t3702\\t3702\\tArabidopsis thaliana\\tecotype=Columbia\\t\\tlatest\\tChromosome\\tMinor\\tFull\\t2018/03/15\\tTAIR10.1\\tThe Arabidopsis Information Resource (TAIR)\\tGCA_000001735.2\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/735/GCF_000001735.4_TAIR10.1\\t\\t\\n', 'GCF_000001905.1\\tPRJNA70973\\tSAMN02953622\\tAAGU00000000.3\\trepresentative genome\\t9785\\t9785\\tLoxodonta africana\\t\\tISIS603380\\tlatest\\tScaffold\\tMajor\\tFull\\t2009/07/15\\tLoxafr3.0\\tBroad Institute\\tGCA_000001905.1\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/905/GCF_000001905.1_Loxafr3.0\\t\\t\\n', 'GCF_000001985.1\\tPRJNA32665\\tSAMN02953685\\tABAR00000000.1\\trepresentative genome\\t441960\\t37727\\tTalaromyces marneffei ATCC 18224\\tstrain=ATCC 18224\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2008/10/29\\tJCVI-PMFA1-2.0\\tJ. Craig Venter Institute\\tGCA_000001985.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/985/GCF_000001985.1_JCVI-PMFA1-2.0\\t\\tassembly from type material\\n', 'GCF_000002035.6\\tPRJNA13922\\tSAMN06930106\\t\\treference genome\\t7955\\t7955\\tDanio rerio\\t\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2017/05/09\\tGRCz11\\tGenome Reference Consortium\\tGCA_000002035.4\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/035/GCF_000002035.6_GRCz11\\t\\t\\n', 'GCF_000002075.1\\tPRJNA209509\\tSAMN02953658\\tAASC00000000.3\\trepresentative genome\\t6500\\t6500\\tAplysia californica\\t\\tF4 #8\\tlatest\\tScaffold\\tMajor\\tFull\\t2013/05/15\\tAplCal3.0\\tBroad Institute\\tGCA_000002075.2\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/075/GCF_000002075.1_AplCal3.0\\t\\t\\n', 'GCF_000002235.5\\tPRJNA13728\\tSAMN00829422\\tAAGJ00000000.6\\trepresentative genome\\t7668\\t7668\\tStrongylocentrotus purpuratus\\t\\tSpur 01\\tlatest\\tScaffold\\tMajor\\tFull\\t2019/09/06\\tSpur_5.0\\tBaylor College of Medicine\\tGCA_000002235.4\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/235/GCF_000002235.5_Spur_5.0\\t\\t\\n', 'GCF_000002285.5\\tPRJNA12384\\tSAMN02953603\\tAAEX00000000.4\\tna\\t9615\\t9612\\tCanis lupus familiaris\\tbreed=boxer\\tTasha\\tlatest\\tChromosome\\tMajor\\tFull\\t2020/10/06\\tDog10K_Boxer_Tasha\\tDog Genome Sequencing Consortium\\tGCA_000002285.4\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/285/GCF_000002285.5_Dog10K_Boxer_Tasha\\t\\t\\n', 'GCF_000002295.2\\tPRJNA15683\\tSAMN02953611\\tAAFR00000000.3\\trepresentative genome\\t13616\\t13616\\tMonodelphis domestica\\t\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2007/01/25\\tMonDom5\\tThe Genome Sequencing Platform, The Genome Assembly Team\\tna\\tna\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/295/GCF_000002295.2_MonDom5\\t\\t\\n', 'GCF_000002315.6\\tPRJNA10808\\tSAMN02981218\\tAADN00000000.5\\trepresentative genome\\t9031\\t9031\\tGallus gallus\\tbreed=Red Jungle fowl, inbred line UCD001\\tRJF #256\\tlatest\\tChromosome\\tMinor\\tFull\\t2018/03/27\\tGRCg6a\\tGenome Reference Consortium\\tGCA_000002315.5\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/315/GCF_000002315.6_GRCg6a\\t\\t\\n', 'GCF_000002335.3\\tPRJNA15718\\tSAMN02953634\\tAAJJ00000000.2\\trepresentative genome\\t7070\\t7070\\tTribolium castaneum\\tstrain=Georgia GA2\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2016/03/10\\tTcas5.2\\tKansas State University\\tGCA_000002335.3\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/335/GCF_000002335.3_Tcas5.2\\t\\t\\n', 'GCF_000002415.2\\tPRJNA20431\\tSAMN02953638\\tAAKM00000000.1\\trepresentative genome\\t5855\\t5855\\tPlasmodium vivax\\t\\tSalvador I\\tlatest\\tChromosome\\tMajor\\tFull\\t2009/05/06\\tASM241v2\\tTIGR\\tGCA_000002415.2\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/415/GCF_000002415.2_ASM241v2\\t\\t\\n', 'GCF_000002425.4\\tPRJNA28131\\tSAMN00120126\\tABEU00000000.2\\trepresentative genome\\t3218\\t3218\\tPhyscomitrium patens\\tecotype=Gransden 2004\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2018/01/24\\tPhypa V3\\tMoss Genome Consortium\\tGCA_000002425.2\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/425/GCF_000002425.4_Phypa_V3\\t\\t\\n', 'GCF_000002435.2\\tPRJNA15590\\tSAMN02952905\\tAACB00000000.3\\trepresentative genome\\t5741\\t5741\\tGiardia intestinalis\\tstrain=WB C6\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2019/11/06\\tUU_WB_2.1\\tMarine Biological Laboratory\\tGCA_000002435.2\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/435/GCF_000002435.2_UU_WB_2.1\\t\\t\\n', 'GCF_000002445.2\\tPRJNA15565\\tSAMN02953625\\t\\tna\\t185431\\t5691\\tTrypanosoma brucei brucei TREU927\\tstrain=927/4 GUTat10.1\\t927/4 GUTat10.1\\tlatest\\tChromosome\\tMajor\\tFull\\t2005/12/14\\tASM244v1\\tTrypanosoma brucei consortium\\tGCA_000002445.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/445/GCF_000002445.2_ASM244v1\\t\\t\\n', 'GCF_000002455.1\\tPRJNA27939\\t\\t\\tna\\t227086\\t227086\\tBigelowiella natans\\t\\t\\tlatest\\tChromosome\\tMajor\\tPartial\\t2006/07/28\\tASM245v1\\tUniversity of Melbourne, Australia\\tGCA_000002455.1\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/455/GCF_000002455.1_ASM245v1\\tpartial\\t\\n', 'GCF_000002495.2\\tPRJNA1433\\tSAMN02953596\\tAACU00000000.3\\trepresentative genome\\t242507\\t318829\\tPyricularia oryzae 70-15\\tstrain=70-15\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2011/10/14\\tMG8\\tInternational Rice Blast Genome Consortium\\tGCA_000002495.2\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/495/GCF_000002495.2_MG8\\t\\t\\n', 'GCF_000002515.2\\tPRJNA12377\\tSAMEA3138170\\t\\trepresentative genome\\t28985\\t28985\\tKluyveromyces lactis\\tstrain=NRRL Y-1140\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2004/07/02\\tASM251v1\\tGenolevures Consortium\\tGCA_000002515.1\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/515/GCF_000002515.2_ASM251v1\\t\\t\\n', 'GCF_000002525.2\\tPRJNA12414\\tSAMEA3138171\\t\\trepresentative genome\\t284591\\t4952\\tYarrowia lipolytica CLIB122\\tstrain=CLIB122\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2004/07/02\\tASM252v1\\tGenolevures Consortium\\tGCA_000002525.1\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/525/GCF_000002525.2_ASM252v1\\t\\t\\n', 'GCF_000002545.3\\tPRJNA12376\\tSAMEA3138172\\t\\trepresentative genome\\t5478\\t5478\\t[Candida] glabrata\\tstrain=CBS 138\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2008/08/12\\tASM254v2\\tGenolevures Consortium\\tGCA_000002545.2\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/545/GCF_000002545.3_ASM254v2\\t\\tassembly from type material\\n', 'GCF_000002595.1\\tPRJNA21061\\tSAMN02953692\\tABCN00000000.1\\trepresentative genome\\t3055\\t3055\\tChlamydomonas reinhardtii\\tstrain=CC-503 cw92 mt+\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2007/10/15\\tv3.0\\tDOE Joint Genome Institute\\tGCA_000002595.2\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/595/GCF_000002595.1_v3.0\\t\\t\\n', 'GCF_000002655.1\\tPRJNA14003\\tSAMN00115746\\tAAHF00000000.1\\trepresentative genome\\t330879\\t746128\\tAspergillus fumigatus Af293\\tstrain=Af293\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2005/06/10\\tASM265v1\\tJ. Craig Venter Institute\\tGCA_000002655.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/655/GCF_000002655.1_ASM265v1\\t\\t\\n', 'GCF_000002715.2\\tPRJNA18467\\tSAMN02953636\\tAAKD00000000.3\\trepresentative genome\\t344612\\t5057\\tAspergillus clavatus NRRL 1\\tstrain=NRRL 1\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2006/12/14\\tASM271v1\\tJ. Craig Venter Institute\\tGCA_000002715.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/715/GCF_000002715.2_ASM271v1\\t\\t\\n', 'GCF_000002725.2\\tPRJNA15564\\tSAMEA3138173\\t\\trepresentative genome\\t347515\\t5664\\tLeishmania major strain Friedlin\\tstrain=Friedlin\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2011/02/14\\tASM272v2\\tFriedlin Consortium\\tGCA_000002725.2\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/725/GCF_000002725.2_ASM272v2\\t\\t\\n', 'GCF_000002765.5\\tPRJNA148\\tSAMN00102897\\t\\trepresentative genome\\t36329\\t5833\\tPlasmodium falciparum 3D7\\t\\t3D7\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2016/04/07\\tGCA_000002765\\tPlasmodium falciparum Genome Sequencing Consortium\\tGCA_000002765.3\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/765/GCF_000002765.5_GCA_000002765\\t\\t\\n', 'GCF_000002775.4\\tPRJNA17973\\tSAMN02953657\\tAARH00000000.3\\trepresentative genome\\t3694\\t3694\\tPopulus trichocarpa\\t\\tNisqually-1\\tlatest\\tChromosome\\tMajor\\tFull\\t2018/01/24\\tPop_tri_v3\\tUS DOE Joint Genome Institute (JGI\\tGCA_000002775.3\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/775/GCF_000002775.4_Pop_tri_v3\\t\\t\\n', 'GCF_000002825.2\\tPRJNA18631\\tSAMN02953626\\tAAHC00000000.1\\trepresentative genome\\t412133\\t5722\\tTrichomonas vaginalis G3\\tstrain=G3; ATCC PRA-98\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2007/01/11\\tASM282v1\\tTIGR\\tGCA_000002825.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/825/GCF_000002825.2_ASM282v1\\t\\t\\n', 'GCF_000002845.2\\tPRJNA19185\\tSAMEA2272271\\tCADA00000000.1\\trepresentative genome\\t420245\\t5660\\tLeishmania braziliensis MHOM/BR/75/M2904\\tstrain=MHOM/BR/75/M2904\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2011/11/07\\tASM284v2\\tThe Sanger Institute\\tGCA_000002845.2\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/845/GCF_000002845.2_ASM284v2\\t\\t\\n', 'GCF_000002855.3\\tPRJNA19263\\tSAMEA3283178\\t\\trepresentative genome\\t425011\\t5061\\tAspergillus niger CBS 513.88\\t\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2007/04/11\\tASM285v2\\tDSM, The Netherlands\\tGCA_000002855.2\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/855/GCF_000002855.3_ASM285v2\\t\\t\\n', 'GCF_000002865.3\\tPRJNA28133\\tSAMN02953695\\tABFJ00000000.1\\trepresentative genome\\t431895\\t81824\\tMonosiga brevicollis MX1\\tstrain=MX1\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2007/12/20\\tV1.0\\tDOE Joint Genome Institute\\tGCA_000002865.1\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/865/GCF_000002865.3_V1.0\\t\\t\\n', 'GCF_000002875.2\\tPRJNA19187\\tSAMEA3138175\\tCACT00000000.1\\trepresentative genome\\t435258\\t5671\\tLeishmania infantum JPCM5\\tstrain=JPCM5\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2011/11/08\\tASM287v2\\tThe Sanger Institute\\tGCA_000002875.2\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/875/GCF_000002875.2_ASM287v2\\t\\t\\n', 'GCF_000002945.1\\tPRJNA127\\tSAMEA3138176\\t\\trepresentative genome\\t4896\\t4896\\tSchizosaccharomyces pombe\\tstrain=972h-\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2007/11/09\\tASM294v2\\tS. pombe European Sequencing Consortium (EUPOM)\\tGCA_000002945.2\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/945/GCF_000002945.1_ASM294v2\\t\\t\\n', 'GCF_000002975.1\\tPRJNA210\\tSAMEA3865277\\t\\tna\\t55529\\t55529\\tGuillardia theta\\t\\t\\tlatest\\tChromosome\\tMajor\\tPartial\\t2001/04/26\\tASM297v1\\tInstitute for Marine Biosciences, Halifax, Canada\\tGCA_000002975.1\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/975/GCF_000002975.1_ASM297v1\\tpartial\\t\\n', 'GCF_000002985.6\\tPRJNA158\\tSAMEA3138177\\t\\treference genome\\t6239\\t6239\\tCaenorhabditis elegans\\tstrain=Bristol N2\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2013/02/07\\tWBcel235\\tC. elegans Sequencing Consortium\\tGCA_000002985.3\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/985/GCF_000002985.6_WBcel235\\t\\t\\n', 'GCF_000002995.3\\tPRJNA27801\\tSAMN02953652\\tAAQA00000000.1\\trepresentative genome\\t6279\\t6279\\tBrugia malayi\\t\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2008/03/04\\tASM299v2\\tTIGR\\tGCA_000002995.2\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/002/995/GCF_000002995.3_ASM299v2\\t\\t\\n', 'GCF_000003025.6\\tPRJNA28993\\tSAMN02953785\\tAEMK00000000.2\\trepresentative genome\\t9823\\t9823\\tSus scrofa\\tbreed=Duroc\\tTJ Tabasco\\tlatest\\tChromosome\\tMajor\\tFull\\t2017/02/07\\tSscrofa11.1\\tThe Swine Genome Sequencing Consortium (SGSC)\\tGCA_000003025.6\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/003/025/GCF_000003025.6_Sscrofa11.1\\t\\t\\n', 'GCF_000003125.1\\tPRJNA38857\\tSAMN02953686\\tABAS00000000.1\\trepresentative genome\\t441959\\t28564\\tTalaromyces stipitatus ATCC 10500\\tstrain=ATCC 10500\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2008/12/22\\tJCVI-TSTA1-3.0\\tJ. Craig Venter Institute\\tGCA_000003125.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/003/125/GCF_000003125.1_JCVI-TSTA1-3.0\\t\\tassembly from type material\\n', 'GCF_000003135.1\\tPRJNA224116\\tSAMN00001475\\tACHI00000000.1\\tna\\t548480\\t216816\\tBifidobacterium longum subsp. longum ATCC 55813\\tstrain=ATCC 55813\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2009/05/05\\tASM313v1\\tBaylor College of Medicine\\tGCA_000003135.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/003/135/GCF_000003135.1_ASM313v1\\t\\t\\n', 'GCF_000003195.3\\tPRJNA38691\\tSAMN02953738\\tABXC00000000.3\\trepresentative genome\\t4558\\t4558\\tSorghum bicolor\\tcultivar=BTx623\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2017/04/07\\tSorghum_bicolor_NCBIv3\\tSorghum Consortium\\tGCA_000003195.3\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/003/195/GCF_000003195.3_Sorghum_bicolor_NCBIv3\\t\\t\\n', 'GCF_000003215.1\\tPRJNA224116\\tSAMN02470531\\tABFD00000000.2\\tna\\t455631\\t1496\\tClostridioides difficile QCD-66c26\\t\\tQCD-66c26\\tlatest\\tChromosome\\tMajor\\tFull\\t2009/01/30\\tASM321v1\\tMcGill University\\tGCA_000003215.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/003/215/GCF_000003215.1_ASM321v1\\t\\t\\n', 'GCF_000003225.4\\tPRJNA16308\\tSAMEA3138180\\t\\trepresentative genome\\t5874\\t5874\\tTheileria annulata\\tstrain=Ankara isolate clone C9\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2005/07/29\\tASM322v1\\tThe Sanger Institute\\tGCA_000003225.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/003/225/GCF_000003225.4_ASM322v1\\t\\t\\n', 'GCF_000003515.1\\tPRJNA39807\\tSAMN02953631\\tAAIW00000000.1\\trepresentative genome\\t336963\\t33188\\tUncinocarpus reesii 1704\\tstrain=1704\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2009/05/21\\tASM351v2\\tBroad Institute\\tGCA_000003515.2\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/003/515/GCF_000003515.1_ASM351v2\\t\\t\\n', 'GCF_000003605.2\\tPRJNA42857\\tSAMN00102747\\tACQM00000000.1\\trepresentative genome\\t10224\\t10224\\tSaccoglossus kowalevskii\\t\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2009/09/09\\tSkow_1.1\\tBaylor College of Medicine\\tGCA_000003605.1\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/003/605/GCF_000003605.2_Skow_1.1\\t\\t\\n', 'GCF_000003625.3\\tPRJNA42933\\tSAMN02953624\\tAAGW00000000.2\\trepresentative genome\\t9986\\t9986\\tOryctolagus cuniculus\\tbreed=Thorbecke inbred\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2009/10/20\\tOryCun2.0\\tThe Broad Institute of MIT and Harvard\\tGCA_000003625.1\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/003/625/GCF_000003625.3_OryCun2.0\\t\\t\\n', 'GCF_000003645.1\\tPRJNA224116\\tSAMN00717290\\tACLS00000000.1\\tna\\t526973\\t1396\\tBacillus cereus m1293\\tstrain=m1293\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2009/05/29\\tASM364v1\\tNaval Medical Research Center\\tGCA_000003645.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/003/645/GCF_000003645.1_ASM364v1\\t\\t\\n', 'GCF_000003745.3\\tPRJNA33471\\tSAMEA2272750\\tCAAP00000000.3\\trepresentative genome\\t29760\\t29760\\tVitis vinifera\\tcultivar=PN40024\\t\\tlatest\\tChromosome\\tMinor\\tFull\\t2009/12/07\\t12X\\tInternational Grape Genome Program\\tGCA_000003745.2\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/003/745/GCF_000003745.3_12X\\t\\t\\n', 'GCF_000003815.2\\tPRJNA33245\\tSAMN02953694\\tABEP00000000.3\\trepresentative genome\\t7739\\t7739\\tBranchiostoma floridae\\tstrain=S238N-H82\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2020/04/29\\tBfl_VNyyK\\tDOE Joint Genome Institute\\tGCA_000003815.2\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/003/815/GCF_000003815.2_Bfl_VNyyK\\t\\t\\n', 'GCF_000003835.1\\tPRJNA41079\\tSAMN02953613\\tAAFT00000000.1\\trepresentative genome\\t306902\\t36911\\tClavispora lusitaniae ATCC 42720\\tstrain=ATCC 42720\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2009/06/02\\tASM383v1\\tBroad Institute\\tGCA_000003835.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/003/835/GCF_000003835.1_ASM383v1\\t\\t\\n', 'GCF_000003855.2\\tPRJNA41099\\tSAMN02953742\\tACBU00000000.1\\trepresentative genome\\t559298\\t1681229\\tBlastomyces gilchristii SLH14081\\tstrain=SLH14081\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2009/02/24\\tBD_SLH14081_V1\\tBroad Institute\\tGCA_000003855.2\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/003/855/GCF_000003855.2_BD_SLH14081_V1\\t\\t\\n', 'GCF_000003925.1\\tPRJNA224116\\tSAMN00727646\\tACMU00000000.1\\tna\\t526997\\t1405\\tBacillus mycoides DSM 2048\\tstrain=DSM 2048\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2009/05/29\\tASM392v1\\tNaval Medical Research Center\\tGCA_000003925.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/003/925/GCF_000003925.1_ASM392v1\\t\\tassembly from type material\\n', 'GCF_000003955.1\\tPRJNA224116\\tSAMN00727677\\tACMT00000000.1\\tna\\t526994\\t1396\\tBacillus cereus AH1273\\tstrain=AH1273\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2009/05/29\\tASM395v1\\tNaval Medical Research Center\\tGCA_000003955.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/003/955/GCF_000003955.1_ASM395v1\\t\\t\\n', 'GCF_000004075.3\\tPRJNA182750\\tSAMN02953750\\tACHR00000000.3\\trepresentative genome\\t3659\\t3659\\tCucumis sativus\\tcultivar=9930\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2019/11/15\\tCucumber_9930_V3\\tThe Cucumber Genome Initiative\\tGCA_000004075.3\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/075/GCF_000004075.3_Cucumber_9930_V3\\t\\t\\n', 'GCF_000004095.1\\tPRJNA31231\\tSAMN00000081\\tACZU00000000.1\\trepresentative genome\\t6087\\t6087\\tHydra vulgaris\\tstrain=105\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2009/10/30\\tHydra_RP_1.0\\tJCVI\\tGCA_000004095.1\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/095/GCF_000004095.1_Hydra_RP_1.0\\t\\t\\n', 'GCF_000004155.1\\tPRJNA264109\\tSAMN02953762\\tACQJ00000000.2\\trepresentative genome\\t653667\\t866546\\tSchizosaccharomyces cryophilus OY26\\tstrain=OY26\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2013/07/31\\tSCY4\\tBroad Institute\\tGCA_000004155.2\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/155/GCF_000004155.1_SCY4\\t\\tassembly from type material\\n', 'GCF_000004195.4\\tPRJNA205740\\tSAMN13041969\\tAAMC00000000.4\\trepresentative genome\\t8364\\t8364\\tXenopus tropicalis\\tstrain=Nigerian\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2019/11/14\\tUCB_Xtro_10.0\\tUniversity of California, Berkeley\\tGCA_000004195.4\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/195/GCF_000004195.4_UCB_Xtro_10.0\\t\\t\\n', 'GCF_000004255.2\\tPRJNA49545\\tSAMN02981250\\tADBK00000000.1\\trepresentative genome\\t81972\\t59689\\tArabidopsis lyrata subsp. lyrata\\t\\t\\tlatest\\tScaffold\\tMinor\\tFull\\t2016/11/26\\tv.1.0\\tUS DOE Joint Genome Institute (JGI-PGF)\\tGCA_000004255.1\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/255/GCF_000004255.2_v.1.0\\t\\t\\n', 'GCF_000004515.5\\tPRJNA48389\\tSAMN00002965\\tACUP00000000.3\\trepresentative genome\\t3847\\t3847\\tGlycine max\\tcultivar=Williams 82\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2018/07/24\\tGlycine_max_v2.1\\tUS DOE Joint Genome Institute (JGI-PGF)\\tGCA_000004515.4\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/515/GCF_000004515.5_Glycine_max_v2.1\\t\\t\\n', 'GCF_000004555.1\\tPRJNA20855\\t\\tCAAC00000000.2\\trepresentative genome\\t6238\\t6238\\tCaenorhabditis briggsae\\tstrain=AF16\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2008/05/01\\tASM455v1\\tThe C.briggsae Sequencing Consortium\\tGCA_000004555.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/555/GCF_000004555.1_ASM455v1\\t\\t\\n', 'GCF_000004695.1\\tPRJNA13925\\tSAMN02953606\\tAAFI00000000.2\\trepresentative genome\\t352472\\t44689\\tDictyostelium discoideum AX4\\tstrain=AX4\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2009/08/24\\tdicty_2.7\\tThe Dictyostelium discoideum Sequencing Consortium\\tGCA_000004695.1\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/695/GCF_000004695.1_dicty_2.7\\t\\t\\n', 'GCF_000004825.1\\tPRJNA46447\\tSAMN02953767\\tADBJ00000000.1\\trepresentative genome\\t670386\\t2086695\\tHeterostelium album PN500\\tstrain=PN500\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2010/01/29\\tPolPal_Dec2009\\tLeibniz Institute for Age Research - Fritz Lipmann Institute\\tGCA_000004825.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/825/GCF_000004825.1_PolPal_Dec2009\\t\\t\\n', 'GCF_000004985.1\\tPRJNA43691\\tSAMN02953745\\tACER00000000.1\\trepresentative genome\\t744533\\t5762\\tNaegleria gruberi strain NEG-M\\tstrain=NEG-M\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2010/01/20\\tV1.0\\tUS DOE Joint Genome Institute (JGI-PGF)\\tGCA_000004985.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/004/985/GCF_000004985.1_V1.0\\t\\t\\n', 'GCF_000005155.2\\tPRJNA29983\\tSAMN02953650\\tAAPT00000000.1\\trepresentative genome\\t7222\\t7222\\tDrosophila grimshawi\\tstrain=TSC#15287-2541.00\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2006/06/30\\tdgri_caf1\\tAgencourt Bioscience Corporation\\tGCA_000005155.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/155/GCF_000005155.2_dgri_caf1\\t\\t\\n', 'GCF_000005175.2\\tPRJNA29977\\tSAMN02953651\\tAAPU00000000.1\\trepresentative genome\\t7230\\t7230\\tDrosophila mojavensis\\tstrain=TSC#15081-1352.22\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2006/06/30\\tdmoj_caf1\\tAgencourt Bioscience Corporation\\tGCA_000005175.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/175/GCF_000005175.2_dmoj_caf1\\t\\t\\n', 'GCF_000005505.3\\tPRJNA74771\\tSAMN02981254\\tADDN00000000.3\\trepresentative genome\\t15368\\t15368\\tBrachypodium distachyon\\tstrain=Bd21\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2018/01/24\\tBrachypodium_distachyon_v3.0\\tJGI-PGF\\tGCA_000005505.4\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/505/GCF_000005505.3_Brachypodium_distachyon_v3.0\\t\\t\\n', 'GCF_000005575.2\\tPRJNA163\\tSAMN02952903\\tAAAB00000000.1\\trepresentative genome\\t180454\\t7165\\tAnopheles gambiae str. PEST\\tstrain=PEST\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2006/10/16\\tAgamP3\\tThe International Consortium for the Sequencing of Anopheles Genome\\tGCA_000005575.1\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/575/GCF_000005575.2_AgamP3\\t\\t\\n', 'GCF_000005825.2\\tPRJNA224116\\tSAMN02603086\\t\\trepresentative genome\\t398511\\t79885\\tBacillus pseudofirmus OF4\\tstrain=OF4\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2010/12/15\\tASM582v2\\tCenter for Genomic Sciences, Allegheny-Singer Research Institute\\tGCA_000005825.2\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/825/GCF_000005825.2_ASM582v2\\t\\t\\n', 'GCF_000005845.2\\tPRJNA57779\\tSAMN02604091\\t\\treference genome\\t511145\\t562\\tEscherichia coli str. K-12 substr. MG1655\\tstrain=K-12 substr. MG1655\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2013/09/26\\tASM584v2\\tUniv. Wisconsin\\tGCA_000005845.2\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/845/GCF_000005845.2_ASM584v2\\t\\t\\n', 'GCF_000005925.1\\tPRJNA29997\\tSAMN02953653\\tAAQB00000000.1\\trepresentative genome\\t7260\\t7260\\tDrosophila willistoni\\tstrain=TSC#14030-0811.24\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2006/08/03\\tdwil_caf1\\tJ. Craig Venter Institute\\tGCA_000005925.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/005/925/GCF_000005925.1_dwil_caf1\\t\\t\\n', 'GCF_000006175.1\\tPRJNA224116\\tSAMN00000040\\t\\trepresentative genome\\t456320\\t2188\\tMethanococcus voltae A3\\tstrain=A3\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2010/06/03\\tASM617v2\\tUS DOE Joint Genome Institute (JGI-PGF)\\tGCA_000006175.2\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/175/GCF_000006175.1_ASM617v2\\t\\t\\n', 'GCF_000006295.1\\tPRJNA19807\\tSAMN02953682\\tAAZO00000000.1\\trepresentative genome\\t121224\\t121225\\tPediculus humanus corporis\\tstrain=USDA\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2007/04/23\\tJCVI_LOUSE_1.0\\tThe human body louse genome consortium\\tGCA_000006295.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/295/GCF_000006295.1_JCVI_LOUSE_1.0\\t\\t\\n', 'GCF_000006335.3\\tPRJNA39569\\tSAMN02953608\\tAAFN00000000.2\\trepresentative genome\\t294747\\t5482\\tCandida tropicalis MYA-3404\\tstrain=MYA-3404\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2009/06/18\\tASM633v3\\tBroad Institute\\tGCA_000006335.3\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/335/GCF_000006335.3_ASM633v3\\t\\t\\n', 'GCF_000006355.2\\tPRJNA33651\\tSAMEA2272753\\tCADCXE000000000.1\\trepresentative genome\\t5851\\t5850\\tPlasmodium knowlesi strain H\\tstrain=H\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2020/02/24\\tGCA_000006355.2\\tWellcome Trust Sanger Institute\\tGCA_000006355.3\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/355/GCF_000006355.2_GCA_000006355.2\\t\\t\\n', 'GCF_000006405.1\\tPRJNA46451\\tSAMN02953677\\tAAXJ00000000.1\\trepresentative genome\\t423536\\t31276\\tPerkinsus marinus ATCC 50983\\tstrain=ATCC 50983\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2009/06/12\\tJCVI_PMG_1.0\\tTIGR\\tGCA_000006405.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/405/GCF_000006405.1_JCVI_PMG_1.0\\t\\t\\n', 'GCF_000006425.1\\tPRJNA15585\\t\\tAAEL00000000.1\\trepresentative genome\\t353151\\t237895\\tCryptosporidium hominis TU502\\tstrain=TU502\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2004/10/27\\tASM642v1\\tVirginia Commonwealth University\\tGCA_000006425.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/425/GCF_000006425.1_ASM642v1\\t\\t\\n', 'GCF_000006445.2\\tPRJNA12410\\tSAMEA3138183\\t\\trepresentative genome\\t284592\\t4959\\tDebaryomyces hansenii CBS767\\tstrain=CBS767\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2008/09/16\\tASM644v2\\tGenolevures Consortium\\tGCA_000006445.2\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/445/GCF_000006445.2_ASM644v2\\t\\tassembly from type material\\n', 'GCF_000006515.1\\tPRJNA32283\\tSAMN02953683\\tAAZY00000000.2\\trepresentative genome\\t441375\\t5808\\tCryptosporidium muris RN66\\tstrain=RN66\\t\\tlatest\\tScaffold\\tMajor\\tFull\\t2008/07/23\\tJCVI-cmg-v1.0\\tTIGR\\tGCA_000006515.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/515/GCF_000006515.1_JCVI_cmg_v1.0\\t\\t\\n', 'GCF_000006565.2\\tPRJNA32719\\tSAMN00255192\\tABPA00000000.2\\trepresentative genome\\t508771\\t5811\\tToxoplasma gondii ME49\\tstrain=ME49\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2013/08/02\\tTGA4\\tJ. Craig Venter Institute\\tGCA_000006565.2\\tdifferent\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/565/GCF_000006565.2_TGA4\\t\\t\\n', 'GCF_000006605.1\\tPRJNA224116\\tSAMEA3283089\\t\\tna\\t306537\\t38289\\tCorynebacterium jeikeium K411\\tstrain=K411 = NCTC 11915\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2005/06/27\\tASM660v1\\tBielefeld Univ\\tGCA_000006605.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/605/GCF_000006605.1_ASM660v1\\t\\t\\n', 'GCF_000006625.1\\tPRJNA224116\\tSAMN02604098\\t\\tna\\t273119\\t134821\\tUreaplasma parvum serovar 3 str. ATCC 700970\\tstrain=ATCC 700970\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2003/07/17\\tASM662v1\\tUniv.AL-Birmingham\\tGCA_000006625.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/625/GCF_000006625.1_ASM662v1\\t\\t\\n', 'GCF_000006645.1\\tPRJNA224116\\tSAMN02604093\\t\\tna\\t187410\\t632\\tYersinia pestis KIM10+\\tstrain=KIM10+\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2002/07/29\\tASM664v1\\tUniv. Wisconsin\\tGCA_000006645.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/645/GCF_000006645.1_ASM664v1\\t\\t\\n', 'GCF_000006665.1\\tPRJNA224116\\tSAMN02604092\\t\\tna\\t155864\\t562\\tEscherichia coli O157:H7 str. EDL933\\tstrain=EDL933\\t\\tlatest\\tChromosome\\tMajor\\tFull\\t2004/12/06\\tASM666v1\\tUniv. Wisconsin\\tGCA_000006665.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/665/GCF_000006665.1_ASM666v1\\t\\t\\n', 'GCF_000006685.1\\tPRJNA224116\\tSAMN02603994\\t\\trepresentative genome\\t243161\\t83560\\tChlamydia muridarum str. Nigg\\tstrain=Nigg\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2003/03/26\\tASM668v1\\tTIGR\\tGCA_000006685.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/685/GCF_000006685.1_ASM668v1\\t\\tassembly from type material\\n', 'GCF_000006725.1\\tPRJNA224116\\tSAMN02603773\\t\\tna\\t160492\\t2371\\tXylella fastidiosa 9a5c\\tstrain=9a5c\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2004/06/04\\tASM672v1\\tSao Paulo state (Brazil) Consortium\\tGCA_000006725.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/725/GCF_000006725.1_ASM672v1\\t\\t\\n', 'GCF_000006745.1\\tPRJNA224116\\tSAMN02603969\\t\\tna\\t243277\\t666\\tVibrio cholerae O1 biovar El Tor str. N16961\\tstrain=N16961\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2001/01/09\\tASM674v1\\tTIGR\\tGCA_000006745.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/745/GCF_000006745.1_ASM674v1\\t\\t\\n', 'GCF_000006765.1\\tPRJNA57945\\tSAMN02603714\\t\\treference genome\\t208964\\t287\\tPseudomonas aeruginosa PAO1\\tstrain=PAO1\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2006/07/07\\tASM676v1\\tPathoGenesis Corporation\\tGCA_000006765.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/765/GCF_000006765.1_ASM676v1\\t\\t\\n', 'GCF_000006785.2\\tPRJNA224116\\tSAMN02604089\\t\\tna\\t160490\\t1314\\tStreptococcus pyogenes M1 GAS\\tstrain=SF370\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2014/04/01\\tASM678v2\\tUniv. Oklahoma\\tGCA_000006785.2\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/785/GCF_000006785.2_ASM678v2\\t\\t\\n', 'GCF_000006805.1\\tPRJNA224116\\tSAMN02604216\\t\\tna\\t64091\\t2242\\tHalobacterium salinarum NRC-1\\tstrain=NRC-1; ATCC 700922\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2001/01/09\\tASM680v1\\tUniversity of Massachusetts-Amherst, University of Washington\\tGCA_000006805.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/805/GCF_000006805.1_ASM680v1\\t\\t\\n', 'GCF_000006825.1\\tPRJNA224116\\tSAMN02604085\\t\\tna\\t272843\\t747\\tPasteurella multocida subsp. multocida str. Pm70\\tstrain=Pm70\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2001/03/21\\tASM682v1\\tUniv. Minnesota\\tGCA_000006825.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/825/GCF_000006825.1_ASM682v1\\t\\t\\n', 'GCF_000006845.1\\tPRJNA224116\\tSAMN02604088\\t\\tna\\t242231\\t485\\tNeisseria gonorrhoeae FA 1090\\tstrain=FA 1090\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2005/02/14\\tASM684v1\\tUniv. Oklahoma\\tGCA_000006845.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/845/GCF_000006845.1_ASM684v1\\t\\t\\n', 'GCF_000006865.1\\tPRJNA224116\\tSAMN02603339\\t\\tna\\t272623\\t1358\\tLactococcus lactis subsp. lactis Il1403\\tstrain=IL1403\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2001/03/21\\tASM686v1\\tINRA\\tGCA_000006865.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/865/GCF_000006865.1_ASM686v1\\t\\t\\n', 'GCF_000006885.1\\tPRJNA224116\\tSAMN02604002\\t\\tna\\t170187\\t1313\\tStreptococcus pneumoniae TIGR4\\tstrain=TIGR4; ATCC BAA-334\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2008/07/03\\tASM688v1\\tTIGR\\tGCA_000006885.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/885/GCF_000006885.1_ASM688v1\\t\\t\\n', 'GCF_000006905.1\\tPRJNA224116\\tSAMN02604005\\t\\tna\\t190650\\t155892\\tCaulobacter vibrioides CB15\\tstrain=CB15\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2002/03/06\\tASM690v1\\tTIGR\\tGCA_000006905.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/905/GCF_000006905.1_ASM690v1\\t\\t\\n', 'GCF_000006925.2\\tPRJNA62907\\tSAMN02603599\\t\\treference genome\\t198214\\t623\\tShigella flexneri 2a str. 301\\tstrain=301\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2011/08/03\\tASM692v2\\tMicrobial Genome Center of ChMPH\\tGCA_000006925.2\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/925/GCF_000006925.2_ASM692v2\\t\\t\\n', 'GCF_000006945.2\\tPRJNA57799\\tSAMN02604315\\t\\treference genome\\t99287\\t28901\\tSalmonella enterica subsp. enterica serovar Typhimurium str. LT2\\t\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2016/01/13\\tASM694v2\\tWashington University Genome Sequencing Center\\tGCA_000006945.2\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/945/GCF_000006945.2_ASM694v2\\t\\tassembly from type material\\n', 'GCF_000006965.1\\tPRJNA224116\\tSAMEA3283068\\t\\tna\\t266834\\t382\\tSinorhizobium meliloti 1021\\tstrain=1021\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2003/05/06\\tASM696v1\\tMELILO\\tGCA_000006965.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/965/GCF_000006965.1_ASM696v1\\t\\t\\n', 'GCF_000006985.1\\tPRJNA224116\\tSAMN02604006\\t\\trepresentative genome\\t194439\\t1097\\tChlorobaculum tepidum TLS\\tstrain=TLS\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2002/07/02\\tASM698v1\\tTIGR\\tGCA_000006985.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/006/985/GCF_000006985.1_ASM698v1\\t\\tassembly from type material\\n', 'GCF_000007005.1\\tPRJNA224116\\tSAMN02603227\\t\\tna\\t273057\\t2287\\tSaccharolobus solfataricus P2\\tstrain=P2\\t\\tlatest\\tComplete Genome\\tMajor\\tFull\\t2002/11/25\\tASM700v1\\tEuropean/Canadian Consortium\\tGCA_000007005.1\\tidentical\\tftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/007/005/GCF_000007005.1_ASM700v1\\t\\t\\n']\n"
     ]
    }
   ],
   "source": [
    "print(first[:100])"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 239,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Length mismatch: Expected axis has 1 elements, new values have 22 elements",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-239-fb4a2354da87>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[0;32m      9\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     10\u001b[0m \u001b[0massembly_sum\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'assembly_summary_refseq.txt'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m'unicode'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mheader\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mNone\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdelimiter\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m'\\t+'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mengine\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m'python'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 11\u001b[1;33m \u001b[0massembly_sum\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;34m\"assembly_accession\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"bioproject\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"biosample\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"wgs_master\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"refseq_category\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"taxid\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"species_taxid\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"organism_name\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"infraspecific_name\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"isolate\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"version_status\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"assembly_level\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"release_type\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"genome_rep\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"seq_rel_date\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"asm_name\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"submitter\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"gbrs_paired_asm\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"paired_asm_comp\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"ftp_path\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"excluded_from_refseq\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"relation_to_type_material\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32mD:\\anaconndaa\\lib\\site-packages\\pandas\\core\\generic.py\u001b[0m in \u001b[0;36m__setattr__\u001b[1;34m(self, name, value)\u001b[0m\n\u001b[0;32m   5285\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   5286\u001b[0m             \u001b[0mobject\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__getattribute__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 5287\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mobject\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__setattr__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   5288\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[0mAttributeError\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   5289\u001b[0m             \u001b[1;32mpass\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\properties.pyx\u001b[0m in \u001b[0;36mpandas._libs.properties.AxisProperty.__set__\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mD:\\anaconndaa\\lib\\site-packages\\pandas\\core\\generic.py\u001b[0m in \u001b[0;36m_set_axis\u001b[1;34m(self, axis, labels)\u001b[0m\n\u001b[0;32m    659\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    660\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_set_axis\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlabels\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m->\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 661\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mset_axis\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlabels\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    662\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_clear_item_cache\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    663\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mD:\\anaconndaa\\lib\\site-packages\\pandas\\core\\internals\\managers.py\u001b[0m in \u001b[0;36mset_axis\u001b[1;34m(self, axis, new_labels)\u001b[0m\n\u001b[0;32m    175\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    176\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mnew_len\u001b[0m \u001b[1;33m!=\u001b[0m \u001b[0mold_len\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 177\u001b[1;33m             raise ValueError(\n\u001b[0m\u001b[0;32m    178\u001b[0m                 \u001b[1;34mf\"Length mismatch: Expected axis has {old_len} elements, new \"\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    179\u001b[0m                 \u001b[1;34mf\"values have {new_len} elements\"\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mValueError\u001b[0m: Length mismatch: Expected axis has 1 elements, new values have 22 elements"
     ]
    }
   ],
   "source": [
    "\n",
    "assembly_summary_file=\"ftp://ftp.ncbi.nih.gov/genomes/refseq/assembly_summary_refseq.txt\"\n",
    "#Download the file using wget sysyem call\n",
    "subprocess.call(\"wget \"+assembly_summary_file, shell=True)\n",
    "#Reformat the file to pandas-friendly format\n",
    "subprocess.call(\"sed -i '1d' assembly_summary_refseq.txt\",shell=True)\n",
    "subprocess.call(\"sed -i 's/^# //' assembly_summary_refseq.txt\", shell=True)\n",
    "#Read the file as a dataframe - using read_table\n",
    "#Use read_table if the column separator is tab\n",
    "\n",
    "assembly_sum = pd.read_csv('assembly_summary_refseq.txt', dtype='unicode', header=None, delimiter='\\t+', engine='python')\n",
    "assembly_sum.columns = [\"assembly_accession\", \"bioproject\", \"biosample\", \"wgs_master\", \"refseq_category\", \"taxid\", \"species_taxid\", \"organism_name\", \"infraspecific_name\", \"isolate\", \"version_status\", \"assembly_level\", \"release_type\", \"genome_rep\", \"seq_rel_date\", \"asm_name\", \"submitter\", \"gbrs_paired_asm\", \"paired_asm_comp\", \"ftp_path\", \"excluded_from_refseq\", \"relation_to_type_material\"]\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 240,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th>0</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>#   See ftp://ftp.ncbi.nlm.nih.gov/genomes/README_assembly_summary.txt for a description of the columns in this file.</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th># assembly_accession</th>\n",
       "      <th>bioproject</th>\n",
       "      <th>biosample</th>\n",
       "      <th>wgs_master</th>\n",
       "      <th>refseq_category</th>\n",
       "      <th>taxid</th>\n",
       "      <th>species_taxid</th>\n",
       "      <th>organism_name</th>\n",
       "      <th>infraspecific_name</th>\n",
       "      <th>isolate</th>\n",
       "      <th>version_status</th>\n",
       "      <th>assembly_level</th>\n",
       "      <th>release_type</th>\n",
       "      <th>genome_rep</th>\n",
       "      <th>seq_rel_date</th>\n",
       "      <th>asm_name</th>\n",
       "      <th>submitter</th>\n",
       "      <th>gbrs_paired_asm</th>\n",
       "      <th>paired_asm_comp</th>\n",
       "      <th>ftp_path</th>\n",
       "      <th>excluded_from_refseq</th>\n",
       "      <td>relation_to_type_material</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000001215.4</th>\n",
       "      <th>PRJNA164</th>\n",
       "      <th>SAMN02803731</th>\n",
       "      <th>reference genome</th>\n",
       "      <th>7227</th>\n",
       "      <th>7227</th>\n",
       "      <th>Drosophila melanogaster</th>\n",
       "      <th>latest</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Major</th>\n",
       "      <th>Full</th>\n",
       "      <th>2014/08/01</th>\n",
       "      <th>Release 6 plus ISO1 MT</th>\n",
       "      <th>The FlyBase Consortium/Berkeley Drosophila Genome Project/Celera Genomics</th>\n",
       "      <th>GCA_000001215.4</th>\n",
       "      <th>identical</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/215/GCF_000001215.4_Release_6_plus_ISO1_MT</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000001405.39</th>\n",
       "      <th>PRJNA168</th>\n",
       "      <th>reference genome</th>\n",
       "      <th>9606</th>\n",
       "      <th>9606</th>\n",
       "      <th>Homo sapiens</th>\n",
       "      <th>latest</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Patch</th>\n",
       "      <th>Full</th>\n",
       "      <th>2019/02/28</th>\n",
       "      <th>GRCh38.p13</th>\n",
       "      <th>Genome Reference Consortium</th>\n",
       "      <th>GCA_000001405.28</th>\n",
       "      <th>different</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/405/GCF_000001405.39_GRCh38.p13</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_000001635.27</th>\n",
       "      <th>PRJNA169</th>\n",
       "      <th>reference genome</th>\n",
       "      <th>10090</th>\n",
       "      <th>10090</th>\n",
       "      <th>Mus musculus</th>\n",
       "      <th>latest</th>\n",
       "      <th>Chromosome</th>\n",
       "      <th>Major</th>\n",
       "      <th>Full</th>\n",
       "      <th>2020/06/24</th>\n",
       "      <th>GRCm39</th>\n",
       "      <th>Genome Reference Consortium</th>\n",
       "      <th>GCA_000001635.9</th>\n",
       "      <th>identical</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/001/635/GCF_000001635.27_GRCm39</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_905236995.1</th>\n",
       "      <th>PRJNA224116</th>\n",
       "      <th>SAMEA7833113</th>\n",
       "      <th>CAJNNO000000000.1</th>\n",
       "      <th>na</th>\n",
       "      <th>150055</th>\n",
       "      <th>150055</th>\n",
       "      <th>Streptococcus lutetiensis</th>\n",
       "      <th>BR_UFV01</th>\n",
       "      <th>latest</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Major</th>\n",
       "      <th>Full</th>\n",
       "      <th>2021/02/20</th>\n",
       "      <th>ASM90523699v1</th>\n",
       "      <th>Universidade Federal de Vicosa</th>\n",
       "      <th>GCA_905236995.1</th>\n",
       "      <th>identical</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/236/995/GCF_905236995.1_ASM90523699v1</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_905237015.1</th>\n",
       "      <th>PRJNA224116</th>\n",
       "      <th>SAMEA7833114</th>\n",
       "      <th>CAJNNP000000000.1</th>\n",
       "      <th>na</th>\n",
       "      <th>150055</th>\n",
       "      <th>150055</th>\n",
       "      <th>Streptococcus lutetiensis</th>\n",
       "      <th>BR_UFV02</th>\n",
       "      <th>latest</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Major</th>\n",
       "      <th>Full</th>\n",
       "      <th>2021/02/20</th>\n",
       "      <th>ASM90523701v1</th>\n",
       "      <th>Universidade Federal de Vicosa</th>\n",
       "      <th>GCA_905237015.1</th>\n",
       "      <th>identical</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/237/015/GCF_905237015.1_ASM90523701v1</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_905237025.1</th>\n",
       "      <th>PRJNA224116</th>\n",
       "      <th>SAMEA7833116</th>\n",
       "      <th>CAJNNS000000000.1</th>\n",
       "      <th>na</th>\n",
       "      <th>150055</th>\n",
       "      <th>150055</th>\n",
       "      <th>Streptococcus lutetiensis</th>\n",
       "      <th>BR_UFV04</th>\n",
       "      <th>latest</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Major</th>\n",
       "      <th>Full</th>\n",
       "      <th>2021/02/20</th>\n",
       "      <th>ASM90523702v1</th>\n",
       "      <th>Universidade Federal de Vicosa</th>\n",
       "      <th>GCA_905237025.1</th>\n",
       "      <th>identical</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/237/025/GCF_905237025.1_ASM90523702v1</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_905237035.1</th>\n",
       "      <th>PRJNA224116</th>\n",
       "      <th>SAMEA7833115</th>\n",
       "      <th>CAJNNR000000000.1</th>\n",
       "      <th>na</th>\n",
       "      <th>150055</th>\n",
       "      <th>150055</th>\n",
       "      <th>Streptococcus lutetiensis</th>\n",
       "      <th>BR_UFV03</th>\n",
       "      <th>latest</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Major</th>\n",
       "      <th>Full</th>\n",
       "      <th>2021/02/20</th>\n",
       "      <th>ASM90523703v1</th>\n",
       "      <th>Universidade Federal de Vicosa</th>\n",
       "      <th>GCA_905237035.1</th>\n",
       "      <th>identical</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/237/035/GCF_905237035.1_ASM90523703v1</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>GCF_905237045.1</th>\n",
       "      <th>PRJNA224116</th>\n",
       "      <th>SAMEA7833117</th>\n",
       "      <th>CAJNNQ000000000.1</th>\n",
       "      <th>na</th>\n",
       "      <th>150055</th>\n",
       "      <th>150055</th>\n",
       "      <th>Streptococcus lutetiensis</th>\n",
       "      <th>BR_UFV05</th>\n",
       "      <th>latest</th>\n",
       "      <th>Contig</th>\n",
       "      <th>Major</th>\n",
       "      <th>Full</th>\n",
       "      <th>2021/02/20</th>\n",
       "      <th>ASM90523704v1</th>\n",
       "      <th>Universidade Federal de Vicosa</th>\n",
       "      <th>GCA_905237045.1</th>\n",
       "      <th>identical</th>\n",
       "      <th>ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/237/045/GCF_905237045.1_ASM90523704v1</th>\n",
       "      <th>NaN</th>\n",
       "      <th>NaN</th>\n",
       "      <td>None</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>222707 rows × 1 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                         0\n",
       "#   See ftp://ftp.ncbi.nlm.nih.gov/genomes/READ... NaN         NaN              NaN               NaN             NaN          NaN                     NaN                       NaN                NaN     NaN            NaN            NaN                         NaN                                                NaN             NaN                                                NaN                                                NaN             NaN                                                NaN      NaN                                        None\n",
       "# assembly_accession                               bioproject  biosample        wgs_master        refseq_category taxid        species_taxid           organism_name             infraspecific_name isolate version_status assembly_level release_type                genome_rep                                         seq_rel_date    asm_name                                           submitter                                          gbrs_paired_asm paired_asm_comp                                    ftp_path excluded_from_refseq  relation_to_type_material\n",
       "GCF_000001215.4                                    PRJNA164    SAMN02803731     reference genome  7227            7227         Drosophila melanogaster latest                    Chromosome         Major   Full           2014/08/01     Release 6 plus ISO1 MT      The FlyBase Consortium/Berkeley Drosophila Geno... GCA_000001215.4 identical                                          ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/... NaN             NaN                                                NaN      NaN                                        None\n",
       "GCF_000001405.39                                   PRJNA168    reference genome 9606              9606            Homo sapiens latest                  Chromosome                Patch              Full    2019/02/28     GRCh38.p13     Genome Reference Consortium GCA_000001405.28                                   different       ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/... NaN                                                NaN             NaN                                                NaN      NaN                                        None\n",
       "GCF_000001635.27                                   PRJNA169    reference genome 10090             10090           Mus musculus latest                  Chromosome                Major              Full    2020/06/24     GRCm39         Genome Reference Consortium GCA_000001635.9                                    identical       ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/000/... NaN                                                NaN             NaN                                                NaN      NaN                                        None\n",
       "...                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                    ...\n",
       "GCF_905236995.1                                    PRJNA224116 SAMEA7833113     CAJNNO000000000.1 na              150055       150055                  Streptococcus lutetiensis BR_UFV01           latest  Contig         Major          Full                        2021/02/20                                         ASM90523699v1   Universidade Federal de Vicosa                     GCA_905236995.1                                    identical       ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/... NaN      NaN                                        None\n",
       "GCF_905237015.1                                    PRJNA224116 SAMEA7833114     CAJNNP000000000.1 na              150055       150055                  Streptococcus lutetiensis BR_UFV02           latest  Contig         Major          Full                        2021/02/20                                         ASM90523701v1   Universidade Federal de Vicosa                     GCA_905237015.1                                    identical       ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/... NaN      NaN                                        None\n",
       "GCF_905237025.1                                    PRJNA224116 SAMEA7833116     CAJNNS000000000.1 na              150055       150055                  Streptococcus lutetiensis BR_UFV04           latest  Contig         Major          Full                        2021/02/20                                         ASM90523702v1   Universidade Federal de Vicosa                     GCA_905237025.1                                    identical       ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/... NaN      NaN                                        None\n",
       "GCF_905237035.1                                    PRJNA224116 SAMEA7833115     CAJNNR000000000.1 na              150055       150055                  Streptococcus lutetiensis BR_UFV03           latest  Contig         Major          Full                        2021/02/20                                         ASM90523703v1   Universidade Federal de Vicosa                     GCA_905237035.1                                    identical       ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/... NaN      NaN                                        None\n",
       "GCF_905237045.1                                    PRJNA224116 SAMEA7833117     CAJNNQ000000000.1 na              150055       150055                  Streptococcus lutetiensis BR_UFV05           latest  Contig         Major          Full                        2021/02/20                                         ASM90523704v1   Universidade Federal de Vicosa                     GCA_905237045.1                                    identical       ftp://ftp.ncbi.nlm.nih.gov/genomes/all/GCF/905/... NaN      NaN                                        None\n",
       "\n",
       "[222707 rows x 1 columns]"
      ]
     },
     "execution_count": 240,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "assembly_sum\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def get_this(asb, outfile='refseq_genome.txt'):\n",
    "for id in asb:\n",
    "        my_df=assembly_sum[(assembly_sum['assembly_accession'] == id) &\n",
    "                           ((assembly_sum['refseq_category'] == 'reference genome') |\n",
    "                            (assembly_sum['refseq_category'] == 'representative genome')\n",
    "                           )]\n",
    "        my_df=my_df[['ftp_path','assembly_accession','asm_name']]\n",
    "        #Process the newly created file and download genomes from NCBI website\n",
    "        my_df.to_csv(outfile,mode='w',index=False,header=None)\n",
    "        process_url_file(outfile)\n",
    "    return"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 238,
   "metadata": {},
   "outputs": [
    {
     "ename": "ValueError",
     "evalue": "Length mismatch: Expected axis has 1 elements, new values have 22 elements",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mValueError\u001b[0m                                Traceback (most recent call last)",
      "\u001b[1;32m<ipython-input-238-a3ffbd3114e3>\u001b[0m in \u001b[0;36m<module>\u001b[1;34m\u001b[0m\n\u001b[1;32m----> 1\u001b[1;33m \u001b[0mget_this\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0masb\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0moutfile\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m'refseq_genome.txt'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[1;32m<ipython-input-237-36f57639c63b>\u001b[0m in \u001b[0;36mget_this\u001b[1;34m(asb, outfile)\u001b[0m\n\u001b[0;32m     10\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     11\u001b[0m     \u001b[0massembly_sum\u001b[0m \u001b[1;33m=\u001b[0m \u001b[0mpd\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mread_csv\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;34m'assembly_summary_refseq.txt'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdtype\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m'unicode'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mheader\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;32mNone\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mdelimiter\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m'\\t+'\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mengine\u001b[0m\u001b[1;33m=\u001b[0m\u001b[1;34m'python'\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m---> 12\u001b[1;33m     \u001b[0massembly_sum\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mcolumns\u001b[0m \u001b[1;33m=\u001b[0m \u001b[1;33m[\u001b[0m\u001b[1;34m\"assembly_accession\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"bioproject\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"biosample\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"wgs_master\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"refseq_category\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"taxid\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"species_taxid\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"organism_name\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"infraspecific_name\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"isolate\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"version_status\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"assembly_level\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"release_type\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"genome_rep\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"seq_rel_date\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"asm_name\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"submitter\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"gbrs_paired_asm\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"paired_asm_comp\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"ftp_path\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"excluded_from_refseq\"\u001b[0m\u001b[1;33m,\u001b[0m \u001b[1;34m\"relation_to_type_material\"\u001b[0m\u001b[1;33m]\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m     13\u001b[0m     \u001b[1;32mfor\u001b[0m \u001b[0mid\u001b[0m \u001b[1;32min\u001b[0m \u001b[0masb\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m     14\u001b[0m         my_df=assembly_sum[(assembly_sum['assembly_accession'] == id) &\n",
      "\u001b[1;32mD:\\anaconndaa\\lib\\site-packages\\pandas\\core\\generic.py\u001b[0m in \u001b[0;36m__setattr__\u001b[1;34m(self, name, value)\u001b[0m\n\u001b[0;32m   5285\u001b[0m         \u001b[1;32mtry\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   5286\u001b[0m             \u001b[0mobject\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__getattribute__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m-> 5287\u001b[1;33m             \u001b[1;32mreturn\u001b[0m \u001b[0mobject\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m__setattr__\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mname\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mvalue\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m   5288\u001b[0m         \u001b[1;32mexcept\u001b[0m \u001b[0mAttributeError\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m   5289\u001b[0m             \u001b[1;32mpass\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mpandas\\_libs\\properties.pyx\u001b[0m in \u001b[0;36mpandas._libs.properties.AxisProperty.__set__\u001b[1;34m()\u001b[0m\n",
      "\u001b[1;32mD:\\anaconndaa\\lib\\site-packages\\pandas\\core\\generic.py\u001b[0m in \u001b[0;36m_set_axis\u001b[1;34m(self, axis, labels)\u001b[0m\n\u001b[0;32m    659\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    660\u001b[0m     \u001b[1;32mdef\u001b[0m \u001b[0m_set_axis\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0mself\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0maxis\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlabels\u001b[0m\u001b[1;33m)\u001b[0m \u001b[1;33m->\u001b[0m \u001b[1;32mNone\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 661\u001b[1;33m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_data\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0mset_axis\u001b[0m\u001b[1;33m(\u001b[0m\u001b[0maxis\u001b[0m\u001b[1;33m,\u001b[0m \u001b[0mlabels\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0m\u001b[0;32m    662\u001b[0m         \u001b[0mself\u001b[0m\u001b[1;33m.\u001b[0m\u001b[0m_clear_item_cache\u001b[0m\u001b[1;33m(\u001b[0m\u001b[1;33m)\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    663\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;32mD:\\anaconndaa\\lib\\site-packages\\pandas\\core\\internals\\managers.py\u001b[0m in \u001b[0;36mset_axis\u001b[1;34m(self, axis, new_labels)\u001b[0m\n\u001b[0;32m    175\u001b[0m \u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    176\u001b[0m         \u001b[1;32mif\u001b[0m \u001b[0mnew_len\u001b[0m \u001b[1;33m!=\u001b[0m \u001b[0mold_len\u001b[0m\u001b[1;33m:\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[1;32m--> 177\u001b[1;33m             raise ValueError(\n\u001b[0m\u001b[0;32m    178\u001b[0m                 \u001b[1;34mf\"Length mismatch: Expected axis has {old_len} elements, new \"\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n\u001b[0;32m    179\u001b[0m                 \u001b[1;34mf\"values have {new_len} elements\"\u001b[0m\u001b[1;33m\u001b[0m\u001b[1;33m\u001b[0m\u001b[0m\n",
      "\u001b[1;31mValueError\u001b[0m: Length mismatch: Expected axis has 1 elements, new values have 22 elements"
     ]
    }
   ],
   "source": [
    "get_this(asb, outfile='refseq_genome.txt')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.8.3"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
